// @aws-sdk/client-ecs/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var utilWaiter = require('@smithy/util-waiter');
var errors = require('./models/errors');
var ECSServiceException = require('./models/ECSServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "ecs",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class ECSClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultECSHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class CreateCapacityProviderCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "CreateCapacityProvider", {})
    .n("ECSClient", "CreateCapacityProviderCommand")
    .sc(schemas_0.CreateCapacityProvider$)
    .build() {
}

class CreateClusterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "CreateCluster", {})
    .n("ECSClient", "CreateClusterCommand")
    .sc(schemas_0.CreateCluster$)
    .build() {
}

class CreateDaemonCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "CreateDaemon", {})
    .n("ECSClient", "CreateDaemonCommand")
    .sc(schemas_0.CreateDaemon$)
    .build() {
}

class CreateExpressGatewayServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "CreateExpressGatewayService", {})
    .n("ECSClient", "CreateExpressGatewayServiceCommand")
    .sc(schemas_0.CreateExpressGatewayService$)
    .build() {
}

class CreateServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "CreateService", {})
    .n("ECSClient", "CreateServiceCommand")
    .sc(schemas_0.CreateService$)
    .build() {
}

class CreateTaskSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "CreateTaskSet", {})
    .n("ECSClient", "CreateTaskSetCommand")
    .sc(schemas_0.CreateTaskSet$)
    .build() {
}

class DeleteAccountSettingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteAccountSetting", {})
    .n("ECSClient", "DeleteAccountSettingCommand")
    .sc(schemas_0.DeleteAccountSetting$)
    .build() {
}

class DeleteAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteAttributes", {})
    .n("ECSClient", "DeleteAttributesCommand")
    .sc(schemas_0.DeleteAttributes$)
    .build() {
}

class DeleteCapacityProviderCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteCapacityProvider", {})
    .n("ECSClient", "DeleteCapacityProviderCommand")
    .sc(schemas_0.DeleteCapacityProvider$)
    .build() {
}

class DeleteClusterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteCluster", {})
    .n("ECSClient", "DeleteClusterCommand")
    .sc(schemas_0.DeleteCluster$)
    .build() {
}

class DeleteDaemonCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteDaemon", {})
    .n("ECSClient", "DeleteDaemonCommand")
    .sc(schemas_0.DeleteDaemon$)
    .build() {
}

class DeleteDaemonTaskDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteDaemonTaskDefinition", {})
    .n("ECSClient", "DeleteDaemonTaskDefinitionCommand")
    .sc(schemas_0.DeleteDaemonTaskDefinition$)
    .build() {
}

class DeleteExpressGatewayServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteExpressGatewayService", {})
    .n("ECSClient", "DeleteExpressGatewayServiceCommand")
    .sc(schemas_0.DeleteExpressGatewayService$)
    .build() {
}

class DeleteServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteService", {})
    .n("ECSClient", "DeleteServiceCommand")
    .sc(schemas_0.DeleteService$)
    .build() {
}

class DeleteTaskDefinitionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteTaskDefinitions", {})
    .n("ECSClient", "DeleteTaskDefinitionsCommand")
    .sc(schemas_0.DeleteTaskDefinitions$)
    .build() {
}

class DeleteTaskSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeleteTaskSet", {})
    .n("ECSClient", "DeleteTaskSetCommand")
    .sc(schemas_0.DeleteTaskSet$)
    .build() {
}

class DeregisterContainerInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeregisterContainerInstance", {})
    .n("ECSClient", "DeregisterContainerInstanceCommand")
    .sc(schemas_0.DeregisterContainerInstance$)
    .build() {
}

class DeregisterTaskDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DeregisterTaskDefinition", {})
    .n("ECSClient", "DeregisterTaskDefinitionCommand")
    .sc(schemas_0.DeregisterTaskDefinition$)
    .build() {
}

class DescribeCapacityProvidersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeCapacityProviders", {})
    .n("ECSClient", "DescribeCapacityProvidersCommand")
    .sc(schemas_0.DescribeCapacityProviders$)
    .build() {
}

class DescribeClustersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeClusters", {})
    .n("ECSClient", "DescribeClustersCommand")
    .sc(schemas_0.DescribeClusters$)
    .build() {
}

class DescribeContainerInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeContainerInstances", {})
    .n("ECSClient", "DescribeContainerInstancesCommand")
    .sc(schemas_0.DescribeContainerInstances$)
    .build() {
}

class DescribeDaemonCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeDaemon", {})
    .n("ECSClient", "DescribeDaemonCommand")
    .sc(schemas_0.DescribeDaemon$)
    .build() {
}

class DescribeDaemonDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeDaemonDeployments", {})
    .n("ECSClient", "DescribeDaemonDeploymentsCommand")
    .sc(schemas_0.DescribeDaemonDeployments$)
    .build() {
}

class DescribeDaemonRevisionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeDaemonRevisions", {})
    .n("ECSClient", "DescribeDaemonRevisionsCommand")
    .sc(schemas_0.DescribeDaemonRevisions$)
    .build() {
}

class DescribeDaemonTaskDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeDaemonTaskDefinition", {})
    .n("ECSClient", "DescribeDaemonTaskDefinitionCommand")
    .sc(schemas_0.DescribeDaemonTaskDefinition$)
    .build() {
}

class DescribeExpressGatewayServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeExpressGatewayService", {})
    .n("ECSClient", "DescribeExpressGatewayServiceCommand")
    .sc(schemas_0.DescribeExpressGatewayService$)
    .build() {
}

class DescribeServiceDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeServiceDeployments", {})
    .n("ECSClient", "DescribeServiceDeploymentsCommand")
    .sc(schemas_0.DescribeServiceDeployments$)
    .build() {
}

class DescribeServiceRevisionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeServiceRevisions", {})
    .n("ECSClient", "DescribeServiceRevisionsCommand")
    .sc(schemas_0.DescribeServiceRevisions$)
    .build() {
}

class DescribeServicesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeServices", {})
    .n("ECSClient", "DescribeServicesCommand")
    .sc(schemas_0.DescribeServices$)
    .build() {
}

class DescribeTaskDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeTaskDefinition", {})
    .n("ECSClient", "DescribeTaskDefinitionCommand")
    .sc(schemas_0.DescribeTaskDefinition$)
    .build() {
}

class DescribeTasksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeTasks", {})
    .n("ECSClient", "DescribeTasksCommand")
    .sc(schemas_0.DescribeTasks$)
    .build() {
}

class DescribeTaskSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DescribeTaskSets", {})
    .n("ECSClient", "DescribeTaskSetsCommand")
    .sc(schemas_0.DescribeTaskSets$)
    .build() {
}

class DiscoverPollEndpointCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "DiscoverPollEndpoint", {})
    .n("ECSClient", "DiscoverPollEndpointCommand")
    .sc(schemas_0.DiscoverPollEndpoint$)
    .build() {
}

class ExecuteCommandCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ExecuteCommand", {})
    .n("ECSClient", "ExecuteCommandCommand")
    .sc(schemas_0.ExecuteCommand$)
    .build() {
}

class GetTaskProtectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "GetTaskProtection", {})
    .n("ECSClient", "GetTaskProtectionCommand")
    .sc(schemas_0.GetTaskProtection$)
    .build() {
}

class ListAccountSettingsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListAccountSettings", {})
    .n("ECSClient", "ListAccountSettingsCommand")
    .sc(schemas_0.ListAccountSettings$)
    .build() {
}

class ListAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListAttributes", {})
    .n("ECSClient", "ListAttributesCommand")
    .sc(schemas_0.ListAttributes$)
    .build() {
}

class ListClustersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListClusters", {})
    .n("ECSClient", "ListClustersCommand")
    .sc(schemas_0.ListClusters$)
    .build() {
}

class ListContainerInstancesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListContainerInstances", {})
    .n("ECSClient", "ListContainerInstancesCommand")
    .sc(schemas_0.ListContainerInstances$)
    .build() {
}

class ListDaemonDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListDaemonDeployments", {})
    .n("ECSClient", "ListDaemonDeploymentsCommand")
    .sc(schemas_0.ListDaemonDeployments$)
    .build() {
}

class ListDaemonsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListDaemons", {})
    .n("ECSClient", "ListDaemonsCommand")
    .sc(schemas_0.ListDaemons$)
    .build() {
}

class ListDaemonTaskDefinitionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListDaemonTaskDefinitions", {})
    .n("ECSClient", "ListDaemonTaskDefinitionsCommand")
    .sc(schemas_0.ListDaemonTaskDefinitions$)
    .build() {
}

class ListServiceDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListServiceDeployments", {})
    .n("ECSClient", "ListServiceDeploymentsCommand")
    .sc(schemas_0.ListServiceDeployments$)
    .build() {
}

class ListServicesByNamespaceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListServicesByNamespace", {})
    .n("ECSClient", "ListServicesByNamespaceCommand")
    .sc(schemas_0.ListServicesByNamespace$)
    .build() {
}

class ListServicesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListServices", {})
    .n("ECSClient", "ListServicesCommand")
    .sc(schemas_0.ListServices$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListTagsForResource", {})
    .n("ECSClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListTaskDefinitionFamiliesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListTaskDefinitionFamilies", {})
    .n("ECSClient", "ListTaskDefinitionFamiliesCommand")
    .sc(schemas_0.ListTaskDefinitionFamilies$)
    .build() {
}

class ListTaskDefinitionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListTaskDefinitions", {})
    .n("ECSClient", "ListTaskDefinitionsCommand")
    .sc(schemas_0.ListTaskDefinitions$)
    .build() {
}

class ListTasksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "ListTasks", {})
    .n("ECSClient", "ListTasksCommand")
    .sc(schemas_0.ListTasks$)
    .build() {
}

class PutAccountSettingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "PutAccountSetting", {})
    .n("ECSClient", "PutAccountSettingCommand")
    .sc(schemas_0.PutAccountSetting$)
    .build() {
}

class PutAccountSettingDefaultCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "PutAccountSettingDefault", {})
    .n("ECSClient", "PutAccountSettingDefaultCommand")
    .sc(schemas_0.PutAccountSettingDefault$)
    .build() {
}

class PutAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "PutAttributes", {})
    .n("ECSClient", "PutAttributesCommand")
    .sc(schemas_0.PutAttributes$)
    .build() {
}

class PutClusterCapacityProvidersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "PutClusterCapacityProviders", {})
    .n("ECSClient", "PutClusterCapacityProvidersCommand")
    .sc(schemas_0.PutClusterCapacityProviders$)
    .build() {
}

class RegisterContainerInstanceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "RegisterContainerInstance", {})
    .n("ECSClient", "RegisterContainerInstanceCommand")
    .sc(schemas_0.RegisterContainerInstance$)
    .build() {
}

class RegisterDaemonTaskDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "RegisterDaemonTaskDefinition", {})
    .n("ECSClient", "RegisterDaemonTaskDefinitionCommand")
    .sc(schemas_0.RegisterDaemonTaskDefinition$)
    .build() {
}

class RegisterTaskDefinitionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "RegisterTaskDefinition", {})
    .n("ECSClient", "RegisterTaskDefinitionCommand")
    .sc(schemas_0.RegisterTaskDefinition$)
    .build() {
}

class RunTaskCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "RunTask", {})
    .n("ECSClient", "RunTaskCommand")
    .sc(schemas_0.RunTask$)
    .build() {
}

class StartTaskCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "StartTask", {})
    .n("ECSClient", "StartTaskCommand")
    .sc(schemas_0.StartTask$)
    .build() {
}

class StopServiceDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "StopServiceDeployment", {})
    .n("ECSClient", "StopServiceDeploymentCommand")
    .sc(schemas_0.StopServiceDeployment$)
    .build() {
}

class StopTaskCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "StopTask", {})
    .n("ECSClient", "StopTaskCommand")
    .sc(schemas_0.StopTask$)
    .build() {
}

class SubmitAttachmentStateChangesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "SubmitAttachmentStateChanges", {})
    .n("ECSClient", "SubmitAttachmentStateChangesCommand")
    .sc(schemas_0.SubmitAttachmentStateChanges$)
    .build() {
}

class SubmitContainerStateChangeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "SubmitContainerStateChange", {})
    .n("ECSClient", "SubmitContainerStateChangeCommand")
    .sc(schemas_0.SubmitContainerStateChange$)
    .build() {
}

class SubmitTaskStateChangeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "SubmitTaskStateChange", {})
    .n("ECSClient", "SubmitTaskStateChangeCommand")
    .sc(schemas_0.SubmitTaskStateChange$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "TagResource", {})
    .n("ECSClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UntagResource", {})
    .n("ECSClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateCapacityProviderCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateCapacityProvider", {})
    .n("ECSClient", "UpdateCapacityProviderCommand")
    .sc(schemas_0.UpdateCapacityProvider$)
    .build() {
}

class UpdateClusterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateCluster", {})
    .n("ECSClient", "UpdateClusterCommand")
    .sc(schemas_0.UpdateCluster$)
    .build() {
}

class UpdateClusterSettingsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateClusterSettings", {})
    .n("ECSClient", "UpdateClusterSettingsCommand")
    .sc(schemas_0.UpdateClusterSettings$)
    .build() {
}

class UpdateContainerAgentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateContainerAgent", {})
    .n("ECSClient", "UpdateContainerAgentCommand")
    .sc(schemas_0.UpdateContainerAgent$)
    .build() {
}

class UpdateContainerInstancesStateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateContainerInstancesState", {})
    .n("ECSClient", "UpdateContainerInstancesStateCommand")
    .sc(schemas_0.UpdateContainerInstancesState$)
    .build() {
}

class UpdateDaemonCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateDaemon", {})
    .n("ECSClient", "UpdateDaemonCommand")
    .sc(schemas_0.UpdateDaemon$)
    .build() {
}

class UpdateExpressGatewayServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateExpressGatewayService", {})
    .n("ECSClient", "UpdateExpressGatewayServiceCommand")
    .sc(schemas_0.UpdateExpressGatewayService$)
    .build() {
}

class UpdateServiceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateService", {})
    .n("ECSClient", "UpdateServiceCommand")
    .sc(schemas_0.UpdateService$)
    .build() {
}

class UpdateServicePrimaryTaskSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateServicePrimaryTaskSet", {})
    .n("ECSClient", "UpdateServicePrimaryTaskSetCommand")
    .sc(schemas_0.UpdateServicePrimaryTaskSet$)
    .build() {
}

class UpdateTaskProtectionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateTaskProtection", {})
    .n("ECSClient", "UpdateTaskProtectionCommand")
    .sc(schemas_0.UpdateTaskProtection$)
    .build() {
}

class UpdateTaskSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonEC2ContainerServiceV20141113", "UpdateTaskSet", {})
    .n("ECSClient", "UpdateTaskSetCommand")
    .sc(schemas_0.UpdateTaskSet$)
    .build() {
}

const paginateListAccountSettings = core.createPaginator(ECSClient, ListAccountSettingsCommand, "nextToken", "nextToken", "maxResults");

const paginateListAttributes = core.createPaginator(ECSClient, ListAttributesCommand, "nextToken", "nextToken", "maxResults");

const paginateListClusters = core.createPaginator(ECSClient, ListClustersCommand, "nextToken", "nextToken", "maxResults");

const paginateListContainerInstances = core.createPaginator(ECSClient, ListContainerInstancesCommand, "nextToken", "nextToken", "maxResults");

const paginateListServicesByNamespace = core.createPaginator(ECSClient, ListServicesByNamespaceCommand, "nextToken", "nextToken", "maxResults");

const paginateListServices = core.createPaginator(ECSClient, ListServicesCommand, "nextToken", "nextToken", "maxResults");

const paginateListTaskDefinitionFamilies = core.createPaginator(ECSClient, ListTaskDefinitionFamiliesCommand, "nextToken", "nextToken", "maxResults");

const paginateListTaskDefinitions = core.createPaginator(ECSClient, ListTaskDefinitionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListTasks = core.createPaginator(ECSClient, ListTasksCommand, "nextToken", "nextToken", "maxResults");

const checkState$8 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeDaemonCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.daemon.status;
            };
            if (returnComparator() === "ACTIVE") {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.daemon.status;
            };
            if (returnComparator() === "DELETE_IN_PROGRESS") {
                return { state: utilWaiter.WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForDaemonActive = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$8);
};
const waitUntilDaemonActive = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$8);
    return utilWaiter.checkExceptions(result);
};

const checkState$7 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeDaemonDeploymentsCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.daemonDeployments);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "STOPPED");
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.failures);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.reason;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "MISSING") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForDaemonDeploymentStopped = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$7);
};
const waitUntilDaemonDeploymentStopped = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$7);
    return utilWaiter.checkExceptions(result);
};

const checkState$6 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeDaemonDeploymentsCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.daemonDeployments);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "SUCCESSFUL");
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.daemonDeployments);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "STOPPED") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.daemonDeployments);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_FAILED") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.daemonDeployments);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "ROLLBACK_SUCCESSFUL") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.failures);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.reason;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "MISSING") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForDaemonDeploymentSuccessful = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$6);
};
const waitUntilDaemonDeploymentSuccessful = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$6);
    return utilWaiter.checkExceptions(result);
};

const checkState$5 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeDaemonTaskDefinitionCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.daemonTaskDefinition.status;
            };
            if (returnComparator() === "ACTIVE") {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.daemonTaskDefinition.status;
            };
            if (returnComparator() === "DELETE_IN_PROGRESS") {
                return { state: utilWaiter.WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                return result.daemonTaskDefinition.status;
            };
            if (returnComparator() === "DELETED") {
                return { state: utilWaiter.WaiterState.FAILURE, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForDaemonTaskDefinitionActive = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$5);
};
const waitUntilDaemonTaskDefinitionActive = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$5);
    return utilWaiter.checkExceptions(result);
};

const checkState$4 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeDaemonTaskDefinitionCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.daemonTaskDefinition.status;
            };
            if (returnComparator() === "DELETED") {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForDaemonTaskDefinitionDeleted = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
};
const waitUntilDaemonTaskDefinitionDeleted = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
    return utilWaiter.checkExceptions(result);
};

const checkState$3 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeServicesCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.failures);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.reason;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "MISSING") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.services);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "INACTIVE") {
                    return { state: utilWaiter.WaiterState.SUCCESS, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForServicesInactive = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 600 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
};
const waitUntilServicesInactive = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 600 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
    return utilWaiter.checkExceptions(result);
};

const checkState$2 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeServicesCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.failures);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.reason;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "MISSING") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.services);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "DRAINING") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.services);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.status;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "INACTIVE") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let filterRes_2 = result.services.filter((element_1) => {
                    return (!((element_1.deployments.length == 1) && (element_1.runningCount == element_1.desiredCount)));
                });
                return (filterRes_2.length == 0);
            };
            if (returnComparator() == true) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForServicesStable = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 600 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
};
const waitUntilServicesStable = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 600 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
    return utilWaiter.checkExceptions(result);
};

const checkState$1 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeTasksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.tasks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.lastStatus;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "STOPPED") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.failures);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.reason;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "MISSING") {
                    return { state: utilWaiter.WaiterState.FAILURE, reason };
                }
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.tasks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.lastStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "RUNNING");
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForTasksRunning = async (params, input) => {
    const serviceDefaults = { minDelay: 6, maxDelay: 600 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
};
const waitUntilTasksRunning = async (params, input) => {
    const serviceDefaults = { minDelay: 6, maxDelay: 600 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
    return utilWaiter.checkExceptions(result);
};

const checkState = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeTasksCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.tasks);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.lastStatus;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "STOPPED");
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForTasksStopped = async (params, input) => {
    const serviceDefaults = { minDelay: 6, maxDelay: 600 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilTasksStopped = async (params, input) => {
    const serviceDefaults = { minDelay: 6, maxDelay: 600 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return utilWaiter.checkExceptions(result);
};

const commands = {
    CreateCapacityProviderCommand,
    CreateClusterCommand,
    CreateDaemonCommand,
    CreateExpressGatewayServiceCommand,
    CreateServiceCommand,
    CreateTaskSetCommand,
    DeleteAccountSettingCommand,
    DeleteAttributesCommand,
    DeleteCapacityProviderCommand,
    DeleteClusterCommand,
    DeleteDaemonCommand,
    DeleteDaemonTaskDefinitionCommand,
    DeleteExpressGatewayServiceCommand,
    DeleteServiceCommand,
    DeleteTaskDefinitionsCommand,
    DeleteTaskSetCommand,
    DeregisterContainerInstanceCommand,
    DeregisterTaskDefinitionCommand,
    DescribeCapacityProvidersCommand,
    DescribeClustersCommand,
    DescribeContainerInstancesCommand,
    DescribeDaemonCommand,
    DescribeDaemonDeploymentsCommand,
    DescribeDaemonRevisionsCommand,
    DescribeDaemonTaskDefinitionCommand,
    DescribeExpressGatewayServiceCommand,
    DescribeServiceDeploymentsCommand,
    DescribeServiceRevisionsCommand,
    DescribeServicesCommand,
    DescribeTaskDefinitionCommand,
    DescribeTasksCommand,
    DescribeTaskSetsCommand,
    DiscoverPollEndpointCommand,
    ExecuteCommandCommand,
    GetTaskProtectionCommand,
    ListAccountSettingsCommand,
    ListAttributesCommand,
    ListClustersCommand,
    ListContainerInstancesCommand,
    ListDaemonDeploymentsCommand,
    ListDaemonsCommand,
    ListDaemonTaskDefinitionsCommand,
    ListServiceDeploymentsCommand,
    ListServicesCommand,
    ListServicesByNamespaceCommand,
    ListTagsForResourceCommand,
    ListTaskDefinitionFamiliesCommand,
    ListTaskDefinitionsCommand,
    ListTasksCommand,
    PutAccountSettingCommand,
    PutAccountSettingDefaultCommand,
    PutAttributesCommand,
    PutClusterCapacityProvidersCommand,
    RegisterContainerInstanceCommand,
    RegisterDaemonTaskDefinitionCommand,
    RegisterTaskDefinitionCommand,
    RunTaskCommand,
    StartTaskCommand,
    StopServiceDeploymentCommand,
    StopTaskCommand,
    SubmitAttachmentStateChangesCommand,
    SubmitContainerStateChangeCommand,
    SubmitTaskStateChangeCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateCapacityProviderCommand,
    UpdateClusterCommand,
    UpdateClusterSettingsCommand,
    UpdateContainerAgentCommand,
    UpdateContainerInstancesStateCommand,
    UpdateDaemonCommand,
    UpdateExpressGatewayServiceCommand,
    UpdateServiceCommand,
    UpdateServicePrimaryTaskSetCommand,
    UpdateTaskProtectionCommand,
    UpdateTaskSetCommand,
};
const paginators = {
    paginateListAccountSettings,
    paginateListAttributes,
    paginateListClusters,
    paginateListContainerInstances,
    paginateListServices,
    paginateListServicesByNamespace,
    paginateListTaskDefinitionFamilies,
    paginateListTaskDefinitions,
    paginateListTasks,
};
const waiters = {
    waitUntilDaemonActive,
    waitUntilDaemonDeploymentSuccessful,
    waitUntilDaemonDeploymentStopped,
    waitUntilDaemonTaskDefinitionActive,
    waitUntilDaemonTaskDefinitionDeleted,
    waitUntilServicesInactive,
    waitUntilServicesStable,
    waitUntilTasksRunning,
    waitUntilTasksStopped,
};
class ECS extends ECSClient {
}
smithyClient.createAggregatedClient(commands, ECS, { paginators, waiters });

const AcceleratorManufacturer = {
    AMAZON_WEB_SERVICES: "amazon-web-services",
    AMD: "amd",
    HABANA: "habana",
    NVIDIA: "nvidia",
    XILINX: "xilinx",
};
const AcceleratorName = {
    A100: "a100",
    A10G: "a10g",
    H100: "h100",
    INFERENTIA: "inferentia",
    K520: "k520",
    K80: "k80",
    M60: "m60",
    RADEON_PRO_V520: "radeon-pro-v520",
    T4: "t4",
    T4G: "t4g",
    V100: "v100",
    VU9P: "vu9p",
};
const AcceleratorType = {
    FPGA: "fpga",
    GPU: "gpu",
    INFERENCE: "inference",
};
const AccessType = {
    PRIVATE: "PRIVATE",
    PUBLIC: "PUBLIC",
};
const AgentUpdateStatus = {
    FAILED: "FAILED",
    PENDING: "PENDING",
    STAGED: "STAGED",
    STAGING: "STAGING",
    UPDATED: "UPDATED",
    UPDATING: "UPDATING",
};
const ManagedDraining = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ManagedScalingStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ManagedTerminationProtection = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AutoRepairActionsStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const CapacityOptionType = {
    ON_DEMAND: "ON_DEMAND",
    RESERVED: "RESERVED",
    SPOT: "SPOT",
};
const CapacityReservationPreference = {
    RESERVATIONS_EXCLUDED: "RESERVATIONS_EXCLUDED",
    RESERVATIONS_FIRST: "RESERVATIONS_FIRST",
    RESERVATIONS_ONLY: "RESERVATIONS_ONLY",
};
const BareMetal = {
    EXCLUDED: "excluded",
    INCLUDED: "included",
    REQUIRED: "required",
};
const BurstablePerformance = {
    EXCLUDED: "excluded",
    INCLUDED: "included",
    REQUIRED: "required",
};
const CpuManufacturer = {
    AMAZON_WEB_SERVICES: "amazon-web-services",
    AMD: "amd",
    INTEL: "intel",
};
const InstanceGeneration = {
    CURRENT: "current",
    PREVIOUS: "previous",
};
const LocalStorage = {
    EXCLUDED: "excluded",
    INCLUDED: "included",
    REQUIRED: "required",
};
const LocalStorageType = {
    HDD: "hdd",
    SSD: "ssd",
};
const ManagedInstancesMonitoringOptions = {
    BASIC: "BASIC",
    DETAILED: "DETAILED",
};
const PropagateMITags = {
    CAPACITY_PROVIDER: "CAPACITY_PROVIDER",
    NONE: "NONE",
};
const CapacityProviderStatus = {
    ACTIVE: "ACTIVE",
    DEPROVISIONING: "DEPROVISIONING",
    INACTIVE: "INACTIVE",
    PROVISIONING: "PROVISIONING",
};
const CapacityProviderType = {
    EC2_AUTOSCALING: "EC2_AUTOSCALING",
    FARGATE: "FARGATE",
    FARGATE_SPOT: "FARGATE_SPOT",
    MANAGED_INSTANCES: "MANAGED_INSTANCES",
};
const CapacityProviderUpdateStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    UPDATE_COMPLETE: "UPDATE_COMPLETE",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATE_IN_PROGRESS: "UPDATE_IN_PROGRESS",
};
const CapacityProviderField = {
    TAGS: "TAGS",
};
const ExecuteCommandLogging = {
    DEFAULT: "DEFAULT",
    NONE: "NONE",
    OVERRIDE: "OVERRIDE",
};
const ClusterSettingName = {
    CONTAINER_INSIGHTS: "containerInsights",
};
const TargetType = {
    CONTAINER_INSTANCE: "container-instance",
};
const InstanceHealthCheckState = {
    IMPAIRED: "IMPAIRED",
    INITIALIZING: "INITIALIZING",
    INSUFFICIENT_DATA: "INSUFFICIENT_DATA",
    OK: "OK",
};
const InstanceHealthCheckType = {
    ACCELERATED_COMPUTE: "ACCELERATED_COMPUTE",
    CONTAINER_RUNTIME: "CONTAINER_RUNTIME",
    DAEMON: "DAEMON",
};
const ClusterField = {
    ATTACHMENTS: "ATTACHMENTS",
    CONFIGURATIONS: "CONFIGURATIONS",
    SETTINGS: "SETTINGS",
    STATISTICS: "STATISTICS",
    TAGS: "TAGS",
};
const ContainerInstanceStatus = {
    ACTIVE: "ACTIVE",
    DEREGISTERING: "DEREGISTERING",
    DRAINING: "DRAINING",
    REGISTERING: "REGISTERING",
    REGISTRATION_FAILED: "REGISTRATION_FAILED",
};
const TransportProtocol = {
    TCP: "tcp",
    UDP: "udp",
};
const ManagedAgentName = {
    ExecuteCommandAgent: "ExecuteCommandAgent",
};
const ContainerInstanceField = {
    CONTAINER_INSTANCE_HEALTH: "CONTAINER_INSTANCE_HEALTH",
    TAGS: "TAGS",
};
const DesiredStatus = {
    PENDING: "PENDING",
    RUNNING: "RUNNING",
    STOPPED: "STOPPED",
};
const LaunchType = {
    EC2: "EC2",
    EXTERNAL: "EXTERNAL",
    FARGATE: "FARGATE",
    MANAGED_INSTANCES: "MANAGED_INSTANCES",
};
const PlatformDeviceType = {
    GPU: "GPU",
};
const DaemonDeploymentRollbackMonitorsStatus = {
    DISABLED: "DISABLED",
    MONITORING: "MONITORING",
    MONITORING_COMPLETE: "MONITORING_COMPLETE",
    TRIGGERED: "TRIGGERED",
};
const DaemonDeploymentStatus = {
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
    ROLLBACK_FAILED: "ROLLBACK_FAILED",
    ROLLBACK_IN_PROGRESS: "ROLLBACK_IN_PROGRESS",
    ROLLBACK_SUCCESSFUL: "ROLLBACK_SUCCESSFUL",
    STOPPED: "STOPPED",
    STOP_REQUESTED: "STOP_REQUESTED",
    SUCCESSFUL: "SUCCESSFUL",
};
const DaemonPropagateTags = {
    DAEMON: "DAEMON",
    NONE: "NONE",
};
const DaemonStatus = {
    ACTIVE: "ACTIVE",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
const ContainerCondition = {
    COMPLETE: "COMPLETE",
    HEALTHY: "HEALTHY",
    START: "START",
    SUCCESS: "SUCCESS",
};
const EnvironmentFileType = {
    S3: "s3",
};
const FirelensConfigurationType = {
    FLUENTBIT: "fluentbit",
    FLUENTD: "fluentd",
};
const DeviceCgroupPermission = {
    MKNOD: "mknod",
    READ: "read",
    WRITE: "write",
};
const LogDriver = {
    AWSFIRELENS: "awsfirelens",
    AWSLOGS: "awslogs",
    FLUENTD: "fluentd",
    GELF: "gelf",
    JOURNALD: "journald",
    JSON_FILE: "json-file",
    SPLUNK: "splunk",
    SYSLOG: "syslog",
};
const UlimitName = {
    CORE: "core",
    CPU: "cpu",
    DATA: "data",
    FSIZE: "fsize",
    LOCKS: "locks",
    MEMLOCK: "memlock",
    MSGQUEUE: "msgqueue",
    NICE: "nice",
    NOFILE: "nofile",
    NPROC: "nproc",
    RSS: "rss",
    RTPRIO: "rtprio",
    RTTIME: "rttime",
    SIGPENDING: "sigpending",
    STACK: "stack",
};
const DaemonTaskDefinitionStatus = {
    ACTIVE: "ACTIVE",
    DELETED: "DELETED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
const DaemonTaskDefinitionRevisionFilter = {
    LAST_REGISTERED: "LAST_REGISTERED",
};
const SortOrder = {
    ASC: "ASC",
    DESC: "DESC",
};
const DaemonTaskDefinitionStatusFilter = {
    ACTIVE: "ACTIVE",
    ALL: "ALL",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
const SettingName = {
    AWSVPC_TRUNKING: "awsvpcTrunking",
    CONTAINER_INSIGHTS: "containerInsights",
    CONTAINER_INSTANCE_LONG_ARN_FORMAT: "containerInstanceLongArnFormat",
    DEFAULT_LOG_DRIVER_MODE: "defaultLogDriverMode",
    FARGATE_EVENT_WINDOWS: "fargateEventWindows",
    FARGATE_FIPS_MODE: "fargateFIPSMode",
    FARGATE_TASK_RETIREMENT_WAIT_PERIOD: "fargateTaskRetirementWaitPeriod",
    GUARD_DUTY_ACTIVATE: "guardDutyActivate",
    SERVICE_LONG_ARN_FORMAT: "serviceLongArnFormat",
    TAG_RESOURCE_AUTHORIZATION: "tagResourceAuthorization",
    TASK_LONG_ARN_FORMAT: "taskLongArnFormat",
};
const SettingType = {
    AWS_MANAGED: "aws_managed",
    USER: "user",
};
const Compatibility = {
    EC2: "EC2",
    EXTERNAL: "EXTERNAL",
    FARGATE: "FARGATE",
    MANAGED_INSTANCES: "MANAGED_INSTANCES",
};
const ApplicationProtocol = {
    GRPC: "grpc",
    HTTP: "http",
    HTTP2: "http2",
};
const ResourceType = {
    GPU: "GPU",
    INFERENCE_ACCELERATOR: "InferenceAccelerator",
};
const VersionConsistency = {
    DISABLED: "disabled",
    ENABLED: "enabled",
};
const IpcMode = {
    HOST: "host",
    NONE: "none",
    TASK: "task",
};
const NetworkMode = {
    AWSVPC: "awsvpc",
    BRIDGE: "bridge",
    HOST: "host",
    NONE: "none",
};
const PidMode = {
    HOST: "host",
    TASK: "task",
};
const TaskDefinitionPlacementConstraintType = {
    MEMBER_OF: "memberOf",
};
const ProxyConfigurationType = {
    APPMESH: "APPMESH",
};
const CPUArchitecture = {
    ARM64: "ARM64",
    X86_64: "X86_64",
};
const OSFamily = {
    LINUX: "LINUX",
    WINDOWS_SERVER_2004_CORE: "WINDOWS_SERVER_2004_CORE",
    WINDOWS_SERVER_2016_FULL: "WINDOWS_SERVER_2016_FULL",
    WINDOWS_SERVER_2019_CORE: "WINDOWS_SERVER_2019_CORE",
    WINDOWS_SERVER_2019_FULL: "WINDOWS_SERVER_2019_FULL",
    WINDOWS_SERVER_2022_CORE: "WINDOWS_SERVER_2022_CORE",
    WINDOWS_SERVER_2022_FULL: "WINDOWS_SERVER_2022_FULL",
    WINDOWS_SERVER_2025_CORE: "WINDOWS_SERVER_2025_CORE",
    WINDOWS_SERVER_2025_FULL: "WINDOWS_SERVER_2025_FULL",
    WINDOWS_SERVER_20H2_CORE: "WINDOWS_SERVER_20H2_CORE",
};
const TaskDefinitionStatus = {
    ACTIVE: "ACTIVE",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    INACTIVE: "INACTIVE",
};
const Scope = {
    SHARED: "shared",
    TASK: "task",
};
const EFSAuthorizationConfigIAM = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const EFSTransitEncryption = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const TaskDefinitionField = {
    TAGS: "TAGS",
};
const TaskDefinitionFamilyStatus = {
    ACTIVE: "ACTIVE",
    ALL: "ALL",
    INACTIVE: "INACTIVE",
};
const ServiceDeploymentRollbackMonitorsStatus = {
    DISABLED: "DISABLED",
    MONITORING: "MONITORING",
    MONITORING_COMPLETE: "MONITORING_COMPLETE",
    TRIGGERED: "TRIGGERED",
};
const DeploymentLifecycleHookStage = {
    POST_PRODUCTION_TRAFFIC_SHIFT: "POST_PRODUCTION_TRAFFIC_SHIFT",
    POST_SCALE_UP: "POST_SCALE_UP",
    POST_TEST_TRAFFIC_SHIFT: "POST_TEST_TRAFFIC_SHIFT",
    PRE_SCALE_UP: "PRE_SCALE_UP",
    PRODUCTION_TRAFFIC_SHIFT: "PRODUCTION_TRAFFIC_SHIFT",
    RECONCILE_SERVICE: "RECONCILE_SERVICE",
    TEST_TRAFFIC_SHIFT: "TEST_TRAFFIC_SHIFT",
};
const DeploymentStrategy = {
    BLUE_GREEN: "BLUE_GREEN",
    CANARY: "CANARY",
    LINEAR: "LINEAR",
    ROLLING: "ROLLING",
};
const ServiceDeploymentLifecycleStage = {
    BAKE_TIME: "BAKE_TIME",
    CLEAN_UP: "CLEAN_UP",
    POST_PRODUCTION_TRAFFIC_SHIFT: "POST_PRODUCTION_TRAFFIC_SHIFT",
    POST_SCALE_UP: "POST_SCALE_UP",
    POST_TEST_TRAFFIC_SHIFT: "POST_TEST_TRAFFIC_SHIFT",
    PRE_SCALE_UP: "PRE_SCALE_UP",
    PRODUCTION_TRAFFIC_SHIFT: "PRODUCTION_TRAFFIC_SHIFT",
    RECONCILE_SERVICE: "RECONCILE_SERVICE",
    SCALE_UP: "SCALE_UP",
    TEST_TRAFFIC_SHIFT: "TEST_TRAFFIC_SHIFT",
};
const ServiceDeploymentStatus = {
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
    ROLLBACK_FAILED: "ROLLBACK_FAILED",
    ROLLBACK_IN_PROGRESS: "ROLLBACK_IN_PROGRESS",
    ROLLBACK_REQUESTED: "ROLLBACK_REQUESTED",
    ROLLBACK_SUCCESSFUL: "ROLLBACK_SUCCESSFUL",
    STOPPED: "STOPPED",
    STOP_REQUESTED: "STOP_REQUESTED",
    SUCCESSFUL: "SUCCESSFUL",
};
const ExpressGatewayServiceScalingMetric = {
    AverageCPUUtilization: "AVERAGE_CPU",
    AverageMemoryUtilization: "AVERAGE_MEMORY",
    RequestCountPerTarget: "REQUEST_COUNT_PER_TARGET",
};
const ExpressGatewayServiceStatusCode = {
    ACTIVE: "ACTIVE",
    DRAINING: "DRAINING",
    INACTIVE: "INACTIVE",
};
const AvailabilityZoneRebalancing = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DeploymentControllerType = {
    CODE_DEPLOY: "CODE_DEPLOY",
    ECS: "ECS",
    EXTERNAL: "EXTERNAL",
};
const AssignPublicIp = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const PlacementConstraintType = {
    DISTINCT_INSTANCE: "distinctInstance",
    MEMBER_OF: "memberOf",
};
const PlacementStrategyType = {
    BINPACK: "binpack",
    RANDOM: "random",
    SPREAD: "spread",
};
const PropagateTags = {
    NONE: "NONE",
    SERVICE: "SERVICE",
    TASK_DEFINITION: "TASK_DEFINITION",
};
const SchedulingStrategy = {
    DAEMON: "DAEMON",
    REPLICA: "REPLICA",
};
const ServiceConnectAccessLoggingFormat = {
    JSON: "JSON",
    TEXT: "TEXT",
};
const ServiceConnectIncludeQueryParameters = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const TaskFilesystemType = {
    EXT3: "ext3",
    EXT4: "ext4",
    NTFS: "ntfs",
    XFS: "xfs",
};
const EBSResourceType = {
    VOLUME: "volume",
};
const DeploymentRolloutState = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
const ResourceManagementType = {
    CUSTOMER: "CUSTOMER",
    ECS: "ECS",
};
const ScaleUnit = {
    PERCENT: "PERCENT",
};
const StabilityStatus = {
    STABILIZING: "STABILIZING",
    STEADY_STATE: "STEADY_STATE",
};
const ExpressGatewayServiceInclude = {
    TAGS: "TAGS",
};
const ServiceField = {
    TAGS: "TAGS",
};
const StopServiceDeploymentStopType = {
    ABORT: "ABORT",
    ROLLBACK: "ROLLBACK",
};
const ManagedResourceStatus = {
    ACTIVE: "ACTIVE",
    DELETED: "DELETED",
    DEPROVISIONING: "DEPROVISIONING",
    FAILED: "FAILED",
    PROVISIONING: "PROVISIONING",
};
const TaskField = {
    TAGS: "TAGS",
};
const Connectivity = {
    CONNECTED: "CONNECTED",
    DISCONNECTED: "DISCONNECTED",
};
const HealthStatus = {
    HEALTHY: "HEALTHY",
    UNHEALTHY: "UNHEALTHY",
    UNKNOWN: "UNKNOWN",
};
const TaskStopCode = {
    ESSENTIAL_CONTAINER_EXITED: "EssentialContainerExited",
    SERVICE_SCHEDULER_INITIATED: "ServiceSchedulerInitiated",
    SPOT_INTERRUPTION: "SpotInterruption",
    TASK_FAILED_TO_START: "TaskFailedToStart",
    TERMINATION_NOTICE: "TerminationNotice",
    USER_INITIATED: "UserInitiated",
};
const TaskSetField = {
    TAGS: "TAGS",
};

exports.$Command = smithyClient.Command;
exports.__Client = smithyClient.Client;
exports.ECSServiceException = ECSServiceException.ECSServiceException;
exports.AcceleratorManufacturer = AcceleratorManufacturer;
exports.AcceleratorName = AcceleratorName;
exports.AcceleratorType = AcceleratorType;
exports.AccessType = AccessType;
exports.AgentUpdateStatus = AgentUpdateStatus;
exports.ApplicationProtocol = ApplicationProtocol;
exports.AssignPublicIp = AssignPublicIp;
exports.AutoRepairActionsStatus = AutoRepairActionsStatus;
exports.AvailabilityZoneRebalancing = AvailabilityZoneRebalancing;
exports.BareMetal = BareMetal;
exports.BurstablePerformance = BurstablePerformance;
exports.CPUArchitecture = CPUArchitecture;
exports.CapacityOptionType = CapacityOptionType;
exports.CapacityProviderField = CapacityProviderField;
exports.CapacityProviderStatus = CapacityProviderStatus;
exports.CapacityProviderType = CapacityProviderType;
exports.CapacityProviderUpdateStatus = CapacityProviderUpdateStatus;
exports.CapacityReservationPreference = CapacityReservationPreference;
exports.ClusterField = ClusterField;
exports.ClusterSettingName = ClusterSettingName;
exports.Compatibility = Compatibility;
exports.Connectivity = Connectivity;
exports.ContainerCondition = ContainerCondition;
exports.ContainerInstanceField = ContainerInstanceField;
exports.ContainerInstanceStatus = ContainerInstanceStatus;
exports.CpuManufacturer = CpuManufacturer;
exports.CreateCapacityProviderCommand = CreateCapacityProviderCommand;
exports.CreateClusterCommand = CreateClusterCommand;
exports.CreateDaemonCommand = CreateDaemonCommand;
exports.CreateExpressGatewayServiceCommand = CreateExpressGatewayServiceCommand;
exports.CreateServiceCommand = CreateServiceCommand;
exports.CreateTaskSetCommand = CreateTaskSetCommand;
exports.DaemonDeploymentRollbackMonitorsStatus = DaemonDeploymentRollbackMonitorsStatus;
exports.DaemonDeploymentStatus = DaemonDeploymentStatus;
exports.DaemonPropagateTags = DaemonPropagateTags;
exports.DaemonStatus = DaemonStatus;
exports.DaemonTaskDefinitionRevisionFilter = DaemonTaskDefinitionRevisionFilter;
exports.DaemonTaskDefinitionStatus = DaemonTaskDefinitionStatus;
exports.DaemonTaskDefinitionStatusFilter = DaemonTaskDefinitionStatusFilter;
exports.DeleteAccountSettingCommand = DeleteAccountSettingCommand;
exports.DeleteAttributesCommand = DeleteAttributesCommand;
exports.DeleteCapacityProviderCommand = DeleteCapacityProviderCommand;
exports.DeleteClusterCommand = DeleteClusterCommand;
exports.DeleteDaemonCommand = DeleteDaemonCommand;
exports.DeleteDaemonTaskDefinitionCommand = DeleteDaemonTaskDefinitionCommand;
exports.DeleteExpressGatewayServiceCommand = DeleteExpressGatewayServiceCommand;
exports.DeleteServiceCommand = DeleteServiceCommand;
exports.DeleteTaskDefinitionsCommand = DeleteTaskDefinitionsCommand;
exports.DeleteTaskSetCommand = DeleteTaskSetCommand;
exports.DeploymentControllerType = DeploymentControllerType;
exports.DeploymentLifecycleHookStage = DeploymentLifecycleHookStage;
exports.DeploymentRolloutState = DeploymentRolloutState;
exports.DeploymentStrategy = DeploymentStrategy;
exports.DeregisterContainerInstanceCommand = DeregisterContainerInstanceCommand;
exports.DeregisterTaskDefinitionCommand = DeregisterTaskDefinitionCommand;
exports.DescribeCapacityProvidersCommand = DescribeCapacityProvidersCommand;
exports.DescribeClustersCommand = DescribeClustersCommand;
exports.DescribeContainerInstancesCommand = DescribeContainerInstancesCommand;
exports.DescribeDaemonCommand = DescribeDaemonCommand;
exports.DescribeDaemonDeploymentsCommand = DescribeDaemonDeploymentsCommand;
exports.DescribeDaemonRevisionsCommand = DescribeDaemonRevisionsCommand;
exports.DescribeDaemonTaskDefinitionCommand = DescribeDaemonTaskDefinitionCommand;
exports.DescribeExpressGatewayServiceCommand = DescribeExpressGatewayServiceCommand;
exports.DescribeServiceDeploymentsCommand = DescribeServiceDeploymentsCommand;
exports.DescribeServiceRevisionsCommand = DescribeServiceRevisionsCommand;
exports.DescribeServicesCommand = DescribeServicesCommand;
exports.DescribeTaskDefinitionCommand = DescribeTaskDefinitionCommand;
exports.DescribeTaskSetsCommand = DescribeTaskSetsCommand;
exports.DescribeTasksCommand = DescribeTasksCommand;
exports.DesiredStatus = DesiredStatus;
exports.DeviceCgroupPermission = DeviceCgroupPermission;
exports.DiscoverPollEndpointCommand = DiscoverPollEndpointCommand;
exports.EBSResourceType = EBSResourceType;
exports.ECS = ECS;
exports.ECSClient = ECSClient;
exports.EFSAuthorizationConfigIAM = EFSAuthorizationConfigIAM;
exports.EFSTransitEncryption = EFSTransitEncryption;
exports.EnvironmentFileType = EnvironmentFileType;
exports.ExecuteCommandCommand = ExecuteCommandCommand;
exports.ExecuteCommandLogging = ExecuteCommandLogging;
exports.ExpressGatewayServiceInclude = ExpressGatewayServiceInclude;
exports.ExpressGatewayServiceScalingMetric = ExpressGatewayServiceScalingMetric;
exports.ExpressGatewayServiceStatusCode = ExpressGatewayServiceStatusCode;
exports.FirelensConfigurationType = FirelensConfigurationType;
exports.GetTaskProtectionCommand = GetTaskProtectionCommand;
exports.HealthStatus = HealthStatus;
exports.InstanceGeneration = InstanceGeneration;
exports.InstanceHealthCheckState = InstanceHealthCheckState;
exports.InstanceHealthCheckType = InstanceHealthCheckType;
exports.IpcMode = IpcMode;
exports.LaunchType = LaunchType;
exports.ListAccountSettingsCommand = ListAccountSettingsCommand;
exports.ListAttributesCommand = ListAttributesCommand;
exports.ListClustersCommand = ListClustersCommand;
exports.ListContainerInstancesCommand = ListContainerInstancesCommand;
exports.ListDaemonDeploymentsCommand = ListDaemonDeploymentsCommand;
exports.ListDaemonTaskDefinitionsCommand = ListDaemonTaskDefinitionsCommand;
exports.ListDaemonsCommand = ListDaemonsCommand;
exports.ListServiceDeploymentsCommand = ListServiceDeploymentsCommand;
exports.ListServicesByNamespaceCommand = ListServicesByNamespaceCommand;
exports.ListServicesCommand = ListServicesCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTaskDefinitionFamiliesCommand = ListTaskDefinitionFamiliesCommand;
exports.ListTaskDefinitionsCommand = ListTaskDefinitionsCommand;
exports.ListTasksCommand = ListTasksCommand;
exports.LocalStorage = LocalStorage;
exports.LocalStorageType = LocalStorageType;
exports.LogDriver = LogDriver;
exports.ManagedAgentName = ManagedAgentName;
exports.ManagedDraining = ManagedDraining;
exports.ManagedInstancesMonitoringOptions = ManagedInstancesMonitoringOptions;
exports.ManagedResourceStatus = ManagedResourceStatus;
exports.ManagedScalingStatus = ManagedScalingStatus;
exports.ManagedTerminationProtection = ManagedTerminationProtection;
exports.NetworkMode = NetworkMode;
exports.OSFamily = OSFamily;
exports.PidMode = PidMode;
exports.PlacementConstraintType = PlacementConstraintType;
exports.PlacementStrategyType = PlacementStrategyType;
exports.PlatformDeviceType = PlatformDeviceType;
exports.PropagateMITags = PropagateMITags;
exports.PropagateTags = PropagateTags;
exports.ProxyConfigurationType = ProxyConfigurationType;
exports.PutAccountSettingCommand = PutAccountSettingCommand;
exports.PutAccountSettingDefaultCommand = PutAccountSettingDefaultCommand;
exports.PutAttributesCommand = PutAttributesCommand;
exports.PutClusterCapacityProvidersCommand = PutClusterCapacityProvidersCommand;
exports.RegisterContainerInstanceCommand = RegisterContainerInstanceCommand;
exports.RegisterDaemonTaskDefinitionCommand = RegisterDaemonTaskDefinitionCommand;
exports.RegisterTaskDefinitionCommand = RegisterTaskDefinitionCommand;
exports.ResourceManagementType = ResourceManagementType;
exports.ResourceType = ResourceType;
exports.RunTaskCommand = RunTaskCommand;
exports.ScaleUnit = ScaleUnit;
exports.SchedulingStrategy = SchedulingStrategy;
exports.Scope = Scope;
exports.ServiceConnectAccessLoggingFormat = ServiceConnectAccessLoggingFormat;
exports.ServiceConnectIncludeQueryParameters = ServiceConnectIncludeQueryParameters;
exports.ServiceDeploymentLifecycleStage = ServiceDeploymentLifecycleStage;
exports.ServiceDeploymentRollbackMonitorsStatus = ServiceDeploymentRollbackMonitorsStatus;
exports.ServiceDeploymentStatus = ServiceDeploymentStatus;
exports.ServiceField = ServiceField;
exports.SettingName = SettingName;
exports.SettingType = SettingType;
exports.SortOrder = SortOrder;
exports.StabilityStatus = StabilityStatus;
exports.StartTaskCommand = StartTaskCommand;
exports.StopServiceDeploymentCommand = StopServiceDeploymentCommand;
exports.StopServiceDeploymentStopType = StopServiceDeploymentStopType;
exports.StopTaskCommand = StopTaskCommand;
exports.SubmitAttachmentStateChangesCommand = SubmitAttachmentStateChangesCommand;
exports.SubmitContainerStateChangeCommand = SubmitContainerStateChangeCommand;
exports.SubmitTaskStateChangeCommand = SubmitTaskStateChangeCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.TargetType = TargetType;
exports.TaskDefinitionFamilyStatus = TaskDefinitionFamilyStatus;
exports.TaskDefinitionField = TaskDefinitionField;
exports.TaskDefinitionPlacementConstraintType = TaskDefinitionPlacementConstraintType;
exports.TaskDefinitionStatus = TaskDefinitionStatus;
exports.TaskField = TaskField;
exports.TaskFilesystemType = TaskFilesystemType;
exports.TaskSetField = TaskSetField;
exports.TaskStopCode = TaskStopCode;
exports.TransportProtocol = TransportProtocol;
exports.UlimitName = UlimitName;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateCapacityProviderCommand = UpdateCapacityProviderCommand;
exports.UpdateClusterCommand = UpdateClusterCommand;
exports.UpdateClusterSettingsCommand = UpdateClusterSettingsCommand;
exports.UpdateContainerAgentCommand = UpdateContainerAgentCommand;
exports.UpdateContainerInstancesStateCommand = UpdateContainerInstancesStateCommand;
exports.UpdateDaemonCommand = UpdateDaemonCommand;
exports.UpdateExpressGatewayServiceCommand = UpdateExpressGatewayServiceCommand;
exports.UpdateServiceCommand = UpdateServiceCommand;
exports.UpdateServicePrimaryTaskSetCommand = UpdateServicePrimaryTaskSetCommand;
exports.UpdateTaskProtectionCommand = UpdateTaskProtectionCommand;
exports.UpdateTaskSetCommand = UpdateTaskSetCommand;
exports.VersionConsistency = VersionConsistency;
exports.paginateListAccountSettings = paginateListAccountSettings;
exports.paginateListAttributes = paginateListAttributes;
exports.paginateListClusters = paginateListClusters;
exports.paginateListContainerInstances = paginateListContainerInstances;
exports.paginateListServices = paginateListServices;
exports.paginateListServicesByNamespace = paginateListServicesByNamespace;
exports.paginateListTaskDefinitionFamilies = paginateListTaskDefinitionFamilies;
exports.paginateListTaskDefinitions = paginateListTaskDefinitions;
exports.paginateListTasks = paginateListTasks;
exports.waitForDaemonActive = waitForDaemonActive;
exports.waitForDaemonDeploymentStopped = waitForDaemonDeploymentStopped;
exports.waitForDaemonDeploymentSuccessful = waitForDaemonDeploymentSuccessful;
exports.waitForDaemonTaskDefinitionActive = waitForDaemonTaskDefinitionActive;
exports.waitForDaemonTaskDefinitionDeleted = waitForDaemonTaskDefinitionDeleted;
exports.waitForServicesInactive = waitForServicesInactive;
exports.waitForServicesStable = waitForServicesStable;
exports.waitForTasksRunning = waitForTasksRunning;
exports.waitForTasksStopped = waitForTasksStopped;
exports.waitUntilDaemonActive = waitUntilDaemonActive;
exports.waitUntilDaemonDeploymentStopped = waitUntilDaemonDeploymentStopped;
exports.waitUntilDaemonDeploymentSuccessful = waitUntilDaemonDeploymentSuccessful;
exports.waitUntilDaemonTaskDefinitionActive = waitUntilDaemonTaskDefinitionActive;
exports.waitUntilDaemonTaskDefinitionDeleted = waitUntilDaemonTaskDefinitionDeleted;
exports.waitUntilServicesInactive = waitUntilServicesInactive;
exports.waitUntilServicesStable = waitUntilServicesStable;
exports.waitUntilTasksRunning = waitUntilTasksRunning;
exports.waitUntilTasksStopped = waitUntilTasksStopped;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
