// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/CohortWidgetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CohortWidgetDefinition = void 0;
/**
 * The cohort widget visualizes user retention over time.
 */
class CohortWidgetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CohortWidgetDefinition.attributeTypeMap;
    }
}
exports.CohortWidgetDefinition = CohortWidgetDefinition;
/**
 * @ignore
 */
CohortWidgetDefinition.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    requests: {
        baseName: "requests",
        type: "Array<RetentionGridRequest>",
        required: true,
    },
    time: {
        baseName: "time",
        type: "WidgetTime",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    titleAlign: {
        baseName: "title_align",
        type: "WidgetTextAlign",
    },
    titleSize: {
        baseName: "title_size",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "CohortWidgetDefinitionType",
        required: true,
    },
};
