// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/DashboardTab.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DashboardTab = void 0;
/**
 * Dashboard tab for organizing widgets.
 */
class DashboardTab {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DashboardTab.attributeTypeMap;
    }
}
exports.DashboardTab = DashboardTab;
/**
 * @ignore
 */
DashboardTab.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    widgetIds: {
        baseName: "widget_ids",
        type: "Array<number>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
