// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/FormulaAndFunctionUserJourneyQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormulaAndFunctionUserJourneyQueryDefinition = void 0;
/**
 * A formula and functions User Journey query for defining funnel, timeseries, and scalar visualizations over journey data.
 */
class FormulaAndFunctionUserJourneyQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormulaAndFunctionUserJourneyQueryDefinition.attributeTypeMap;
    }
}
exports.FormulaAndFunctionUserJourneyQueryDefinition = FormulaAndFunctionUserJourneyQueryDefinition;
/**
 * @ignore
 */
FormulaAndFunctionUserJourneyQueryDefinition.attributeTypeMap = {
    compute: {
        baseName: "compute",
        type: "UserJourneyFormulaCompute",
        required: true,
    },
    dataSource: {
        baseName: "data_source",
        type: "ProductAnalyticsFunnelDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<UserJourneyFormulaGroupBy>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    search: {
        baseName: "search",
        type: "UserJourneySearch",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
