// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/FormulaAndFunctionRetentionQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormulaAndFunctionRetentionQueryDefinition = void 0;
/**
 * A formula and functions Retention query for defining timeseries and scalar visualizations.
 */
class FormulaAndFunctionRetentionQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormulaAndFunctionRetentionQueryDefinition.attributeTypeMap;
    }
}
exports.FormulaAndFunctionRetentionQueryDefinition = FormulaAndFunctionRetentionQueryDefinition;
/**
 * @ignore
 */
FormulaAndFunctionRetentionQueryDefinition.attributeTypeMap = {
    compute: {
        baseName: "compute",
        type: "RetentionCompute",
        required: true,
    },
    dataSource: {
        baseName: "data_source",
        type: "RetentionDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<RetentionGroupBy>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    search: {
        baseName: "search",
        type: "RetentionSearch",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
