// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/FormulaAndFunctionApmMetricsQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormulaAndFunctionApmMetricsQueryDefinition = void 0;
/**
 * A formula and functions APM metrics query.
 */
class FormulaAndFunctionApmMetricsQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormulaAndFunctionApmMetricsQueryDefinition.attributeTypeMap;
    }
}
exports.FormulaAndFunctionApmMetricsQueryDefinition = FormulaAndFunctionApmMetricsQueryDefinition;
/**
 * @ignore
 */
FormulaAndFunctionApmMetricsQueryDefinition.attributeTypeMap = {
    dataSource: {
        baseName: "data_source",
        type: "FormulaAndFunctionApmMetricsDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    operationMode: {
        baseName: "operation_mode",
        type: "string",
    },
    operationName: {
        baseName: "operation_name",
        type: "string",
    },
    peerTags: {
        baseName: "peer_tags",
        type: "Array<string>",
    },
    queryFilter: {
        baseName: "query_filter",
        type: "string",
    },
    resourceHash: {
        baseName: "resource_hash",
        type: "string",
    },
    resourceName: {
        baseName: "resource_name",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
    },
    spanKind: {
        baseName: "span_kind",
        type: "FormulaAndFunctionApmMetricsSpanKind",
    },
    stat: {
        baseName: "stat",
        type: "FormulaAndFunctionApmMetricStatName",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
