// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/FormulaAndFunctionProductAnalyticsExtendedQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormulaAndFunctionProductAnalyticsExtendedQueryDefinition = void 0;
/**
 * A formula and functions Product Analytics Extended query for advanced analytics features.
 */
class FormulaAndFunctionProductAnalyticsExtendedQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormulaAndFunctionProductAnalyticsExtendedQueryDefinition.attributeTypeMap;
    }
}
exports.FormulaAndFunctionProductAnalyticsExtendedQueryDefinition = FormulaAndFunctionProductAnalyticsExtendedQueryDefinition;
/**
 * @ignore
 */
FormulaAndFunctionProductAnalyticsExtendedQueryDefinition.attributeTypeMap = {
    audienceFilters: {
        baseName: "audience_filters",
        type: "ProductAnalyticsAudienceFilters",
    },
    compute: {
        baseName: "compute",
        type: "ProductAnalyticsExtendedCompute",
        required: true,
    },
    dataSource: {
        baseName: "data_source",
        type: "FormulaAndFunctionProductAnalyticsExtendedDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<ProductAnalyticsExtendedGroupBy>",
    },
    indexes: {
        baseName: "indexes",
        type: "Array<FormulaAndFunctionProductAnalyticsExtendedQueryDefinitionIndexesItems>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "any",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
