// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/FormulaAndFunctionEventsDataSource.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ON_CALL_EVENTS = exports.PRODUCT_ANALYTICS = exports.INCIDENT_ANALYTICS = exports.CI_PIPELINES = exports.CI_TESTS = exports.EVENTS = exports.AUDIT = exports.PROFILES = exports.SECURITY_SIGNALS = exports.RUM = exports.NETWORK = exports.SPANS = exports.LOGS = void 0;
exports.LOGS = "logs";
exports.SPANS = "spans";
exports.NETWORK = "network";
exports.RUM = "rum";
exports.SECURITY_SIGNALS = "security_signals";
exports.PROFILES = "profiles";
exports.AUDIT = "audit";
exports.EVENTS = "events";
exports.CI_TESTS = "ci_tests";
exports.CI_PIPELINES = "ci_pipelines";
exports.INCIDENT_ANALYTICS = "incident_analytics";
exports.PRODUCT_ANALYTICS = "product_analytics";
exports.ON_CALL_EVENTS = "on_call_events";
