// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/FunnelComparisonCustomTimeframe.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunnelComparisonCustomTimeframe = void 0;
/**
 * Custom timeframe for funnel comparison.
 */
class FunnelComparisonCustomTimeframe {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FunnelComparisonCustomTimeframe.attributeTypeMap;
    }
}
exports.FunnelComparisonCustomTimeframe = FunnelComparisonCustomTimeframe;
/**
 * @ignore
 */
FunnelComparisonCustomTimeframe.attributeTypeMap = {
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "double",
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "double",
    },
};
