// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HourlyUsageAttributionUsageType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LOGS_INDEXED_180DAY_USAGE = exports.LOGS_INDEXED_15DAY_USAGE = exports.LLM_SPANS_USAGE = exports.LLM_OBSERVABILITY_USAGE = exports.LAMBDA_TRACED_INVOCATIONS_USAGE = exports.INVOCATIONS_USAGE = exports.INGESTED_SPANS_BYTES_USAGE = exports.INGESTED_LOGS_BYTES_USAGE = exports.INFRA_HOST_USAGE = exports.INDEXED_SPANS_USAGE = exports.INCIDENT_MANAGEMENT_MONTHLY_ACTIVE_USERS_USAGE = exports.FUNCTIONS_USAGE = exports.FLEX_STORED_LOGS = exports.FLEX_LOGS_STARTER = exports.FARGATE_USAGE = exports.ESTIMATED_INGESTED_SPANS_USAGE = exports.ESTIMATED_INDEXED_SPANS_USAGE = exports.ERROR_TRACKING_PERCENTAGE = exports.ERROR_TRACKING_USAGE = exports.DBM_QUERIES_USAGE = exports.DBM_HOSTS_USAGE = exports.DATA_STREAM_MONITORING_USAGE = exports.DATA_JOBS_MONITORING_USAGE = exports.CWS_HOSTS_USAGE = exports.CWS_FARGATE_TASK_USAGE = exports.CWS_CONTAINERS_USAGE = exports.CUSTOM_TIMESERIES_USAGE = exports.CUSTOM_INGESTED_TIMESERIES_USAGE = exports.CUSTOM_EVENT_USAGE = exports.CSPM_HOSTS_USAGE = exports.CSPM_CONTAINERS_USAGE = exports.CONTAINER_USAGE = exports.CONTAINER_EXCL_AGENT_USAGE = exports.CODE_SECURITY_HOST_USAGE = exports.CLOUD_SIEM_USAGE = exports.CI_VISIBILITY_ITR_USAGE = exports.CI_TEST_INDEXED_SPANS_USAGE = exports.CI_PIPELINE_INDEXED_SPANS_USAGE = exports.CI_CODE_COVERAGE_COMMITTERS_USAGE = exports.CI_CODE_COVERAGE_COMMITTERS_PERCENTAGE = exports.BROWSER_USAGE = exports.BITS_AI_INVESTIGATIONS_USAGE = exports.ASM_SERVERLESS_TRACED_INVOCATIONS_PERCENTAGE = exports.ASM_SERVERLESS_TRACED_INVOCATIONS_USAGE = exports.APPSEC_USAGE = exports.APPSEC_FARGATE_USAGE = exports.APM_USM_USAGE = exports.APM_HOST_USAGE = exports.APM_FARGATE_USAGE = exports.API_USAGE = void 0;
exports.WORKFLOW_EXECUTIONS_USAGE = exports.VULN_MANAGEMENT_HOSTS_USAGE = exports.UNIVERSAL_SERVICE_MONITORING_USAGE = exports.SNMP_USAGE = exports.SIEM_INGESTED_BYTES_USAGE = exports.SIEM_ANALYZED_LOGS_ADD_ON_USAGE = exports.SERVERLESS_APPS_APM_USAGE = exports.SERVERLESS_APPS_USAGE = exports.SDS_SCANNED_BYTES_USAGE = exports.SCA_FARGATE_USAGE = exports.RUM_SESSION_REPLAY_ADD_ON_USAGE = exports.RUM_REPLAY_SESSIONS_USAGE = exports.RUM_INVESTIGATE_USAGE = exports.RUM_INGESTED_USAGE = exports.RUM_BROWSER_MOBILE_SESSIONS_USAGE = exports.PUBLISHED_APP_USAGE = exports.PROFILED_HOST_USAGE = exports.PROFILED_FARGATE_USAGE = exports.PROFILED_CONTAINER_USAGE = exports.PRODUCT_ANALYTICS_SESSION_USAGE = exports.ONLINE_ARCHIVE_USAGE = exports.OBS_PIPELINE_VCPU_USAGE = exports.OBS_PIPELINE_BYTES_USAGE = exports.NPM_HOST_USAGE = exports.NETWORK_DEVICE_WIRELESS_USAGE = exports.NDM_NETFLOW_USAGE = exports.MOBILE_APP_TESTING_USAGE = exports.LOGS_INDEXED_CUSTOM_RETENTION_USAGE = exports.LOGS_INDEXED_90DAY_USAGE = exports.LOGS_INDEXED_7DAY_USAGE = exports.LOGS_INDEXED_60DAY_USAGE = exports.LOGS_INDEXED_45DAY_USAGE = exports.LOGS_INDEXED_3DAY_USAGE = exports.LOGS_INDEXED_360DAY_USAGE = exports.LOGS_INDEXED_30DAY_USAGE = exports.LOGS_INDEXED_1DAY_USAGE = void 0;
exports.API_USAGE = "api_usage";
exports.APM_FARGATE_USAGE = "apm_fargate_usage";
exports.APM_HOST_USAGE = "apm_host_usage";
exports.APM_USM_USAGE = "apm_usm_usage";
exports.APPSEC_FARGATE_USAGE = "appsec_fargate_usage";
exports.APPSEC_USAGE = "appsec_usage";
exports.ASM_SERVERLESS_TRACED_INVOCATIONS_USAGE = "asm_serverless_traced_invocations_usage";
exports.ASM_SERVERLESS_TRACED_INVOCATIONS_PERCENTAGE = "asm_serverless_traced_invocations_percentage";
exports.BITS_AI_INVESTIGATIONS_USAGE = "bits_ai_investigations_usage";
exports.BROWSER_USAGE = "browser_usage";
exports.CI_CODE_COVERAGE_COMMITTERS_PERCENTAGE = "ci_code_coverage_committers_percentage";
exports.CI_CODE_COVERAGE_COMMITTERS_USAGE = "ci_code_coverage_committers_usage";
exports.CI_PIPELINE_INDEXED_SPANS_USAGE = "ci_pipeline_indexed_spans_usage";
exports.CI_TEST_INDEXED_SPANS_USAGE = "ci_test_indexed_spans_usage";
exports.CI_VISIBILITY_ITR_USAGE = "ci_visibility_itr_usage";
exports.CLOUD_SIEM_USAGE = "cloud_siem_usage";
exports.CODE_SECURITY_HOST_USAGE = "code_security_host_usage";
exports.CONTAINER_EXCL_AGENT_USAGE = "container_excl_agent_usage";
exports.CONTAINER_USAGE = "container_usage";
exports.CSPM_CONTAINERS_USAGE = "cspm_containers_usage";
exports.CSPM_HOSTS_USAGE = "cspm_hosts_usage";
exports.CUSTOM_EVENT_USAGE = "custom_event_usage";
exports.CUSTOM_INGESTED_TIMESERIES_USAGE = "custom_ingested_timeseries_usage";
exports.CUSTOM_TIMESERIES_USAGE = "custom_timeseries_usage";
exports.CWS_CONTAINERS_USAGE = "cws_containers_usage";
exports.CWS_FARGATE_TASK_USAGE = "cws_fargate_task_usage";
exports.CWS_HOSTS_USAGE = "cws_hosts_usage";
exports.DATA_JOBS_MONITORING_USAGE = "data_jobs_monitoring_usage";
exports.DATA_STREAM_MONITORING_USAGE = "data_stream_monitoring_usage";
exports.DBM_HOSTS_USAGE = "dbm_hosts_usage";
exports.DBM_QUERIES_USAGE = "dbm_queries_usage";
exports.ERROR_TRACKING_USAGE = "error_tracking_usage";
exports.ERROR_TRACKING_PERCENTAGE = "error_tracking_percentage";
exports.ESTIMATED_INDEXED_SPANS_USAGE = "estimated_indexed_spans_usage";
exports.ESTIMATED_INGESTED_SPANS_USAGE = "estimated_ingested_spans_usage";
exports.FARGATE_USAGE = "fargate_usage";
exports.FLEX_LOGS_STARTER = "flex_logs_starter";
exports.FLEX_STORED_LOGS = "flex_stored_logs";
exports.FUNCTIONS_USAGE = "functions_usage";
exports.INCIDENT_MANAGEMENT_MONTHLY_ACTIVE_USERS_USAGE = "incident_management_monthly_active_users_usage";
exports.INDEXED_SPANS_USAGE = "indexed_spans_usage";
exports.INFRA_HOST_USAGE = "infra_host_usage";
exports.INGESTED_LOGS_BYTES_USAGE = "ingested_logs_bytes_usage";
exports.INGESTED_SPANS_BYTES_USAGE = "ingested_spans_bytes_usage";
exports.INVOCATIONS_USAGE = "invocations_usage";
exports.LAMBDA_TRACED_INVOCATIONS_USAGE = "lambda_traced_invocations_usage";
exports.LLM_OBSERVABILITY_USAGE = "llm_observability_usage";
exports.LLM_SPANS_USAGE = "llm_spans_usage";
exports.LOGS_INDEXED_15DAY_USAGE = "logs_indexed_15day_usage";
exports.LOGS_INDEXED_180DAY_USAGE = "logs_indexed_180day_usage";
exports.LOGS_INDEXED_1DAY_USAGE = "logs_indexed_1day_usage";
exports.LOGS_INDEXED_30DAY_USAGE = "logs_indexed_30day_usage";
exports.LOGS_INDEXED_360DAY_USAGE = "logs_indexed_360day_usage";
exports.LOGS_INDEXED_3DAY_USAGE = "logs_indexed_3day_usage";
exports.LOGS_INDEXED_45DAY_USAGE = "logs_indexed_45day_usage";
exports.LOGS_INDEXED_60DAY_USAGE = "logs_indexed_60day_usage";
exports.LOGS_INDEXED_7DAY_USAGE = "logs_indexed_7day_usage";
exports.LOGS_INDEXED_90DAY_USAGE = "logs_indexed_90day_usage";
exports.LOGS_INDEXED_CUSTOM_RETENTION_USAGE = "logs_indexed_custom_retention_usage";
exports.MOBILE_APP_TESTING_USAGE = "mobile_app_testing_usage";
exports.NDM_NETFLOW_USAGE = "ndm_netflow_usage";
exports.NETWORK_DEVICE_WIRELESS_USAGE = "npm_host_usage";
exports.NPM_HOST_USAGE = "network_device_wireless_usage";
exports.OBS_PIPELINE_BYTES_USAGE = "obs_pipeline_bytes_usage";
exports.OBS_PIPELINE_VCPU_USAGE = "obs_pipelines_vcpu_usage";
exports.ONLINE_ARCHIVE_USAGE = "online_archive_usage";
exports.PRODUCT_ANALYTICS_SESSION_USAGE = "product_analytics_session_usage";
exports.PROFILED_CONTAINER_USAGE = "profiled_container_usage";
exports.PROFILED_FARGATE_USAGE = "profiled_fargate_usage";
exports.PROFILED_HOST_USAGE = "profiled_host_usage";
exports.PUBLISHED_APP_USAGE = "published_app";
exports.RUM_BROWSER_MOBILE_SESSIONS_USAGE = "rum_browser_mobile_sessions_usage";
exports.RUM_INGESTED_USAGE = "rum_ingested_usage";
exports.RUM_INVESTIGATE_USAGE = "rum_investigate_usage";
exports.RUM_REPLAY_SESSIONS_USAGE = "rum_replay_sessions_usage";
exports.RUM_SESSION_REPLAY_ADD_ON_USAGE = "rum_session_replay_add_on_usage";
exports.SCA_FARGATE_USAGE = "sca_fargate_usage";
exports.SDS_SCANNED_BYTES_USAGE = "sds_scanned_bytes_usage";
exports.SERVERLESS_APPS_USAGE = "serverless_apps_usage";
exports.SERVERLESS_APPS_APM_USAGE = "serverless_apps_apm_usage";
exports.SIEM_ANALYZED_LOGS_ADD_ON_USAGE = "siem_analyzed_logs_add_on_usage";
exports.SIEM_INGESTED_BYTES_USAGE = "siem_ingested_bytes_usage";
exports.SNMP_USAGE = "snmp_usage";
exports.UNIVERSAL_SERVICE_MONITORING_USAGE = "universal_service_monitoring_usage";
exports.VULN_MANAGEMENT_HOSTS_USAGE = "vuln_management_hosts_usage";
exports.WORKFLOW_EXECUTIONS_USAGE = "workflow_executions_usage";
