// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ListStreamSource.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLM_OBSERVABILITY_STREAM = exports.RUM_STREAM = exports.EVENT_STREAM = exports.LOGS_TRANSACTION_STREAM = exports.LOGS_PATTERN_STREAM = exports.LOGS_ISSUE_STREAM = exports.TRACE_STREAM = exports.APM_ISSUE_STREAM = exports.RUM_ISSUE_STREAM = exports.CI_TEST_STREAM = exports.CI_PIPELINE_STREAM = exports.AUDIT_STREAM = exports.LOGS_STREAM = void 0;
exports.LOGS_STREAM = "logs_stream";
exports.AUDIT_STREAM = "audit_stream";
exports.CI_PIPELINE_STREAM = "ci_pipeline_stream";
exports.CI_TEST_STREAM = "ci_test_stream";
exports.RUM_ISSUE_STREAM = "rum_issue_stream";
exports.APM_ISSUE_STREAM = "apm_issue_stream";
exports.TRACE_STREAM = "trace_stream";
exports.LOGS_ISSUE_STREAM = "logs_issue_stream";
exports.LOGS_PATTERN_STREAM = "logs_pattern_stream";
exports.LOGS_TRANSACTION_STREAM = "logs_transaction_stream";
exports.EVENT_STREAM = "event_stream";
exports.RUM_STREAM = "rum_stream";
exports.LLM_OBSERVABILITY_STREAM = "llm_observability_stream";
