// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition = void 0;
/**
 * A formula and functions aggregate augmented query. Used to enrich base query results with data from a reference table.
 */
class MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition = MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition;
/**
 * @ignore
 */
MonitorFormulaAndFunctionAggregateAugmentedQueryDefinition.attributeTypeMap = {
    augmentQuery: {
        baseName: "augment_query",
        type: "MonitorFormulaAndFunctionAggregateAugmentQuery",
        required: true,
    },
    baseQuery: {
        baseName: "base_query",
        type: "MonitorFormulaAndFunctionAggregateBaseQuery",
        required: true,
    },
    compute: {
        baseName: "compute",
        type: "Array<MonitorFormulaAndFunctionEventQueryDefinitionCompute>",
        required: true,
    },
    dataSource: {
        baseName: "data_source",
        type: "MonitorFormulaAndFunctionAggregateAugmentedDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<MonitorFormulaAndFunctionEventQueryGroupBy>",
        required: true,
    },
    joinCondition: {
        baseName: "join_condition",
        type: "MonitorFormulaAndFunctionAggregateQueryJoinCondition",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
};
