// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionAggregateQueryFilter.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionAggregateQueryFilter = void 0;
/**
 * Filter definition for aggregate filtered queries.
 */
class MonitorFormulaAndFunctionAggregateQueryFilter {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionAggregateQueryFilter.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionAggregateQueryFilter = MonitorFormulaAndFunctionAggregateQueryFilter;
/**
 * @ignore
 */
MonitorFormulaAndFunctionAggregateQueryFilter.attributeTypeMap = {
    baseAttribute: {
        baseName: "base_attribute",
        type: "string",
        required: true,
    },
    exclude: {
        baseName: "exclude",
        type: "boolean",
    },
    filterAttribute: {
        baseName: "filter_attribute",
        type: "string",
        required: true,
    },
};
