// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionAggregateFilteredQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionAggregateFilteredQueryDefinition = void 0;
/**
 * A formula and functions aggregate filtered query. Used to filter base query results using data from another source.
 */
class MonitorFormulaAndFunctionAggregateFilteredQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionAggregateFilteredQueryDefinition.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionAggregateFilteredQueryDefinition = MonitorFormulaAndFunctionAggregateFilteredQueryDefinition;
/**
 * @ignore
 */
MonitorFormulaAndFunctionAggregateFilteredQueryDefinition.attributeTypeMap = {
    baseQuery: {
        baseName: "base_query",
        type: "MonitorFormulaAndFunctionAggregateBaseQuery",
        required: true,
    },
    compute: {
        baseName: "compute",
        type: "Array<MonitorFormulaAndFunctionEventQueryDefinitionCompute>",
    },
    dataSource: {
        baseName: "data_source",
        type: "MonitorFormulaAndFunctionAggregateFilteredDataSource",
        required: true,
    },
    filterQuery: {
        baseName: "filter_query",
        type: "MonitorFormulaAndFunctionAggregateFilterQuery",
        required: true,
    },
    filters: {
        baseName: "filters",
        type: "Array<MonitorFormulaAndFunctionAggregateQueryFilter>",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<MonitorFormulaAndFunctionEventQueryGroupBy>",
    },
    name: {
        baseName: "name",
        type: "string",
    },
};
