// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NETWORK_PATH_ALERT = exports.DATA_QUALITY_ALERT = exports.COST_ALERT = exports.NETWORK_PERFORMANCE_ALERT = exports.DATABASE_MONITORING_ALERT = exports.ERROR_TRACKING_ALERT = exports.CI_TESTS_ALERT = exports.CI_PIPELINES_ALERT = exports.AUDIT_ALERT = exports.EVENT_V2_ALERT = exports.SLO_ALERT = exports.TRACE_ANALYTICS_ALERT = exports.SYNTHETICS_ALERT = exports.SERVICE_CHECK = exports.RUM_ALERT = exports.QUERY_ALERT = exports.PROCESS_ALERT = exports.METRIC_ALERT = exports.LOG_ALERT = exports.EVENT_ALERT = exports.COMPOSITE = void 0;
exports.COMPOSITE = "composite";
exports.EVENT_ALERT = "event alert";
exports.LOG_ALERT = "log alert";
exports.METRIC_ALERT = "metric alert";
exports.PROCESS_ALERT = "process alert";
exports.QUERY_ALERT = "query alert";
exports.RUM_ALERT = "rum alert";
exports.SERVICE_CHECK = "service check";
exports.SYNTHETICS_ALERT = "synthetics alert";
exports.TRACE_ANALYTICS_ALERT = "trace-analytics alert";
exports.SLO_ALERT = "slo alert";
exports.EVENT_V2_ALERT = "event-v2 alert";
exports.AUDIT_ALERT = "audit alert";
exports.CI_PIPELINES_ALERT = "ci-pipelines alert";
exports.CI_TESTS_ALERT = "ci-tests alert";
exports.ERROR_TRACKING_ALERT = "error-tracking alert";
exports.DATABASE_MONITORING_ALERT = "database-monitoring alert";
exports.NETWORK_PERFORMANCE_ALERT = "network-performance alert";
exports.COST_ALERT = "cost alert";
exports.DATA_QUALITY_ALERT = "data-quality alert";
exports.NETWORK_PATH_ALERT = "network-path alert";
