// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionMetricsQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionMetricsQueryDefinition = void 0;
/**
 * A formula and functions metrics query for use in aggregate queries.
 */
class MonitorFormulaAndFunctionMetricsQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionMetricsQueryDefinition.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionMetricsQueryDefinition = MonitorFormulaAndFunctionMetricsQueryDefinition;
/**
 * @ignore
 */
MonitorFormulaAndFunctionMetricsQueryDefinition.attributeTypeMap = {
    aggregator: {
        baseName: "aggregator",
        type: "MonitorFormulaAndFunctionMetricsAggregator",
    },
    dataSource: {
        baseName: "data_source",
        type: "MonitorFormulaAndFunctionMetricsDataSource",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
};
