// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionReferenceTableColumn.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionReferenceTableColumn = void 0;
/**
 * A column definition for reference table queries.
 */
class MonitorFormulaAndFunctionReferenceTableColumn {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionReferenceTableColumn.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionReferenceTableColumn = MonitorFormulaAndFunctionReferenceTableColumn;
/**
 * @ignore
 */
MonitorFormulaAndFunctionReferenceTableColumn.attributeTypeMap = {
    alias: {
        baseName: "alias",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
};
