// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionReferenceTableQueryDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionReferenceTableQueryDefinition = void 0;
/**
 * A reference table query for use in aggregate queries.
 */
class MonitorFormulaAndFunctionReferenceTableQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionReferenceTableQueryDefinition.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionReferenceTableQueryDefinition = MonitorFormulaAndFunctionReferenceTableQueryDefinition;
/**
 * @ignore
 */
MonitorFormulaAndFunctionReferenceTableQueryDefinition.attributeTypeMap = {
    columns: {
        baseName: "columns",
        type: "Array<MonitorFormulaAndFunctionReferenceTableColumn>",
    },
    dataSource: {
        baseName: "data_source",
        type: "MonitorFormulaAndFunctionReferenceTableDataSource",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    queryFilter: {
        baseName: "query_filter",
        type: "string",
    },
    tableName: {
        baseName: "table_name",
        type: "string",
        required: true,
    },
};
