// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionAggregateQueryJoinCondition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionAggregateQueryJoinCondition = void 0;
/**
 * Join condition for aggregate augmented queries.
 */
class MonitorFormulaAndFunctionAggregateQueryJoinCondition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionAggregateQueryJoinCondition.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionAggregateQueryJoinCondition = MonitorFormulaAndFunctionAggregateQueryJoinCondition;
/**
 * @ignore
 */
MonitorFormulaAndFunctionAggregateQueryJoinCondition.attributeTypeMap = {
    augmentAttribute: {
        baseName: "augment_attribute",
        type: "string",
        required: true,
    },
    baseAttribute: {
        baseName: "base_attribute",
        type: "string",
        required: true,
    },
    joinType: {
        baseName: "join_type",
        type: "MonitorFormulaAndFunctionAggregateQueryJoinType",
        required: true,
    },
};
