// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonthlyUsageAttributionSupportedMetrics.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DBM_QUERIES_USAGE = exports.DBM_QUERIES_PERCENTAGE = exports.DBM_HOSTS_USAGE = exports.DBM_HOSTS_PERCENTAGE = exports.DATA_STREAM_MONITORING_PERCENTAGE = exports.DATA_STREAM_MONITORING_USAGE = exports.DATA_JOBS_MONITORING_PERCENTAGE = exports.DATA_JOBS_MONITORING_USAGE = exports.CWS_HOSTS_USAGE = exports.CWS_HOSTS_PERCENTAGE = exports.CWS_FARGATE_TASK_USAGE = exports.CWS_FARGATE_TASK_PERCENTAGE = exports.CWS_CONTAINERS_USAGE = exports.CWS_CONTAINERS_PERCENTAGE = exports.CUSTOM_INGESTED_TIMESERIES_PERCENTAGE = exports.CUSTOM_INGESTED_TIMESERIES_USAGE = exports.CUSTOM_TIMESERIES_PERCENTAGE = exports.CUSTOM_TIMESERIES_USAGE = exports.CSPM_HOSTS_USAGE = exports.CSPM_HOSTS_PERCENTAGE = exports.CSPM_CONTAINERS_USAGE = exports.CSPM_CONTAINERS_PERCENTAGE = exports.CONTAINER_PERCENTAGE = exports.CONTAINER_USAGE = exports.CONTAINER_EXCL_AGENT_PERCENTAGE = exports.CONTAINER_EXCL_AGENT_USAGE = exports.CODE_SECURITY_HOST_PERCENTAGE = exports.CODE_SECURITY_HOST_USAGE = exports.CLOUD_SIEM_PERCENTAGE = exports.CLOUD_SIEM_USAGE = exports.CI_VISIBILITY_ITR_PERCENTAGE = exports.CI_VISIBILITY_ITR_USAGE = exports.BROWSER_PERCENTAGE = exports.BROWSER_USAGE = exports.BITS_AI_INVESTIGATIONS_PERCENTAGE = exports.BITS_AI_INVESTIGATIONS_USAGE = exports.ASM_SERVERLESS_TRACED_INVOCATIONS_PERCENTAGE = exports.ASM_SERVERLESS_TRACED_INVOCATIONS_USAGE = exports.APPSEC_PERCENTAGE = exports.APPSEC_USAGE = exports.APM_USM_PERCENTAGE = exports.APM_USM_USAGE = exports.APM_HOST_PERCENTAGE = exports.APM_HOST_USAGE = exports.APPSEC_FARGATE_PERCENTAGE = exports.APPSEC_FARGATE_USAGE = exports.APM_FARGATE_PERCENTAGE = exports.APM_FARGATE_USAGE = exports.API_PERCENTAGE = exports.API_USAGE = void 0;
exports.PUBLISHED_APP_PERCENTAGE = exports.PUBLISHED_APP_USAGE = exports.PROFILED_HOST_PERCENTAGE = exports.PROFILED_HOST_USAGE = exports.PROFILED_FARGATE_PERCENTAGE = exports.PROFILED_FARGATE_USAGE = exports.PROFILED_CONTAINER_PERCENTAGE = exports.PROFILED_CONTAINER_USAGE = exports.PRODUCT_ANALYTICS_SESSION_PERCENTAGE = exports.PRODUCT_ANALYTICS_SESSION_USAGE = exports.ONLINE_ARCHIVE_PERCENTAGE = exports.ONLINE_ARCHIVE_USAGE = exports.OBS_PIPELINES_VCPU_PERCENTAGE = exports.OBS_PIPELINES_VCPU_USAGE = exports.OBS_PIPELINE_BYTES_PERCENTAGE = exports.OBS_PIPELINE_BYTES_USAGE = exports.NPM_HOST_PERCENTAGE = exports.NPM_HOST_USAGE = exports.NETWORK_DEVICE_WIRELESS_PERCENTAGE = exports.NETWORK_DEVICE_WIRELESS_USAGE = exports.NDM_NETFLOW_PERCENTAGE = exports.NDM_NETFLOW_USAGE = exports.MOBILE_APP_TESTING_PERCENTAGE = exports.MOBILE_APP_TESTING_USAGE = exports.LLM_SPANS_PERCENTAGE = exports.LLM_SPANS_USAGE = exports.LLM_OBSERVABILITY_PERCENTAGE = exports.LLM_OBSERVABILITY_USAGE = exports.LAMBDA_TRACED_INVOCATIONS_PERCENTAGE = exports.LAMBDA_TRACED_INVOCATIONS_USAGE = exports.INVOCATIONS_PERCENTAGE = exports.INVOCATIONS_USAGE = exports.INFRA_HOST_PERCENTAGE = exports.INFRA_HOST_USAGE = exports.INCIDENT_MANAGEMENT_MONTHLY_ACTIVE_USERS_PERCENTAGE = exports.INCIDENT_MANAGEMENT_MONTHLY_ACTIVE_USERS_USAGE = exports.FUNCTIONS_PERCENTAGE = exports.FUNCTIONS_USAGE = exports.FLEX_STORED_LOGS_PERCENTAGE = exports.FLEX_STORED_LOGS_USAGE = exports.FLEX_LOGS_STARTER_PERCENTAGE = exports.FLEX_LOGS_STARTER_USAGE = exports.FARGATE_PERCENTAGE = exports.FARGATE_USAGE = exports.ESTIMATED_INGESTED_SPANS_PERCENTAGE = exports.ESTIMATED_INGESTED_SPANS_USAGE = exports.ESTIMATED_INDEXED_SPANS_PERCENTAGE = exports.ESTIMATED_INDEXED_SPANS_USAGE = exports.ERROR_TRACKING_PERCENTAGE = exports.ERROR_TRACKING_USAGE = void 0;
exports.RUM_REPLAY_SESSIONS_PERCENTAGE = exports.RUM_REPLAY_SESSIONS_USAGE = exports.RUM_INVESTIGATE_PERCENTAGE = exports.RUM_INVESTIGATE_USAGE = exports.RUM_INGESTED_PERCENTAGE = exports.RUM_INGESTED_USAGE = exports.LOGS_INDEXED_1DAY_PERCENTAGE = exports.LOGS_INDEXED_1DAY_USAGE = exports.LOGS_INDEXED_3DAY_PERCENTAGE = exports.LOGS_INDEXED_3DAY_USAGE = exports.LOGS_INDEXED_7DAY_PERCENTAGE = exports.LOGS_INDEXED_7DAY_USAGE = exports.LOGS_INDEXED_15DAY_PERCENTAGE = exports.LOGS_INDEXED_15DAY_USAGE = exports.LOGS_INDEXED_30DAY_PERCENTAGE = exports.LOGS_INDEXED_30DAY_USAGE = exports.LOGS_INDEXED_45DAY_PERCENTAGE = exports.LOGS_INDEXED_45DAY_USAGE = exports.LOGS_INDEXED_60DAY_PERCENTAGE = exports.LOGS_INDEXED_60DAY_USAGE = exports.LOGS_INDEXED_90DAY_PERCENTAGE = exports.LOGS_INDEXED_90DAY_USAGE = exports.LOGS_INDEXED_180DAY_PERCENTAGE = exports.LOGS_INDEXED_180DAY_USAGE = exports.LOGS_INDEXED_360DAY_PERCENTAGE = exports.LOGS_INDEXED_360DAY_USAGE = exports.LOGS_INDEXED_CUSTOM_RETENTION_PERCENTAGE = exports.LOGS_INDEXED_CUSTOM_RETENTION_USAGE = exports.CUSTOM_EVENT_PERCENTAGE = exports.CUSTOM_EVENT_USAGE = exports.INDEXED_SPANS_PERCENTAGE = exports.INDEXED_SPANS_USAGE = exports.CI_PIPELINE_INDEXED_SPANS_PERCENTAGE = exports.CI_PIPELINE_INDEXED_SPANS_USAGE = exports.INGESTED_LOGS_BYTES_PERCENTAGE = exports.INGESTED_LOGS_BYTES_USAGE = exports.CI_TEST_INDEXED_SPANS_PERCENTAGE = exports.CI_TEST_INDEXED_SPANS_USAGE = exports.SDS_SCANNED_BYTES_PERCENTAGE = exports.SDS_SCANNED_BYTES_USAGE = exports.VULN_MANAGEMENT_HOSTS_PERCENTAGE = exports.VULN_MANAGEMENT_HOSTS_USAGE = exports.UNIVERSAL_SERVICE_MONITORING_PERCENTAGE = exports.UNIVERSAL_SERVICE_MONITORING_USAGE = exports.SNMP_PERCENTAGE = exports.SNMP_USAGE = exports.SERVERLESS_APPS_APM_PERCENTAGE = exports.SERVERLESS_APPS_APM_USAGE = exports.SERVERLESS_APPS_PERCENTAGE = exports.SERVERLESS_APPS_USAGE = void 0;
exports.ALL = exports.SCA_FARGATE_PERCENTAGE = exports.SCA_FARGATE_USAGE = exports.WORKFLOW_EXECUTIONS_PERCENTAGE = exports.WORKFLOW_EXECUTIONS_USAGE = exports.SIEM_INGESTED_BYTES_PERCENTAGE = exports.SIEM_INGESTED_BYTES_USAGE = exports.SIEM_ANALYZED_LOGS_ADD_ON_PERCENTAGE = exports.SIEM_ANALYZED_LOGS_ADD_ON_USAGE = exports.INGESTED_SPANS_BYTES_PERCENTAGE = exports.INGESTED_SPANS_BYTES_USAGE = exports.RUM_BROWSER_MOBILE_SESSIONS_PERCENTAGE = exports.RUM_BROWSER_MOBILE_SESSIONS_USAGE = exports.RUM_SESSION_REPLAY_ADD_ON_PERCENTAGE = exports.RUM_SESSION_REPLAY_ADD_ON_USAGE = void 0;
exports.API_USAGE = "api_usage";
exports.API_PERCENTAGE = "api_percentage";
exports.APM_FARGATE_USAGE = "apm_fargate_usage";
exports.APM_FARGATE_PERCENTAGE = "apm_fargate_percentage";
exports.APPSEC_FARGATE_USAGE = "appsec_fargate_usage";
exports.APPSEC_FARGATE_PERCENTAGE = "appsec_fargate_percentage";
exports.APM_HOST_USAGE = "apm_host_usage";
exports.APM_HOST_PERCENTAGE = "apm_host_percentage";
exports.APM_USM_USAGE = "apm_usm_usage";
exports.APM_USM_PERCENTAGE = "apm_usm_percentage";
exports.APPSEC_USAGE = "appsec_usage";
exports.APPSEC_PERCENTAGE = "appsec_percentage";
exports.ASM_SERVERLESS_TRACED_INVOCATIONS_USAGE = "asm_serverless_traced_invocations_usage";
exports.ASM_SERVERLESS_TRACED_INVOCATIONS_PERCENTAGE = "asm_serverless_traced_invocations_percentage";
exports.BITS_AI_INVESTIGATIONS_USAGE = "bits_ai_investigations_usage";
exports.BITS_AI_INVESTIGATIONS_PERCENTAGE = "bits_ai_investigations_percentage";
exports.BROWSER_USAGE = "browser_usage";
exports.BROWSER_PERCENTAGE = "browser_percentage";
exports.CI_VISIBILITY_ITR_USAGE = "ci_visibility_itr_usage";
exports.CI_VISIBILITY_ITR_PERCENTAGE = "ci_visibility_itr_percentage";
exports.CLOUD_SIEM_USAGE = "cloud_siem_usage";
exports.CLOUD_SIEM_PERCENTAGE = "cloud_siem_percentage";
exports.CODE_SECURITY_HOST_USAGE = "code_security_host_usage";
exports.CODE_SECURITY_HOST_PERCENTAGE = "code_security_host_percentage";
exports.CONTAINER_EXCL_AGENT_USAGE = "container_excl_agent_usage";
exports.CONTAINER_EXCL_AGENT_PERCENTAGE = "container_excl_agent_percentage";
exports.CONTAINER_USAGE = "container_usage";
exports.CONTAINER_PERCENTAGE = "container_percentage";
exports.CSPM_CONTAINERS_PERCENTAGE = "cspm_containers_percentage";
exports.CSPM_CONTAINERS_USAGE = "cspm_containers_usage";
exports.CSPM_HOSTS_PERCENTAGE = "cspm_hosts_percentage";
exports.CSPM_HOSTS_USAGE = "cspm_hosts_usage";
exports.CUSTOM_TIMESERIES_USAGE = "custom_timeseries_usage";
exports.CUSTOM_TIMESERIES_PERCENTAGE = "custom_timeseries_percentage";
exports.CUSTOM_INGESTED_TIMESERIES_USAGE = "custom_ingested_timeseries_usage";
exports.CUSTOM_INGESTED_TIMESERIES_PERCENTAGE = "custom_ingested_timeseries_percentage";
exports.CWS_CONTAINERS_PERCENTAGE = "cws_containers_percentage";
exports.CWS_CONTAINERS_USAGE = "cws_containers_usage";
exports.CWS_FARGATE_TASK_PERCENTAGE = "cws_fargate_task_percentage";
exports.CWS_FARGATE_TASK_USAGE = "cws_fargate_task_usage";
exports.CWS_HOSTS_PERCENTAGE = "cws_hosts_percentage";
exports.CWS_HOSTS_USAGE = "cws_hosts_usage";
exports.DATA_JOBS_MONITORING_USAGE = "data_jobs_monitoring_usage";
exports.DATA_JOBS_MONITORING_PERCENTAGE = "data_jobs_monitoring_percentage";
exports.DATA_STREAM_MONITORING_USAGE = "data_stream_monitoring_usage";
exports.DATA_STREAM_MONITORING_PERCENTAGE = "data_stream_monitoring_percentage";
exports.DBM_HOSTS_PERCENTAGE = "dbm_hosts_percentage";
exports.DBM_HOSTS_USAGE = "dbm_hosts_usage";
exports.DBM_QUERIES_PERCENTAGE = "dbm_queries_percentage";
exports.DBM_QUERIES_USAGE = "dbm_queries_usage";
exports.ERROR_TRACKING_USAGE = "error_tracking_usage";
exports.ERROR_TRACKING_PERCENTAGE = "error_tracking_percentage";
exports.ESTIMATED_INDEXED_SPANS_USAGE = "estimated_indexed_spans_usage";
exports.ESTIMATED_INDEXED_SPANS_PERCENTAGE = "estimated_indexed_spans_percentage";
exports.ESTIMATED_INGESTED_SPANS_USAGE = "estimated_ingested_spans_usage";
exports.ESTIMATED_INGESTED_SPANS_PERCENTAGE = "estimated_ingested_spans_percentage";
exports.FARGATE_USAGE = "fargate_usage";
exports.FARGATE_PERCENTAGE = "fargate_percentage";
exports.FLEX_LOGS_STARTER_USAGE = "flex_logs_starter_usage";
exports.FLEX_LOGS_STARTER_PERCENTAGE = "flex_logs_starter_percentage";
exports.FLEX_STORED_LOGS_USAGE = "flex_stored_logs_usage";
exports.FLEX_STORED_LOGS_PERCENTAGE = "flex_stored_logs_percentage";
exports.FUNCTIONS_USAGE = "functions_usage";
exports.FUNCTIONS_PERCENTAGE = "functions_percentage";
exports.INCIDENT_MANAGEMENT_MONTHLY_ACTIVE_USERS_USAGE = "incident_management_monthly_active_users_usage";
exports.INCIDENT_MANAGEMENT_MONTHLY_ACTIVE_USERS_PERCENTAGE = "incident_management_monthly_active_users_percentage";
exports.INFRA_HOST_USAGE = "infra_host_usage";
exports.INFRA_HOST_PERCENTAGE = "infra_host_percentage";
exports.INVOCATIONS_USAGE = "invocations_usage";
exports.INVOCATIONS_PERCENTAGE = "invocations_percentage";
exports.LAMBDA_TRACED_INVOCATIONS_USAGE = "lambda_traced_invocations_usage";
exports.LAMBDA_TRACED_INVOCATIONS_PERCENTAGE = "lambda_traced_invocations_percentage";
exports.LLM_OBSERVABILITY_USAGE = "llm_observability_usage";
exports.LLM_OBSERVABILITY_PERCENTAGE = "llm_observability_percentage";
exports.LLM_SPANS_USAGE = "llm_spans_usage";
exports.LLM_SPANS_PERCENTAGE = "llm_spans_percentage";
exports.MOBILE_APP_TESTING_USAGE = "mobile_app_testing_percentage";
exports.MOBILE_APP_TESTING_PERCENTAGE = "mobile_app_testing_usage";
exports.NDM_NETFLOW_USAGE = "ndm_netflow_usage";
exports.NDM_NETFLOW_PERCENTAGE = "ndm_netflow_percentage";
exports.NETWORK_DEVICE_WIRELESS_USAGE = "network_device_wireless_usage";
exports.NETWORK_DEVICE_WIRELESS_PERCENTAGE = "network_device_wireless_percentage";
exports.NPM_HOST_USAGE = "npm_host_usage";
exports.NPM_HOST_PERCENTAGE = "npm_host_percentage";
exports.OBS_PIPELINE_BYTES_USAGE = "obs_pipeline_bytes_usage";
exports.OBS_PIPELINE_BYTES_PERCENTAGE = "obs_pipeline_bytes_percentage";
exports.OBS_PIPELINES_VCPU_USAGE = "obs_pipelines_vcpu_usage";
exports.OBS_PIPELINES_VCPU_PERCENTAGE = "obs_pipelines_vcpu_percentage";
exports.ONLINE_ARCHIVE_USAGE = "online_archive_usage";
exports.ONLINE_ARCHIVE_PERCENTAGE = "online_archive_percentage";
exports.PRODUCT_ANALYTICS_SESSION_USAGE = "product_analytics_session_usage";
exports.PRODUCT_ANALYTICS_SESSION_PERCENTAGE = "product_analytics_session_percentage";
exports.PROFILED_CONTAINER_USAGE = "profiled_container_usage";
exports.PROFILED_CONTAINER_PERCENTAGE = "profiled_container_percentage";
exports.PROFILED_FARGATE_USAGE = "profiled_fargate_usage";
exports.PROFILED_FARGATE_PERCENTAGE = "profiled_fargate_percentage";
exports.PROFILED_HOST_USAGE = "profiled_host_usage";
exports.PROFILED_HOST_PERCENTAGE = "profiled_host_percentage";
exports.PUBLISHED_APP_USAGE = "published_app_usage";
exports.PUBLISHED_APP_PERCENTAGE = "published_app_percentage";
exports.SERVERLESS_APPS_USAGE = "serverless_apps_usage";
exports.SERVERLESS_APPS_PERCENTAGE = "serverless_apps_percentage";
exports.SERVERLESS_APPS_APM_USAGE = "serverless_apps_apm_usage";
exports.SERVERLESS_APPS_APM_PERCENTAGE = "serverless_apps_apm_percentage";
exports.SNMP_USAGE = "snmp_usage";
exports.SNMP_PERCENTAGE = "snmp_percentage";
exports.UNIVERSAL_SERVICE_MONITORING_USAGE = "universal_service_monitoring_usage";
exports.UNIVERSAL_SERVICE_MONITORING_PERCENTAGE = "universal_service_monitoring_percentage";
exports.VULN_MANAGEMENT_HOSTS_USAGE = "vuln_management_hosts_usage";
exports.VULN_MANAGEMENT_HOSTS_PERCENTAGE = "vuln_management_hosts_percentage";
exports.SDS_SCANNED_BYTES_USAGE = "sds_scanned_bytes_usage";
exports.SDS_SCANNED_BYTES_PERCENTAGE = "sds_scanned_bytes_percentage";
exports.CI_TEST_INDEXED_SPANS_USAGE = "ci_test_indexed_spans_usage";
exports.CI_TEST_INDEXED_SPANS_PERCENTAGE = "ci_test_indexed_spans_percentage";
exports.INGESTED_LOGS_BYTES_USAGE = "ingested_logs_bytes_usage";
exports.INGESTED_LOGS_BYTES_PERCENTAGE = "ingested_logs_bytes_percentage";
exports.CI_PIPELINE_INDEXED_SPANS_USAGE = "ci_pipeline_indexed_spans_usage";
exports.CI_PIPELINE_INDEXED_SPANS_PERCENTAGE = "ci_pipeline_indexed_spans_percentage";
exports.INDEXED_SPANS_USAGE = "indexed_spans_usage";
exports.INDEXED_SPANS_PERCENTAGE = "indexed_spans_percentage";
exports.CUSTOM_EVENT_USAGE = "custom_event_usage";
exports.CUSTOM_EVENT_PERCENTAGE = "custom_event_percentage";
exports.LOGS_INDEXED_CUSTOM_RETENTION_USAGE = "logs_indexed_custom_retention_usage";
exports.LOGS_INDEXED_CUSTOM_RETENTION_PERCENTAGE = "logs_indexed_custom_retention_percentage";
exports.LOGS_INDEXED_360DAY_USAGE = "logs_indexed_360day_usage";
exports.LOGS_INDEXED_360DAY_PERCENTAGE = "logs_indexed_360day_percentage";
exports.LOGS_INDEXED_180DAY_USAGE = "logs_indexed_180day_usage";
exports.LOGS_INDEXED_180DAY_PERCENTAGE = "logs_indexed_180day_percentage";
exports.LOGS_INDEXED_90DAY_USAGE = "logs_indexed_90day_usage";
exports.LOGS_INDEXED_90DAY_PERCENTAGE = "logs_indexed_90day_percentage";
exports.LOGS_INDEXED_60DAY_USAGE = "logs_indexed_60day_usage";
exports.LOGS_INDEXED_60DAY_PERCENTAGE = "logs_indexed_60day_percentage";
exports.LOGS_INDEXED_45DAY_USAGE = "logs_indexed_45day_usage";
exports.LOGS_INDEXED_45DAY_PERCENTAGE = "logs_indexed_45day_percentage";
exports.LOGS_INDEXED_30DAY_USAGE = "logs_indexed_30day_usage";
exports.LOGS_INDEXED_30DAY_PERCENTAGE = "logs_indexed_30day_percentage";
exports.LOGS_INDEXED_15DAY_USAGE = "logs_indexed_15day_usage";
exports.LOGS_INDEXED_15DAY_PERCENTAGE = "logs_indexed_15day_percentage";
exports.LOGS_INDEXED_7DAY_USAGE = "logs_indexed_7day_usage";
exports.LOGS_INDEXED_7DAY_PERCENTAGE = "logs_indexed_7day_percentage";
exports.LOGS_INDEXED_3DAY_USAGE = "logs_indexed_3day_usage";
exports.LOGS_INDEXED_3DAY_PERCENTAGE = "logs_indexed_3day_percentage";
exports.LOGS_INDEXED_1DAY_USAGE = "logs_indexed_1day_usage";
exports.LOGS_INDEXED_1DAY_PERCENTAGE = "logs_indexed_1day_percentage";
exports.RUM_INGESTED_USAGE = "rum_ingested_usage";
exports.RUM_INGESTED_PERCENTAGE = "rum_ingested_percentage";
exports.RUM_INVESTIGATE_USAGE = "rum_investigate_usage";
exports.RUM_INVESTIGATE_PERCENTAGE = "rum_investigate_percentage";
exports.RUM_REPLAY_SESSIONS_USAGE = "rum_replay_sessions_usage";
exports.RUM_REPLAY_SESSIONS_PERCENTAGE = "rum_replay_sessions_percentage";
exports.RUM_SESSION_REPLAY_ADD_ON_USAGE = "rum_session_replay_add_on_usage";
exports.RUM_SESSION_REPLAY_ADD_ON_PERCENTAGE = "rum_session_replay_add_on_percentage";
exports.RUM_BROWSER_MOBILE_SESSIONS_USAGE = "rum_browser_mobile_sessions_usage";
exports.RUM_BROWSER_MOBILE_SESSIONS_PERCENTAGE = "rum_browser_mobile_sessions_percentage";
exports.INGESTED_SPANS_BYTES_USAGE = "ingested_spans_bytes_usage";
exports.INGESTED_SPANS_BYTES_PERCENTAGE = "ingested_spans_bytes_percentage";
exports.SIEM_ANALYZED_LOGS_ADD_ON_USAGE = "siem_analyzed_logs_add_on_usage";
exports.SIEM_ANALYZED_LOGS_ADD_ON_PERCENTAGE = "siem_analyzed_logs_add_on_percentage";
exports.SIEM_INGESTED_BYTES_USAGE = "siem_ingested_bytes_usage";
exports.SIEM_INGESTED_BYTES_PERCENTAGE = "siem_ingested_bytes_percentage";
exports.WORKFLOW_EXECUTIONS_USAGE = "workflow_executions_usage";
exports.WORKFLOW_EXECUTIONS_PERCENTAGE = "workflow_executions_percentage";
exports.SCA_FARGATE_USAGE = "sca_fargate_usage";
exports.SCA_FARGATE_PERCENTAGE = "sca_fargate_percentage";
exports.ALL = "*";
