// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ProductAnalyticsAudienceOccurrenceFilter.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsAudienceOccurrenceFilter = void 0;
/**
 * Filter applied to occurrence counts when building a Product Analytics audience.
 */
class ProductAnalyticsAudienceOccurrenceFilter {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsAudienceOccurrenceFilter.attributeTypeMap;
    }
}
exports.ProductAnalyticsAudienceOccurrenceFilter = ProductAnalyticsAudienceOccurrenceFilter;
/**
 * @ignore
 */
ProductAnalyticsAudienceOccurrenceFilter.attributeTypeMap = {
    operator: {
        baseName: "operator",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
