// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ProductAnalyticsFunnelWidgetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsFunnelWidgetDefinition = void 0;
/**
 * The user journey funnel visualization displays conversion funnels based on user journey data from Product Analytics.
 */
class ProductAnalyticsFunnelWidgetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsFunnelWidgetDefinition.attributeTypeMap;
    }
}
exports.ProductAnalyticsFunnelWidgetDefinition = ProductAnalyticsFunnelWidgetDefinition;
/**
 * @ignore
 */
ProductAnalyticsFunnelWidgetDefinition.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    groupedDisplay: {
        baseName: "grouped_display",
        type: "FunnelGroupedDisplay",
    },
    requests: {
        baseName: "requests",
        type: "Array<ProductAnalyticsFunnelRequest>",
        required: true,
    },
    time: {
        baseName: "time",
        type: "WidgetTime",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    titleAlign: {
        baseName: "title_align",
        type: "WidgetTextAlign",
    },
    titleSize: {
        baseName: "title_size",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "FunnelWidgetDefinitionType",
        required: true,
    },
};
