// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ProductAnalyticsFunnelQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsFunnelQuery = void 0;
/**
 * User journey funnel query definition.
 */
class ProductAnalyticsFunnelQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsFunnelQuery.attributeTypeMap;
    }
}
exports.ProductAnalyticsFunnelQuery = ProductAnalyticsFunnelQuery;
/**
 * @ignore
 */
ProductAnalyticsFunnelQuery.attributeTypeMap = {
    compute: {
        baseName: "compute",
        type: "ProductAnalyticsFunnelCompute",
    },
    dataSource: {
        baseName: "data_source",
        type: "ProductAnalyticsFunnelDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<ProductAnalyticsFunnelGroupBy>",
    },
    search: {
        baseName: "search",
        type: "UserJourneySearch",
        required: true,
    },
    subqueryId: {
        baseName: "subquery_id",
        type: "string",
    },
};
