// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ProductAnalyticsExtendedGroupBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsExtendedGroupBy = void 0;
/**
 * Group by configuration for Product Analytics Extended queries.
 */
class ProductAnalyticsExtendedGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsExtendedGroupBy.attributeTypeMap;
    }
}
exports.ProductAnalyticsExtendedGroupBy = ProductAnalyticsExtendedGroupBy;
/**
 * @ignore
 */
ProductAnalyticsExtendedGroupBy.attributeTypeMap = {
    facet: {
        baseName: "facet",
        type: "string",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int32",
    },
    shouldExcludeMissing: {
        baseName: "should_exclude_missing",
        type: "boolean",
    },
    sort: {
        baseName: "sort",
        type: "FormulaAndFunctionEventQueryGroupBySort",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
