// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ProductAnalyticsExtendedCompute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsExtendedCompute = void 0;
/**
 * Compute configuration for Product Analytics Extended queries.
 */
class ProductAnalyticsExtendedCompute {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsExtendedCompute.attributeTypeMap;
    }
}
exports.ProductAnalyticsExtendedCompute = ProductAnalyticsExtendedCompute;
/**
 * @ignore
 */
ProductAnalyticsExtendedCompute.attributeTypeMap = {
    aggregation: {
        baseName: "aggregation",
        type: "FormulaAndFunctionEventAggregation",
        required: true,
    },
    interval: {
        baseName: "interval",
        type: "number",
        format: "double",
    },
    metric: {
        baseName: "metric",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    rollup: {
        baseName: "rollup",
        type: "CalendarInterval",
    },
};
