// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ProductAnalyticsFunnelGroupBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsFunnelGroupBy = void 0;
/**
 * Group by configuration for user journey funnel.
 */
class ProductAnalyticsFunnelGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsFunnelGroupBy.attributeTypeMap;
    }
}
exports.ProductAnalyticsFunnelGroupBy = ProductAnalyticsFunnelGroupBy;
/**
 * @ignore
 */
ProductAnalyticsFunnelGroupBy.attributeTypeMap = {
    facet: {
        baseName: "facet",
        type: "string",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    shouldExcludeMissing: {
        baseName: "should_exclude_missing",
        type: "boolean",
    },
    sort: {
        baseName: "sort",
        type: "ProductAnalyticsFunnelGroupBySort",
    },
    target: {
        baseName: "target",
        type: "UserJourneySearchTarget",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
