// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/RetentionSearch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetentionSearch = void 0;
/**
 * Search configuration for retention queries.
 */
class RetentionSearch {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RetentionSearch.attributeTypeMap;
    }
}
exports.RetentionSearch = RetentionSearch;
/**
 * @ignore
 */
RetentionSearch.attributeTypeMap = {
    cohortCriteria: {
        baseName: "cohort_criteria",
        type: "RetentionCohortCriteria",
        required: true,
    },
    filters: {
        baseName: "filters",
        type: "RetentionFilters",
    },
    retentionEntity: {
        baseName: "retention_entity",
        type: "RetentionEntity",
        required: true,
    },
    returnCondition: {
        baseName: "return_condition",
        type: "RetentionReturnCondition",
        required: true,
    },
    returnCriteria: {
        baseName: "return_criteria",
        type: "RetentionReturnCriteria",
    },
};
