// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/RetentionCurveWidgetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetentionCurveWidgetDefinition = void 0;
/**
 * The retention curve widget visualizes user retention rates over time.
 */
class RetentionCurveWidgetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RetentionCurveWidgetDefinition.attributeTypeMap;
    }
}
exports.RetentionCurveWidgetDefinition = RetentionCurveWidgetDefinition;
/**
 * @ignore
 */
RetentionCurveWidgetDefinition.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    requests: {
        baseName: "requests",
        type: "Array<RetentionCurveWidgetRequest>",
        required: true,
    },
    time: {
        baseName: "time",
        type: "WidgetTime",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    titleAlign: {
        baseName: "title_align",
        type: "WidgetTextAlign",
    },
    titleSize: {
        baseName: "title_size",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "RetentionCurveWidgetDefinitionType",
        required: true,
    },
};
