// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/RetentionGroupBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetentionGroupBy = void 0;
/**
 * Group by configuration for retention queries.
 */
class RetentionGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RetentionGroupBy.attributeTypeMap;
    }
}
exports.RetentionGroupBy = RetentionGroupBy;
/**
 * @ignore
 */
RetentionGroupBy.attributeTypeMap = {
    facet: {
        baseName: "facet",
        type: "string",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    shouldExcludeMissing: {
        baseName: "should_exclude_missing",
        type: "boolean",
    },
    sort: {
        baseName: "sort",
        type: "RetentionGroupBySort",
    },
    source: {
        baseName: "source",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "RetentionGroupByTarget",
        required: true,
    },
};
