// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SankeyWidgetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SankeyWidgetDefinition = void 0;
/**
 * The Sankey diagram visualizes the flow of data between categories, stages or sets of values.
 */
class SankeyWidgetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SankeyWidgetDefinition.attributeTypeMap;
    }
}
exports.SankeyWidgetDefinition = SankeyWidgetDefinition;
/**
 * @ignore
 */
SankeyWidgetDefinition.attributeTypeMap = {
    requests: {
        baseName: "requests",
        type: "Array<SankeyWidgetRequest>",
        required: true,
    },
    showOtherLinks: {
        baseName: "show_other_links",
        type: "boolean",
    },
    sortNodes: {
        baseName: "sort_nodes",
        type: "boolean",
    },
    time: {
        baseName: "time",
        type: "WidgetTime",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    titleAlign: {
        baseName: "title_align",
        type: "WidgetTextAlign",
    },
    titleSize: {
        baseName: "title_size",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "SankeyWidgetDefinitionType",
        required: true,
    },
};
