// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SankeyRumQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SankeyRumQuery = void 0;
/**
 * Sankey widget with RUM data source query.
 */
class SankeyRumQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SankeyRumQuery.attributeTypeMap;
    }
}
exports.SankeyRumQuery = SankeyRumQuery;
/**
 * @ignore
 */
SankeyRumQuery.attributeTypeMap = {
    audienceFilters: {
        baseName: "audience_filters",
        type: "ProductAnalyticsAudienceFilters",
    },
    dataSource: {
        baseName: "data_source",
        type: "SankeyRumDataSource",
        required: true,
    },
    entriesPerStep: {
        baseName: "entries_per_step",
        type: "number",
        format: "int64",
    },
    joinKeys: {
        baseName: "join_keys",
        type: "SankeyJoinKeys",
    },
    mode: {
        baseName: "mode",
        type: "SankeyRumQueryMode",
        required: true,
    },
    numberOfSteps: {
        baseName: "number_of_steps",
        type: "number",
        format: "int64",
    },
    occurrences: {
        baseName: "occurrences",
        type: "ProductAnalyticsAudienceOccurrenceFilter",
    },
    queryString: {
        baseName: "query_string",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "string",
    },
    subqueryId: {
        baseName: "subquery_id",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "string",
    },
};
