// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SankeyNetworkQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SankeyNetworkQuery = void 0;
/**
 * Query configuration for Sankey network widget.
 */
class SankeyNetworkQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SankeyNetworkQuery.attributeTypeMap;
    }
}
exports.SankeyNetworkQuery = SankeyNetworkQuery;
/**
 * @ignore
 */
SankeyNetworkQuery.attributeTypeMap = {
    compute: {
        baseName: "compute",
        type: "SankeyNetworkQueryCompute",
    },
    dataSource: {
        baseName: "data_source",
        type: "SankeyNetworkDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<string>",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        required: true,
        format: "int64",
    },
    mode: {
        baseName: "mode",
        type: "SankeyNetworkQueryMode",
    },
    queryString: {
        baseName: "query_string",
        type: "string",
        required: true,
    },
    shouldExcludeMissing: {
        baseName: "should_exclude_missing",
        type: "boolean",
    },
    sort: {
        baseName: "sort",
        type: "SankeyNetworkQuerySort",
    },
};
