// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SLOCountDefinitionWithTotalEventsFormula.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SLOCountDefinitionWithTotalEventsFormula = void 0;
/**
 * SLO count definition using a total events formula alongside a good events formula.
 */
class SLOCountDefinitionWithTotalEventsFormula {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SLOCountDefinitionWithTotalEventsFormula.attributeTypeMap;
    }
}
exports.SLOCountDefinitionWithTotalEventsFormula = SLOCountDefinitionWithTotalEventsFormula;
/**
 * @ignore
 */
SLOCountDefinitionWithTotalEventsFormula.attributeTypeMap = {
    goodEventsFormula: {
        baseName: "good_events_formula",
        type: "SLOFormula",
        required: true,
    },
    queries: {
        baseName: "queries",
        type: "Array<SLODataSourceQueryDefinition>",
        required: true,
    },
    totalEventsFormula: {
        baseName: "total_events_formula",
        type: "SLOFormula",
        required: true,
    },
};
