// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsMobileStepType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WAIT = exports.TYPETEXT = exports.TOGGLEWIFI = exports.TAP = exports.SCROLLTOELEMENT = exports.SCROLL = exports.ROTATE = exports.RESTARTAPPLICATION = exports.PRESSBACK = exports.PLAYSUBTEST = exports.OPENDEEPLINK = exports.FLICK = exports.EXTRACTVARIABLE = exports.DOUBLETAP = exports.ASSERTSCREENLACKS = exports.ASSERTSCREENCONTAINS = exports.ASSERTELEMENTCONTENT = void 0;
exports.ASSERTELEMENTCONTENT = "assertElementContent";
exports.ASSERTSCREENCONTAINS = "assertScreenContains";
exports.ASSERTSCREENLACKS = "assertScreenLacks";
exports.DOUBLETAP = "doubleTap";
exports.EXTRACTVARIABLE = "extractVariable";
exports.FLICK = "flick";
exports.OPENDEEPLINK = "openDeeplink";
exports.PLAYSUBTEST = "playSubTest";
exports.PRESSBACK = "pressBack";
exports.RESTARTAPPLICATION = "restartApplication";
exports.ROTATE = "rotate";
exports.SCROLL = "scroll";
exports.SCROLLTOELEMENT = "scrollToElement";
exports.TAP = "tap";
exports.TOGGLEWIFI = "toggleWiFi";
exports.TYPETEXT = "typeText";
exports.WAIT = "wait";
