// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsAssertionType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JITTER = exports.MULTI_NETWORK_HOP = exports.CONNECTION = exports.GRPC_PROTO = exports.GRPC_METADATA = exports.GRPC_HEALTHCHECK_STATUS = exports.RECEIVED_MESSAGE = exports.NETWORK_HOP = exports.PACKETS_RECEIVED = exports.PACKET_LOSS_PERCENTAGE = exports.LATENCY = exports.MIN_TLS_VERSION = exports.TLS_VERSION = exports.RECORD_SOME = exports.RECORD_EVERY = exports.PROPERTY = exports.RESPONSE_TIME = exports.CERTIFICATE = exports.STATUS_CODE = exports.HEADER = exports.BODY = void 0;
exports.BODY = "body";
exports.HEADER = "header";
exports.STATUS_CODE = "statusCode";
exports.CERTIFICATE = "certificate";
exports.RESPONSE_TIME = "responseTime";
exports.PROPERTY = "property";
exports.RECORD_EVERY = "recordEvery";
exports.RECORD_SOME = "recordSome";
exports.TLS_VERSION = "tlsVersion";
exports.MIN_TLS_VERSION = "minTlsVersion";
exports.LATENCY = "latency";
exports.PACKET_LOSS_PERCENTAGE = "packetLossPercentage";
exports.PACKETS_RECEIVED = "packetsReceived";
exports.NETWORK_HOP = "networkHop";
exports.RECEIVED_MESSAGE = "receivedMessage";
exports.GRPC_HEALTHCHECK_STATUS = "grpcHealthcheckStatus";
exports.GRPC_METADATA = "grpcMetadata";
exports.GRPC_PROTO = "grpcProto";
exports.CONNECTION = "connection";
exports.MULTI_NETWORK_HOP = "multiNetworkHop";
exports.JITTER = "jitter";
