// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsAssertionJSONSchemaTarget.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsAssertionJSONSchemaTarget = void 0;
/**
 * An assertion for the `validatesJSONSchema` operator.
 */
class SyntheticsAssertionJSONSchemaTarget {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsAssertionJSONSchemaTarget.attributeTypeMap;
    }
}
exports.SyntheticsAssertionJSONSchemaTarget = SyntheticsAssertionJSONSchemaTarget;
/**
 * @ignore
 */
SyntheticsAssertionJSONSchemaTarget.attributeTypeMap = {
    operator: {
        baseName: "operator",
        type: "SyntheticsAssertionJSONSchemaOperator",
        required: true,
    },
    target: {
        baseName: "target",
        type: "SyntheticsAssertionJSONSchemaTargetTarget",
    },
    type: {
        baseName: "type",
        type: "SyntheticsAssertionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
