// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsApiTestFailureCode.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.INTERNAL_ERROR = exports.UNKNOWN = exports.WEBSOCKET = exports.TUNNEL = exports.INVALID_TEST = exports.OCSP = exports.SSL = exports.TIMEOUT = exports.NETUNREACH = exports.HOSTUNREACH = exports.DNS = exports.CONNRESET = exports.CONNREFUSED = exports.INCORRECT_ASSERTION = exports.MALFORMED_RESPONSE = exports.UNESCAPED_CHARACTERS_IN_REQUEST_PATH = exports.REQUIRES_UPDATE = exports.INVALID_REQUEST = exports.HEADERS_INCOMPATIBLE_CONTENT_LENGTH = exports.HEADER_TOO_LARGE = exports.INVALID_CHAR_IN_HEADER = exports.DECRYPTION = exports.AUTHENTICATION_ERROR = exports.TOO_MANY_REDIRECTS = exports.DENIED = exports.BODY_TOO_LARGE = void 0;
exports.BODY_TOO_LARGE = "BODY_TOO_LARGE";
exports.DENIED = "DENIED";
exports.TOO_MANY_REDIRECTS = "TOO_MANY_REDIRECTS";
exports.AUTHENTICATION_ERROR = "AUTHENTICATION_ERROR";
exports.DECRYPTION = "DECRYPTION";
exports.INVALID_CHAR_IN_HEADER = "INVALID_CHAR_IN_HEADER";
exports.HEADER_TOO_LARGE = "HEADER_TOO_LARGE";
exports.HEADERS_INCOMPATIBLE_CONTENT_LENGTH = "HEADERS_INCOMPATIBLE_CONTENT_LENGTH";
exports.INVALID_REQUEST = "INVALID_REQUEST";
exports.REQUIRES_UPDATE = "REQUIRES_UPDATE";
exports.UNESCAPED_CHARACTERS_IN_REQUEST_PATH = "UNESCAPED_CHARACTERS_IN_REQUEST_PATH";
exports.MALFORMED_RESPONSE = "MALFORMED_RESPONSE";
exports.INCORRECT_ASSERTION = "INCORRECT_ASSERTION";
exports.CONNREFUSED = "CONNREFUSED";
exports.CONNRESET = "CONNRESET";
exports.DNS = "DNS";
exports.HOSTUNREACH = "HOSTUNREACH";
exports.NETUNREACH = "NETUNREACH";
exports.TIMEOUT = "TIMEOUT";
exports.SSL = "SSL";
exports.OCSP = "OCSP";
exports.INVALID_TEST = "INVALID_TEST";
exports.TUNNEL = "TUNNEL";
exports.WEBSOCKET = "WEBSOCKET";
exports.UNKNOWN = "UNKNOWN";
exports.INTERNAL_ERROR = "INTERNAL_ERROR";
