// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsBrowserTestFailureCode.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UPLOAD_FILES_NAME = exports.UPLOAD_FILES_DYNAMIC_ELEMENT = exports.UPLOAD_FILES_DIALOG = exports.UPLOAD_FILES_ELEMENT_TYPE = exports.UNSUPPORTED_AUTH_SCHEMA = exports.UNKNOWN = exports.UNAVAILABLE_BROWSER = exports.TOO_MANY_HTTP_REQUESTS = exports.TEST_TIMEOUT = exports.SUB_TEST_NOT_PASSED = exports.STEP_TIMEOUT = exports.SELECT_OPTION = exports.SERVER_CERTIFICATE = exports.PRESS_KEY = exports.OPEN_URL = exports.NAVIGATE_TO_LINK = exports.LOCATE_ELEMENT = exports.INVISIBLE_ELEMENT = exports.INVALID_VARIABLE_PATTERN = exports.INVALID_URL = exports.INVALID_TYPE_TEXT_DELAY = exports.INTERNAL_ERROR = exports.INCONSISTENCIES = exports.FRAME_DETACHED = exports.FORBIDDEN_URL = exports.EXTRACT_VARIABLE = exports.EVALUATE_JAVASCRIPT_CONTEXT = exports.EVALUATE_JAVASCRIPT = exports.EMAIL_VARIABLE_NOT_DEFINED = exports.ELEMENT_NOT_INTERACTABLE = exports.DOWNLOAD_FILE_TOO_LARGE = exports.ASSERTION_FAILURE = exports.API_REQUEST_FAILURE = void 0;
exports.API_REQUEST_FAILURE = "API_REQUEST_FAILURE";
exports.ASSERTION_FAILURE = "ASSERTION_FAILURE";
exports.DOWNLOAD_FILE_TOO_LARGE = "DOWNLOAD_FILE_TOO_LARGE";
exports.ELEMENT_NOT_INTERACTABLE = "ELEMENT_NOT_INTERACTABLE";
exports.EMAIL_VARIABLE_NOT_DEFINED = "EMAIL_VARIABLE_NOT_DEFINED";
exports.EVALUATE_JAVASCRIPT = "EVALUATE_JAVASCRIPT";
exports.EVALUATE_JAVASCRIPT_CONTEXT = "EVALUATE_JAVASCRIPT_CONTEXT";
exports.EXTRACT_VARIABLE = "EXTRACT_VARIABLE";
exports.FORBIDDEN_URL = "FORBIDDEN_URL";
exports.FRAME_DETACHED = "FRAME_DETACHED";
exports.INCONSISTENCIES = "INCONSISTENCIES";
exports.INTERNAL_ERROR = "INTERNAL_ERROR";
exports.INVALID_TYPE_TEXT_DELAY = "INVALID_TYPE_TEXT_DELAY";
exports.INVALID_URL = "INVALID_URL";
exports.INVALID_VARIABLE_PATTERN = "INVALID_VARIABLE_PATTERN";
exports.INVISIBLE_ELEMENT = "INVISIBLE_ELEMENT";
exports.LOCATE_ELEMENT = "LOCATE_ELEMENT";
exports.NAVIGATE_TO_LINK = "NAVIGATE_TO_LINK";
exports.OPEN_URL = "OPEN_URL";
exports.PRESS_KEY = "PRESS_KEY";
exports.SERVER_CERTIFICATE = "SERVER_CERTIFICATE";
exports.SELECT_OPTION = "SELECT_OPTION";
exports.STEP_TIMEOUT = "STEP_TIMEOUT";
exports.SUB_TEST_NOT_PASSED = "SUB_TEST_NOT_PASSED";
exports.TEST_TIMEOUT = "TEST_TIMEOUT";
exports.TOO_MANY_HTTP_REQUESTS = "TOO_MANY_HTTP_REQUESTS";
exports.UNAVAILABLE_BROWSER = "UNAVAILABLE_BROWSER";
exports.UNKNOWN = "UNKNOWN";
exports.UNSUPPORTED_AUTH_SCHEMA = "UNSUPPORTED_AUTH_SCHEMA";
exports.UPLOAD_FILES_ELEMENT_TYPE = "UPLOAD_FILES_ELEMENT_TYPE";
exports.UPLOAD_FILES_DIALOG = "UPLOAD_FILES_DIALOG";
exports.UPLOAD_FILES_DYNAMIC_ELEMENT = "UPLOAD_FILES_DYNAMIC_ELEMENT";
exports.UPLOAD_FILES_NAME = "UPLOAD_FILES_NAME";
