// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/TableWidgetTextFormatPalette.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CUSTOM_TEXT = exports.CUSTOM_BG = exports.GREEN_ON_WHITE = exports.YELLOW_ON_WHITE = exports.RED_ON_WHITE = exports.BLACK_ON_LIGHT_GREEN = exports.BLACK_ON_LIGHT_YELLOW = exports.BLACK_ON_LIGHT_RED = exports.WHITE_ON_GREEN = exports.WHITE_ON_YELLOW = exports.WHITE_ON_RED = void 0;
exports.WHITE_ON_RED = "white_on_red";
exports.WHITE_ON_YELLOW = "white_on_yellow";
exports.WHITE_ON_GREEN = "white_on_green";
exports.BLACK_ON_LIGHT_RED = "black_on_light_red";
exports.BLACK_ON_LIGHT_YELLOW = "black_on_light_yellow";
exports.BLACK_ON_LIGHT_GREEN = "black_on_light_green";
exports.RED_ON_WHITE = "red_on_white";
exports.YELLOW_ON_WHITE = "yellow_on_white";
exports.GREEN_ON_WHITE = "green_on_white";
exports.CUSTOM_BG = "custom_bg";
exports.CUSTOM_TEXT = "custom_text";
