// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/TimeseriesRequestStyle.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeseriesRequestStyle = void 0;
/**
 * Define request widget style for timeseries widgets.
 */
class TimeseriesRequestStyle {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TimeseriesRequestStyle.attributeTypeMap;
    }
}
exports.TimeseriesRequestStyle = TimeseriesRequestStyle;
/**
 * @ignore
 */
TimeseriesRequestStyle.attributeTypeMap = {
    hasValueLabels: {
        baseName: "has_value_labels",
        type: "boolean",
    },
    lineType: {
        baseName: "line_type",
        type: "WidgetLineType",
    },
    lineWidth: {
        baseName: "line_width",
        type: "WidgetLineWidth",
    },
    orderBy: {
        baseName: "order_by",
        type: "WidgetStyleOrderBy",
    },
    palette: {
        baseName: "palette",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
