// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/UsageIoTHour.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageIoTHour = void 0;
/**
 * IoT usage for a given organization for a given hour.
 */
class UsageIoTHour {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UsageIoTHour.attributeTypeMap;
    }
}
exports.UsageIoTHour = UsageIoTHour;
/**
 * @ignore
 */
UsageIoTHour.attributeTypeMap = {
    hour: {
        baseName: "hour",
        type: "Date",
        format: "date-time",
    },
    iotDeviceCount: {
        baseName: "iot_device_count",
        type: "number",
        format: "int64",
    },
    orgName: {
        baseName: "org_name",
        type: "string",
    },
    publicId: {
        baseName: "public_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
