// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/UserJourneyFormulaGroupBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserJourneyFormulaGroupBy = void 0;
/**
 * Group by configuration for User Journey formula queries.
 */
class UserJourneyFormulaGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UserJourneyFormulaGroupBy.attributeTypeMap;
    }
}
exports.UserJourneyFormulaGroupBy = UserJourneyFormulaGroupBy;
/**
 * @ignore
 */
UserJourneyFormulaGroupBy.attributeTypeMap = {
    facet: {
        baseName: "facet",
        type: "string",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int32",
    },
    shouldExcludeMissing: {
        baseName: "should_exclude_missing",
        type: "boolean",
    },
    sort: {
        baseName: "sort",
        type: "FormulaAndFunctionEventQueryGroupBySort",
    },
    target: {
        baseName: "target",
        type: "UserJourneySearchTarget",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
