// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/WidgetPalette.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BLACK_ON_LIGHT_RED = exports.BLACK_ON_LIGHT_GREEN = exports.BLACK_ON_LIGHT_YELLOW = exports.YELLOW_ON_WHITE = exports.WHITE_ON_YELLOW = exports.WHITE_ON_RED = exports.GREEN_ON_WHITE = exports.WHITE_ON_GREEN = exports.WHITE_ON_GRAY = exports.RED_ON_WHITE = exports.RED = exports.ORANGE = exports.GREEN = exports.GREY = exports.GRAY_ON_WHITE = exports.CUSTOM_TEXT = exports.CUSTOM_IMAGE = exports.CUSTOM_BACKGROUND = exports.BLUE = void 0;
exports.BLUE = "blue";
exports.CUSTOM_BACKGROUND = "custom_bg";
exports.CUSTOM_IMAGE = "custom_image";
exports.CUSTOM_TEXT = "custom_text";
exports.GRAY_ON_WHITE = "gray_on_white";
exports.GREY = "grey";
exports.GREEN = "green";
exports.ORANGE = "orange";
exports.RED = "red";
exports.RED_ON_WHITE = "red_on_white";
exports.WHITE_ON_GRAY = "white_on_gray";
exports.WHITE_ON_GREEN = "white_on_green";
exports.GREEN_ON_WHITE = "green_on_white";
exports.WHITE_ON_RED = "white_on_red";
exports.WHITE_ON_YELLOW = "white_on_yellow";
exports.YELLOW_ON_WHITE = "yellow_on_white";
exports.BLACK_ON_LIGHT_YELLOW = "black_on_light_yellow";
exports.BLACK_ON_LIGHT_GREEN = "black_on_light_green";
exports.BLACK_ON_LIGHT_RED = "black_on_light_red";
