// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/WildcardWidgetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WildcardWidgetDefinition = void 0;
/**
 * Custom visualization widget using Vega or Vega-Lite specifications. Combines standard Datadog data requests with a Vega or Vega-Lite JSON specification for flexible, custom visualizations.
 */
class WildcardWidgetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WildcardWidgetDefinition.attributeTypeMap;
    }
}
exports.WildcardWidgetDefinition = WildcardWidgetDefinition;
/**
 * @ignore
 */
WildcardWidgetDefinition.attributeTypeMap = {
    customLinks: {
        baseName: "custom_links",
        type: "Array<WidgetCustomLink>",
    },
    requests: {
        baseName: "requests",
        type: "Array<WildcardWidgetRequest>",
        required: true,
    },
    specification: {
        baseName: "specification",
        type: "WildcardWidgetSpecification",
        required: true,
    },
    time: {
        baseName: "time",
        type: "WidgetTime",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    titleAlign: {
        baseName: "title_align",
        type: "WidgetTextAlign",
    },
    titleSize: {
        baseName: "title_size",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "WildcardWidgetDefinitionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
