// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AllocationExposureSchedule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllocationExposureSchedule = void 0;
/**
 * Progressive release details for a targeting rule allocation.
 */
class AllocationExposureSchedule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AllocationExposureSchedule.attributeTypeMap;
    }
}
exports.AllocationExposureSchedule = AllocationExposureSchedule;
/**
 * @ignore
 */
AllocationExposureSchedule.attributeTypeMap = {
    absoluteStartTime: {
        baseName: "absolute_start_time",
        type: "Date",
        format: "date-time",
    },
    allocationId: {
        baseName: "allocation_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    controlVariantId: {
        baseName: "control_variant_id",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    guardrailTriggeredAction: {
        baseName: "guardrail_triggered_action",
        type: "string",
    },
    guardrailTriggers: {
        baseName: "guardrail_triggers",
        type: "Array<AllocationExposureGuardrailTrigger>",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    rolloutOptions: {
        baseName: "rollout_options",
        type: "RolloutOptions",
        required: true,
    },
    rolloutSteps: {
        baseName: "rollout_steps",
        type: "Array<AllocationExposureRolloutStep>",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
