// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/Allocation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Allocation = void 0;
/**
 * Targeting rule (allocation) details for a feature flag environment.
 */
class Allocation {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return Allocation.attributeTypeMap;
    }
}
exports.Allocation = Allocation;
/**
 * @ignore
 */
Allocation.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    environmentIds: {
        baseName: "environment_ids",
        type: "Array<string>",
        required: true,
    },
    experimentId: {
        baseName: "experiment_id",
        type: "string",
    },
    exposureSchedule: {
        baseName: "exposure_schedule",
        type: "AllocationExposureSchedule",
    },
    guardrailMetrics: {
        baseName: "guardrail_metrics",
        type: "Array<GuardrailMetric>",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    orderPosition: {
        baseName: "order_position",
        type: "number",
        required: true,
        format: "int64",
    },
    targetingRules: {
        baseName: "targeting_rules",
        type: "Array<TargetingRule>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AllocationType",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    variantWeights: {
        baseName: "variant_weights",
        type: "Array<VariantWeight>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
