// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AllocationExposureGuardrailTrigger.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllocationExposureGuardrailTrigger = void 0;
/**
 * Guardrail trigger details for a progressive rollout.
 */
class AllocationExposureGuardrailTrigger {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AllocationExposureGuardrailTrigger.attributeTypeMap;
    }
}
exports.AllocationExposureGuardrailTrigger = AllocationExposureGuardrailTrigger;
/**
 * @ignore
 */
AllocationExposureGuardrailTrigger.attributeTypeMap = {
    allocationExposureScheduleId: {
        baseName: "allocation_exposure_schedule_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    flaggingVariantId: {
        baseName: "flagging_variant_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    metricId: {
        baseName: "metric_id",
        type: "string",
        required: true,
    },
    triggeredAction: {
        baseName: "triggered_action",
        type: "string",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
