// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AllocationExposureScheduleData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllocationExposureScheduleData = void 0;
/**
 * Data wrapper for progressive rollout schedule responses.
 */
class AllocationExposureScheduleData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AllocationExposureScheduleData.attributeTypeMap;
    }
}
exports.AllocationExposureScheduleData = AllocationExposureScheduleData;
/**
 * @ignore
 */
AllocationExposureScheduleData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AllocationExposureSchedule",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "AllocationExposureScheduleDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
