// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AllocationExposureRolloutStep.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllocationExposureRolloutStep = void 0;
/**
 * Exposure progression step details.
 */
class AllocationExposureRolloutStep {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AllocationExposureRolloutStep.attributeTypeMap;
    }
}
exports.AllocationExposureRolloutStep = AllocationExposureRolloutStep;
/**
 * @ignore
 */
AllocationExposureRolloutStep.attributeTypeMap = {
    allocationExposureScheduleId: {
        baseName: "allocation_exposure_schedule_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    exposureRatio: {
        baseName: "exposure_ratio",
        type: "number",
        required: true,
        format: "double",
    },
    groupedStepIndex: {
        baseName: "grouped_step_index",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    intervalMs: {
        baseName: "interval_ms",
        type: "number",
        format: "int64",
    },
    isPauseRecord: {
        baseName: "is_pause_record",
        type: "boolean",
        required: true,
    },
    orderPosition: {
        baseName: "order_position",
        type: "number",
        required: true,
        format: "int64",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
