// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApmDependencyStatsQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApmDependencyStatsQuery = void 0;
/**
 * A query for APM dependency statistics between services, such as call latency and error rates.
 */
class ApmDependencyStatsQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApmDependencyStatsQuery.attributeTypeMap;
    }
}
exports.ApmDependencyStatsQuery = ApmDependencyStatsQuery;
/**
 * @ignore
 */
ApmDependencyStatsQuery.attributeTypeMap = {
    dataSource: {
        baseName: "data_source",
        type: "ApmDependencyStatsDataSource",
        required: true,
    },
    env: {
        baseName: "env",
        type: "string",
        required: true,
    },
    isUpstream: {
        baseName: "is_upstream",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    operationName: {
        baseName: "operation_name",
        type: "string",
        required: true,
    },
    primaryTagName: {
        baseName: "primary_tag_name",
        type: "string",
    },
    primaryTagValue: {
        baseName: "primary_tag_value",
        type: "string",
    },
    resourceName: {
        baseName: "resource_name",
        type: "string",
        required: true,
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    stat: {
        baseName: "stat",
        type: "ApmDependencyStatName",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
