// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApmMetricsStat.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TOTAL_TIME = exports.LATENCY_DISTRIBUTION = exports.LATENCY_P999 = exports.LATENCY_P99 = exports.LATENCY_P95 = exports.LATENCY_P90 = exports.LATENCY_P75 = exports.LATENCY_P50 = exports.LATENCY_MAX = exports.LATENCY_AVG = exports.APDEX = exports.HITS_PER_SECOND = exports.HITS = exports.ERRORS_PER_SECOND = exports.ERRORS = exports.ERROR_RATE = void 0;
exports.ERROR_RATE = "error_rate";
exports.ERRORS = "errors";
exports.ERRORS_PER_SECOND = "errors_per_second";
exports.HITS = "hits";
exports.HITS_PER_SECOND = "hits_per_second";
exports.APDEX = "apdex";
exports.LATENCY_AVG = "latency_avg";
exports.LATENCY_MAX = "latency_max";
exports.LATENCY_P50 = "latency_p50";
exports.LATENCY_P75 = "latency_p75";
exports.LATENCY_P90 = "latency_p90";
exports.LATENCY_P95 = "latency_p95";
exports.LATENCY_P99 = "latency_p99";
exports.LATENCY_P999 = "latency_p999";
exports.LATENCY_DISTRIBUTION = "latency_distribution";
exports.TOTAL_TIME = "total_time";
