// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApmMetricsQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApmMetricsQuery = void 0;
/**
 * A query for APM trace metrics such as hits, errors, and latency percentiles, aggregated across services.
 */
class ApmMetricsQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApmMetricsQuery.attributeTypeMap;
    }
}
exports.ApmMetricsQuery = ApmMetricsQuery;
/**
 * @ignore
 */
ApmMetricsQuery.attributeTypeMap = {
    dataSource: {
        baseName: "data_source",
        type: "ApmMetricsDataSource",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    operationMode: {
        baseName: "operation_mode",
        type: "string",
    },
    operationName: {
        baseName: "operation_name",
        type: "string",
    },
    peerTags: {
        baseName: "peer_tags",
        type: "Array<string>",
    },
    queryFilter: {
        baseName: "query_filter",
        type: "string",
    },
    resourceHash: {
        baseName: "resource_hash",
        type: "string",
    },
    resourceName: {
        baseName: "resource_name",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
    },
    spanKind: {
        baseName: "span_kind",
        type: "ApmMetricsSpanKind",
    },
    stat: {
        baseName: "stat",
        type: "ApmMetricsStat",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
