// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApmResourceStatsQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApmResourceStatsQuery = void 0;
/**
 * A query for APM resource statistics such as latency, error rate, and hit count, grouped by resource name.
 */
class ApmResourceStatsQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApmResourceStatsQuery.attributeTypeMap;
    }
}
exports.ApmResourceStatsQuery = ApmResourceStatsQuery;
/**
 * @ignore
 */
ApmResourceStatsQuery.attributeTypeMap = {
    dataSource: {
        baseName: "data_source",
        type: "ApmResourceStatsDataSource",
        required: true,
    },
    env: {
        baseName: "env",
        type: "string",
        required: true,
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    operationName: {
        baseName: "operation_name",
        type: "string",
    },
    primaryTagName: {
        baseName: "primary_tag_name",
        type: "string",
    },
    primaryTagValue: {
        baseName: "primary_tag_value",
        type: "string",
    },
    resourceName: {
        baseName: "resource_name",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    stat: {
        baseName: "stat",
        type: "ApmResourceStatName",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
