// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityWafCustomRuleConditionOperator.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NOT_EQUALS = exports.EQUALS = exports.NOT_EXISTS = exports.EXISTS = exports.CAPTURE_DATA = exports.NOT_IP_MATCH = exports.IP_MATCH = exports.NOT_EXACT_MATCH = exports.EXACT_MATCH = exports.IS_SQLI = exports.IS_XSS = exports.NOT_PHRASE_MATCH = exports.PHRASE_MATCH = exports.NOT_MATCH_REGEX = exports.MATCH_REGEX = void 0;
exports.MATCH_REGEX = "match_regex";
exports.NOT_MATCH_REGEX = "!match_regex";
exports.PHRASE_MATCH = "phrase_match";
exports.NOT_PHRASE_MATCH = "!phrase_match";
exports.IS_XSS = "is_xss";
exports.IS_SQLI = "is_sqli";
exports.EXACT_MATCH = "exact_match";
exports.NOT_EXACT_MATCH = "!exact_match";
exports.IP_MATCH = "ip_match";
exports.NOT_IP_MATCH = "!ip_match";
exports.CAPTURE_DATA = "capture_data";
exports.EXISTS = "exists";
exports.NOT_EXISTS = "!exists";
exports.EQUALS = "equals";
exports.NOT_EQUALS = "!equals";
