// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AuthNMappingsSort.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAML_ASSERTION_ATTRIBUTE_VALUE_DESCENDING = exports.SAML_ASSERTION_ATTRIBUTE_VALUE_ASCENDING = exports.SAML_ASSERTION_ATTRIBUTE_KEY_DESCENDING = exports.SAML_ASSERTION_ATTRIBUTE_KEY_ASCENDING = exports.ROLE_NAME_DESCENDING = exports.ROLE_NAME_ASCENDING = exports.SAML_ASSERTION_ATTRIBUTE_ID_DESCENDING = exports.SAML_ASSERTION_ATTRIBUTE_ID_ASCENDING = exports.ROLE_ID_DESCENDING = exports.ROLE_ID_ASCENDING = exports.CREATED_AT_DESCENDING = exports.CREATED_AT_ASCENDING = void 0;
exports.CREATED_AT_ASCENDING = "created_at";
exports.CREATED_AT_DESCENDING = "-created_at";
exports.ROLE_ID_ASCENDING = "role_id";
exports.ROLE_ID_DESCENDING = "-role_id";
exports.SAML_ASSERTION_ATTRIBUTE_ID_ASCENDING = "saml_assertion_attribute_id";
exports.SAML_ASSERTION_ATTRIBUTE_ID_DESCENDING = "-saml_assertion_attribute_id";
exports.ROLE_NAME_ASCENDING = "role.name";
exports.ROLE_NAME_DESCENDING = "-role.name";
exports.SAML_ASSERTION_ATTRIBUTE_KEY_ASCENDING = "saml_assertion_attribute.attribute_key";
exports.SAML_ASSERTION_ATTRIBUTE_KEY_DESCENDING = "-saml_assertion_attribute.attribute_key";
exports.SAML_ASSERTION_ATTRIBUTE_VALUE_ASCENDING = "saml_assertion_attribute.attribute_value";
exports.SAML_ASSERTION_ATTRIBUTE_VALUE_DESCENDING = "-saml_assertion_attribute.attribute_value";
