// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AzureStorageDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureStorageDestination = void 0;
/**
 * The `azure_storage` destination forwards logs to an Azure Blob Storage container.
 *
 * **Supported pipeline types:** logs
 */
class AzureStorageDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AzureStorageDestination.attributeTypeMap;
    }
}
exports.AzureStorageDestination = AzureStorageDestination;
/**
 * @ignore
 */
AzureStorageDestination.attributeTypeMap = {
    blobPrefix: {
        baseName: "blob_prefix",
        type: "string",
    },
    buffer: {
        baseName: "buffer",
        type: "ObservabilityPipelineBufferOptions",
    },
    connectionStringKey: {
        baseName: "connection_string_key",
        type: "string",
    },
    containerName: {
        baseName: "container_name",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AzureStorageDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
