// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BatchRowsQueryRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchRowsQueryRequestDataAttributes = void 0;
/**
 * Attributes for a batch rows query request.
 */
class BatchRowsQueryRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BatchRowsQueryRequestDataAttributes.attributeTypeMap;
    }
}
exports.BatchRowsQueryRequestDataAttributes = BatchRowsQueryRequestDataAttributes;
/**
 * @ignore
 */
BatchRowsQueryRequestDataAttributes.attributeTypeMap = {
    rowIds: {
        baseName: "row_ids",
        type: "Array<string>",
        required: true,
    },
    tableId: {
        baseName: "table_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
