// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BudgetWithEntriesDataAttributesEntriesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetWithEntriesDataAttributesEntriesItems = void 0;
/**
 * A single monthly budget entry defining the allocated amount and optional tag filters for a specific month.
 */
class BudgetWithEntriesDataAttributesEntriesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BudgetWithEntriesDataAttributesEntriesItems.attributeTypeMap;
    }
}
exports.BudgetWithEntriesDataAttributesEntriesItems = BudgetWithEntriesDataAttributesEntriesItems;
/**
 * @ignore
 */
BudgetWithEntriesDataAttributesEntriesItems.attributeTypeMap = {
    amount: {
        baseName: "amount",
        type: "number",
        format: "double",
    },
    month: {
        baseName: "month",
        type: "number",
        format: "int64",
    },
    tagFilters: {
        baseName: "tag_filters",
        type: "Array<BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
