// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems = void 0;
/**
 * A tag filter used to scope a budget entry to specific resource tags.
 */
class BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems.attributeTypeMap;
    }
}
exports.BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems = BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems;
/**
 * @ignore
 */
BudgetWithEntriesDataAttributesEntriesItemsTagFiltersItems.attributeTypeMap = {
    tagKey: {
        baseName: "tag_key",
        type: "string",
    },
    tagValue: {
        baseName: "tag_value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
