// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BudgetValidationResponseDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetValidationResponseDataAttributes = void 0;
/**
 * The attributes of a budget validation response, including any validation errors and the validity status.
 */
class BudgetValidationResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BudgetValidationResponseDataAttributes.attributeTypeMap;
    }
}
exports.BudgetValidationResponseDataAttributes = BudgetValidationResponseDataAttributes;
/**
 * @ignore
 */
BudgetValidationResponseDataAttributes.attributeTypeMap = {
    errors: {
        baseName: "errors",
        type: "Array<string>",
    },
    valid: {
        baseName: "valid",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
