// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CampaignResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CampaignResponseAttributes = void 0;
/**
 * Campaign attributes.
 */
class CampaignResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CampaignResponseAttributes.attributeTypeMap;
    }
}
exports.CampaignResponseAttributes = CampaignResponseAttributes;
/**
 * @ignore
 */
CampaignResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    dueDate: {
        baseName: "due_date",
        type: "Date",
        format: "date-time",
    },
    entityScope: {
        baseName: "entity_scope",
        type: "string",
    },
    guidance: {
        baseName: "guidance",
        type: "string",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    owner: {
        baseName: "owner",
        type: "string",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
