// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CompletionConditionOperator.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OPERATOR_IS_NOT_EMPTY = exports.OPERATOR_IS_EMPTY = exports.OPERATOR_IS_NOT_NULL = exports.OPERATOR_IS_NULL = exports.OPERATOR_DOES_NOT_CONTAIN = exports.OPERATOR_CONTAINS = exports.OPERATOR_LESS_THAN_OR_EQUAL_TO = exports.OPERATOR_GREATER_THAN_OR_EQUAL_TO = exports.OPERATOR_LESS_THAN = exports.OPERATOR_GREATER_THAN = exports.OPERATOR_NOT_EQUAL = exports.OPERATOR_EQUAL = void 0;
exports.OPERATOR_EQUAL = "OPERATOR_EQUAL";
exports.OPERATOR_NOT_EQUAL = "OPERATOR_NOT_EQUAL";
exports.OPERATOR_GREATER_THAN = "OPERATOR_GREATER_THAN";
exports.OPERATOR_LESS_THAN = "OPERATOR_LESS_THAN";
exports.OPERATOR_GREATER_THAN_OR_EQUAL_TO = "OPERATOR_GREATER_THAN_OR_EQUAL_TO";
exports.OPERATOR_LESS_THAN_OR_EQUAL_TO = "OPERATOR_LESS_THAN_OR_EQUAL_TO";
exports.OPERATOR_CONTAINS = "OPERATOR_CONTAINS";
exports.OPERATOR_DOES_NOT_CONTAIN = "OPERATOR_DOES_NOT_CONTAIN";
exports.OPERATOR_IS_NULL = "OPERATOR_IS_NULL";
exports.OPERATOR_IS_NOT_NULL = "OPERATOR_IS_NOT_NULL";
exports.OPERATOR_IS_EMPTY = "OPERATOR_IS_EMPTY";
exports.OPERATOR_IS_NOT_EMPTY = "OPERATOR_IS_NOT_EMPTY";
