// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ComponentType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CALLOUTVALUE = exports.CONTAINER = exports.SEARCH = exports.DATERANGEPICKER = exports.GRIDCELL = exports.JSONINPUT = exports.FILEINPUT = exports.NUMBERINPUT = exports.RADIOBUTTONS = exports.VEGACHART = exports.TABS = exports.CHECKBOX = exports.SCHEMAFORM = exports.MODAL = exports.SELECT = exports.TEXT = exports.BUTTON = exports.TEXTAREA = exports.TEXTINPUT = exports.TABLE = void 0;
exports.TABLE = "table";
exports.TEXTINPUT = "textInput";
exports.TEXTAREA = "textArea";
exports.BUTTON = "button";
exports.TEXT = "text";
exports.SELECT = "select";
exports.MODAL = "modal";
exports.SCHEMAFORM = "schemaForm";
exports.CHECKBOX = "checkbox";
exports.TABS = "tabs";
exports.VEGACHART = "vegaChart";
exports.RADIOBUTTONS = "radioButtons";
exports.NUMBERINPUT = "numberInput";
exports.FILEINPUT = "fileInput";
exports.JSONINPUT = "jsonInput";
exports.GRIDCELL = "gridCell";
exports.DATERANGEPICKER = "dateRangePicker";
exports.SEARCH = "search";
exports.CONTAINER = "container";
exports.CALLOUTVALUE = "calloutValue";
