// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/Condition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Condition = void 0;
/**
 * Targeting condition details.
 */
class Condition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return Condition.attributeTypeMap;
    }
}
exports.Condition = Condition;
/**
 * @ignore
 */
Condition.attributeTypeMap = {
    attribute: {
        baseName: "attribute",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    operator: {
        baseName: "operator",
        type: "ConditionOperator",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    value: {
        baseName: "value",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
