// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ContainerTimeseriesQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerTimeseriesQuery = void 0;
/**
 * A query for container-level metrics such as CPU and memory usage.
 */
class ContainerTimeseriesQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ContainerTimeseriesQuery.attributeTypeMap;
    }
}
exports.ContainerTimeseriesQuery = ContainerTimeseriesQuery;
/**
 * @ignore
 */
ContainerTimeseriesQuery.attributeTypeMap = {
    dataSource: {
        baseName: "data_source",
        type: "ContainerDataSource",
        required: true,
    },
    isNormalizedCpu: {
        baseName: "is_normalized_cpu",
        type: "boolean",
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    metric: {
        baseName: "metric",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    sort: {
        baseName: "sort",
        type: "QuerySortOrder",
    },
    tagFilters: {
        baseName: "tag_filters",
        type: "Array<string>",
    },
    textFilter: {
        baseName: "text_filter",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
