// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateFeatureFlagAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFeatureFlagAttributes = void 0;
/**
 * Attributes for creating a new feature flag.
 */
class CreateFeatureFlagAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateFeatureFlagAttributes.attributeTypeMap;
    }
}
exports.CreateFeatureFlagAttributes = CreateFeatureFlagAttributes;
/**
 * @ignore
 */
CreateFeatureFlagAttributes.attributeTypeMap = {
    defaultVariantKey: {
        baseName: "default_variant_key",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    jsonSchema: {
        baseName: "json_schema",
        type: "string",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    valueType: {
        baseName: "value_type",
        type: "ValueType",
        required: true,
    },
    variants: {
        baseName: "variants",
        type: "Array<CreateVariant>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
