// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateEnvironmentAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateEnvironmentAttributes = void 0;
/**
 * Attributes for creating a new environment.
 */
class CreateEnvironmentAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateEnvironmentAttributes.attributeTypeMap;
    }
}
exports.CreateEnvironmentAttributes = CreateEnvironmentAttributes;
/**
 * @ignore
 */
CreateEnvironmentAttributes.attributeTypeMap = {
    isProduction: {
        baseName: "is_production",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    queries: {
        baseName: "queries",
        type: "Array<string>",
        required: true,
    },
    requireFeatureFlagApproval: {
        baseName: "require_feature_flag_approval",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
