// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateTenancyConfigData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTenancyConfigData = void 0;
/**
 * The data object for creating a new OCI tenancy integration configuration, including the tenancy ID, type, and configuration attributes.
 */
class CreateTenancyConfigData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateTenancyConfigData.attributeTypeMap;
    }
}
exports.CreateTenancyConfigData = CreateTenancyConfigData;
/**
 * @ignore
 */
CreateTenancyConfigData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CreateTenancyConfigDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "UpdateTenancyConfigDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
