// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateCampaignRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCampaignRequestAttributes = void 0;
/**
 * Attributes for creating a new campaign.
 */
class CreateCampaignRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateCampaignRequestAttributes.attributeTypeMap;
    }
}
exports.CreateCampaignRequestAttributes = CreateCampaignRequestAttributes;
/**
 * @ignore
 */
CreateCampaignRequestAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    dueDate: {
        baseName: "due_date",
        type: "Date",
        format: "date-time",
    },
    entityScope: {
        baseName: "entity_scope",
        type: "string",
    },
    guidance: {
        baseName: "guidance",
        type: "string",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ownerId: {
        baseName: "owner_id",
        type: "string",
        required: true,
    },
    ruleIds: {
        baseName: "rule_ids",
        type: "Array<string>",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "CampaignStatus",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
