// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomDestinationUpdateRequestDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomDestinationUpdateRequestDefinition = void 0;
/**
 * The definition of a custom destination.
 */
class CustomDestinationUpdateRequestDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomDestinationUpdateRequestDefinition.attributeTypeMap;
    }
}
exports.CustomDestinationUpdateRequestDefinition = CustomDestinationUpdateRequestDefinition;
/**
 * @ignore
 */
CustomDestinationUpdateRequestDefinition.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CustomDestinationUpdateRequestAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "CustomDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
