// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomRulesetRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomRulesetRequestDataAttributes = void 0;
/**
 * Attributes for creating or updating a custom ruleset.
 */
class CustomRulesetRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomRulesetRequestDataAttributes.attributeTypeMap;
    }
}
exports.CustomRulesetRequestDataAttributes = CustomRulesetRequestDataAttributes;
/**
 * @ignore
 */
CustomRulesetRequestDataAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    rules: {
        baseName: "rules",
        type: "Array<CustomRule>",
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
