// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DataExportConfig.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataExportConfig = void 0;
/**
 * AWS Cost and Usage Report data export configuration.
 */
class DataExportConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DataExportConfig.attributeTypeMap;
    }
}
exports.DataExportConfig = DataExportConfig;
/**
 * @ignore
 */
DataExportConfig.attributeTypeMap = {
    bucketName: {
        baseName: "bucket_name",
        type: "string",
        required: true,
    },
    bucketRegion: {
        baseName: "bucket_region",
        type: "string",
        required: true,
    },
    reportName: {
        baseName: "report_name",
        type: "string",
        required: true,
    },
    reportPrefix: {
        baseName: "report_prefix",
        type: "string",
        required: true,
    },
    reportType: {
        baseName: "report_type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
