// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DegradationDataRelationships.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DegradationDataRelationships = void 0;
/**
 * The relationships of a degradation.
 */
class DegradationDataRelationships {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DegradationDataRelationships.attributeTypeMap;
    }
}
exports.DegradationDataRelationships = DegradationDataRelationships;
/**
 * @ignore
 */
DegradationDataRelationships.attributeTypeMap = {
    createdByUser: {
        baseName: "created_by_user",
        type: "DegradationDataRelationshipsCreatedByUser",
    },
    lastModifiedByUser: {
        baseName: "last_modified_by_user",
        type: "DegradationDataRelationshipsLastModifiedByUser",
    },
    statusPage: {
        baseName: "status_page",
        type: "DegradationDataRelationshipsStatusPage",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
