// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationResponseAttributes = void 0;
/**
 * Attributes for a deployment gate evaluation response.
 */
class DeploymentGatesEvaluationResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationResponseAttributes.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationResponseAttributes = DeploymentGatesEvaluationResponseAttributes;
/**
 * @ignore
 */
DeploymentGatesEvaluationResponseAttributes.attributeTypeMap = {
    evaluationId: {
        baseName: "evaluation_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
