// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationRequestAttributes = void 0;
/**
 * Attributes for a deployment gate evaluation request.
 */
class DeploymentGatesEvaluationRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationRequestAttributes.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationRequestAttributes = DeploymentGatesEvaluationRequestAttributes;
/**
 * @ignore
 */
DeploymentGatesEvaluationRequestAttributes.attributeTypeMap = {
    env: {
        baseName: "env",
        type: "string",
        required: true,
    },
    identifier: {
        baseName: "identifier",
        type: "string",
    },
    primaryTag: {
        baseName: "primary_tag",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
