// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationResponseData = void 0;
/**
 * Data for a deployment gate evaluation response.
 */
class DeploymentGatesEvaluationResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationResponseData.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationResponseData = DeploymentGatesEvaluationResponseData;
/**
 * @ignore
 */
DeploymentGatesEvaluationResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "DeploymentGatesEvaluationResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "DeploymentGatesEvaluationResponseDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
