// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationResultResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationResultResponseData = void 0;
/**
 * Data for a deployment gate evaluation result response.
 */
class DeploymentGatesEvaluationResultResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationResultResponseData.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationResultResponseData = DeploymentGatesEvaluationResultResponseData;
/**
 * @ignore
 */
DeploymentGatesEvaluationResultResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "DeploymentGatesEvaluationResultResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "DeploymentGatesEvaluationResultResponseDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
