// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationResultResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationResultResponseAttributes = void 0;
/**
 * Attributes for a deployment gate evaluation result response.
 */
class DeploymentGatesEvaluationResultResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationResultResponseAttributes.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationResultResponseAttributes = DeploymentGatesEvaluationResultResponseAttributes;
/**
 * @ignore
 */
DeploymentGatesEvaluationResultResponseAttributes.attributeTypeMap = {
    dryRun: {
        baseName: "dry_run",
        type: "boolean",
        required: true,
    },
    evaluationId: {
        baseName: "evaluation_id",
        type: "string",
        required: true,
    },
    evaluationUrl: {
        baseName: "evaluation_url",
        type: "string",
        required: true,
    },
    gateId: {
        baseName: "gate_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    gateStatus: {
        baseName: "gate_status",
        type: "DeploymentGatesEvaluationResultResponseAttributesGateStatus",
        required: true,
    },
    rules: {
        baseName: "rules",
        type: "Array<DeploymentGatesRuleResponse>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
