// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DnsMetricKey.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DNS_FAILURE_LATENCY_PERCENTILE = exports.DNS_SUCCESS_LATENCY_PERCENTILE = exports.DNS_RESPONSES_OTHER = exports.DNS_RESPONSES_SERVFAIL = exports.DNS_RESPONSES_NXDOMAIN = exports.DNS_TIMEOUTS = exports.DNS_FAILED_RESPONSES = exports.DNS_SUCCESSFUL_RESPONSES = exports.DNS_FAILURES = exports.DNS_TOTAL_REQUESTS = void 0;
exports.DNS_TOTAL_REQUESTS = "dns_total_requests";
exports.DNS_FAILURES = "dns_failures";
exports.DNS_SUCCESSFUL_RESPONSES = "dns_successful_responses";
exports.DNS_FAILED_RESPONSES = "dns_failed_responses";
exports.DNS_TIMEOUTS = "dns_timeouts";
exports.DNS_RESPONSES_NXDOMAIN = "dns_responses.nxdomain";
exports.DNS_RESPONSES_SERVFAIL = "dns_responses.servfail";
exports.DNS_RESPONSES_OTHER = "dns_responses.other";
exports.DNS_SUCCESS_LATENCY_PERCENTILE = "dns_success_latency_percentile";
exports.DNS_FAILURE_LATENCY_PERCENTILE = "dns_failure_latency_percentile";
