// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DORAGitInfoResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DORAGitInfoResponse = void 0;
/**
 * Git info returned by DORA Metrics events.
 */
class DORAGitInfoResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DORAGitInfoResponse.attributeTypeMap;
    }
}
exports.DORAGitInfoResponse = DORAGitInfoResponse;
/**
 * @ignore
 */
DORAGitInfoResponse.attributeTypeMap = {
    commitSha: {
        baseName: "commit_sha",
        type: "string",
        required: true,
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
