// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityResponseDataRelationshipsOncallsDataItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityResponseDataRelationshipsOncallsDataItems = void 0;
/**
 * Oncall relationship data item containing the oncall resource identifier and type.
 */
class EntityResponseDataRelationshipsOncallsDataItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityResponseDataRelationshipsOncallsDataItems.attributeTypeMap;
    }
}
exports.EntityResponseDataRelationshipsOncallsDataItems = EntityResponseDataRelationshipsOncallsDataItems;
/**
 * @ignore
 */
EntityResponseDataRelationshipsOncallsDataItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "EntityResponseDataRelationshipsOncallsDataItemsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
