// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityResponseDataRelationships.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityResponseDataRelationships = void 0;
/**
 * Entity relationships including incidents, oncalls, schemas, and related entities.
 */
class EntityResponseDataRelationships {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityResponseDataRelationships.attributeTypeMap;
    }
}
exports.EntityResponseDataRelationships = EntityResponseDataRelationships;
/**
 * @ignore
 */
EntityResponseDataRelationships.attributeTypeMap = {
    incidents: {
        baseName: "incidents",
        type: "EntityResponseDataRelationshipsIncidents",
    },
    oncalls: {
        baseName: "oncalls",
        type: "EntityResponseDataRelationshipsOncalls",
    },
    rawSchema: {
        baseName: "rawSchema",
        type: "EntityResponseDataRelationshipsRawSchema",
    },
    relatedEntities: {
        baseName: "relatedEntities",
        type: "EntityResponseDataRelationshipsRelatedEntities",
    },
    schema: {
        baseName: "schema",
        type: "EntityResponseDataRelationshipsSchema",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
