// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityResponseDataRelationshipsRelatedEntitiesDataItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityResponseDataRelationshipsRelatedEntitiesDataItems = void 0;
/**
 * Related entity relationship data item containing the related entity resource identifier and type.
 */
class EntityResponseDataRelationshipsRelatedEntitiesDataItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityResponseDataRelationshipsRelatedEntitiesDataItems.attributeTypeMap;
    }
}
exports.EntityResponseDataRelationshipsRelatedEntitiesDataItems = EntityResponseDataRelationshipsRelatedEntitiesDataItems;
/**
 * @ignore
 */
EntityResponseDataRelationshipsRelatedEntitiesDataItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "EntityResponseDataRelationshipsRelatedEntitiesDataItemsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
