// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityResponseDataRelationshipsIncidentsDataItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityResponseDataRelationshipsIncidentsDataItems = void 0;
/**
 * Incident relationship data item containing the incident resource identifier and type.
 */
class EntityResponseDataRelationshipsIncidentsDataItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityResponseDataRelationshipsIncidentsDataItems.attributeTypeMap;
    }
}
exports.EntityResponseDataRelationshipsIncidentsDataItems = EntityResponseDataRelationshipsIncidentsDataItems;
/**
 * @ignore
 */
EntityResponseDataRelationshipsIncidentsDataItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "EntityResponseDataRelationshipsIncidentsDataItemsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
