// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EnvironmentAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentAttributes = void 0;
/**
 * Attributes of an environment.
 */
class EnvironmentAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EnvironmentAttributes.attributeTypeMap;
    }
}
exports.EnvironmentAttributes = EnvironmentAttributes;
/**
 * @ignore
 */
EnvironmentAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    isProduction: {
        baseName: "is_production",
        type: "boolean",
    },
    key: {
        baseName: "key",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    queries: {
        baseName: "queries",
        type: "Array<string>",
    },
    requireFeatureFlagApproval: {
        baseName: "require_feature_flag_approval",
        type: "boolean",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
