// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ExposureRolloutStepRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExposureRolloutStepRequest = void 0;
/**
 * Rollout step request payload.
 */
class ExposureRolloutStepRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ExposureRolloutStepRequest.attributeTypeMap;
    }
}
exports.ExposureRolloutStepRequest = ExposureRolloutStepRequest;
/**
 * @ignore
 */
ExposureRolloutStepRequest.attributeTypeMap = {
    exposureRatio: {
        baseName: "exposure_ratio",
        type: "number",
        required: true,
        format: "double",
    },
    groupedStepIndex: {
        baseName: "grouped_step_index",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    intervalMs: {
        baseName: "interval_ms",
        type: "number",
        format: "int64",
    },
    isPauseRecord: {
        baseName: "is_pause_record",
        type: "boolean",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
