// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ExposureScheduleRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExposureScheduleRequest = void 0;
/**
 * Progressive release request payload.
 */
class ExposureScheduleRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ExposureScheduleRequest.attributeTypeMap;
    }
}
exports.ExposureScheduleRequest = ExposureScheduleRequest;
/**
 * @ignore
 */
ExposureScheduleRequest.attributeTypeMap = {
    absoluteStartTime: {
        baseName: "absolute_start_time",
        type: "Date",
        format: "date-time",
    },
    controlVariantId: {
        baseName: "control_variant_id",
        type: "string",
    },
    controlVariantKey: {
        baseName: "control_variant_key",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    rolloutOptions: {
        baseName: "rollout_options",
        type: "RolloutOptionsRequest",
        required: true,
    },
    rolloutSteps: {
        baseName: "rollout_steps",
        type: "Array<ExposureRolloutStepRequest>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
