// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FacetInfoResponseDataAttributesResultRange.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FacetInfoResponseDataAttributesResultRange = void 0;
/**
 * The numeric range of a facet attribute, representing the minimum and maximum observed values.
 */
class FacetInfoResponseDataAttributesResultRange {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FacetInfoResponseDataAttributesResultRange.attributeTypeMap;
    }
}
exports.FacetInfoResponseDataAttributesResultRange = FacetInfoResponseDataAttributesResultRange;
/**
 * @ignore
 */
FacetInfoResponseDataAttributesResultRange.attributeTypeMap = {
    max: {
        baseName: "max",
        type: "any",
    },
    min: {
        baseName: "min",
        type: "any",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
