// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FacetInfoResponseDataAttributesResultValuesItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FacetInfoResponseDataAttributesResultValuesItems = void 0;
/**
 * A single facet value with its occurrence count in the dataset.
 */
class FacetInfoResponseDataAttributesResultValuesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FacetInfoResponseDataAttributesResultValuesItems.attributeTypeMap;
    }
}
exports.FacetInfoResponseDataAttributesResultValuesItems = FacetInfoResponseDataAttributesResultValuesItems;
/**
 * @ignore
 */
FacetInfoResponseDataAttributesResultValuesItems.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        format: "int64",
    },
    value: {
        baseName: "value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
