// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FacetInfoRequestDataAttributesSearch.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FacetInfoRequestDataAttributesSearch = void 0;
/**
 * Query-based search configuration for filtering the audience context when retrieving facet values.
 */
class FacetInfoRequestDataAttributesSearch {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FacetInfoRequestDataAttributesSearch.attributeTypeMap;
    }
}
exports.FacetInfoRequestDataAttributesSearch = FacetInfoRequestDataAttributesSearch;
/**
 * @ignore
 */
FacetInfoRequestDataAttributesSearch.attributeTypeMap = {
    query: {
        baseName: "query",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
