// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FeatureFlagEnvironment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureFlagEnvironment = void 0;
/**
 * Environment-specific settings for a feature flag.
 */
class FeatureFlagEnvironment {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FeatureFlagEnvironment.attributeTypeMap;
    }
}
exports.FeatureFlagEnvironment = FeatureFlagEnvironment;
/**
 * @ignore
 */
FeatureFlagEnvironment.attributeTypeMap = {
    allocations: {
        baseName: "allocations",
        type: "{ [key: string]: any; }",
    },
    defaultAllocationKey: {
        baseName: "default_allocation_key",
        type: "string",
    },
    defaultVariantId: {
        baseName: "default_variant_id",
        type: "string",
    },
    environmentId: {
        baseName: "environment_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    environmentName: {
        baseName: "environment_name",
        type: "string",
    },
    environmentQueries: {
        baseName: "environment_queries",
        type: "Array<string>",
    },
    isProduction: {
        baseName: "is_production",
        type: "boolean",
    },
    overrideAllocationKey: {
        baseName: "override_allocation_key",
        type: "string",
    },
    overrideVariantId: {
        baseName: "override_variant_id",
        type: "string",
    },
    pendingSuggestionId: {
        baseName: "pending_suggestion_id",
        type: "string",
    },
    requireFeatureFlagApproval: {
        baseName: "require_feature_flag_approval",
        type: "boolean",
    },
    rolloutPercentage: {
        baseName: "rollout_percentage",
        type: "number",
        format: "int64",
    },
    rules: {
        baseName: "rules",
        type: "Array<{ [key: string]: any; }>",
    },
    status: {
        baseName: "status",
        type: "FeatureFlagStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
