// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FeatureFlagAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureFlagAttributes = void 0;
/**
 * Attributes of a feature flag.
 */
class FeatureFlagAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FeatureFlagAttributes.attributeTypeMap;
    }
}
exports.FeatureFlagAttributes = FeatureFlagAttributes;
/**
 * @ignore
 */
FeatureFlagAttributes.attributeTypeMap = {
    archivedAt: {
        baseName: "archived_at",
        type: "Date",
        format: "date-time",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        format: "uuid",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    distributionChannel: {
        baseName: "distribution_channel",
        type: "string",
    },
    featureFlagEnvironments: {
        baseName: "feature_flag_environments",
        type: "Array<FeatureFlagEnvironment>",
    },
    jsonSchema: {
        baseName: "json_schema",
        type: "string",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    lastUpdatedBy: {
        baseName: "last_updated_by",
        type: "string",
        format: "uuid",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    requireApproval: {
        baseName: "require_approval",
        type: "boolean",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        format: "date-time",
    },
    valueType: {
        baseName: "value_type",
        type: "ValueType",
        required: true,
    },
    variants: {
        baseName: "variants",
        type: "Array<Variant>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
