// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FlakyTestHistoryPolicyMeta.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlakyTestHistoryPolicyMeta = void 0;
/**
 * Metadata about the policy that triggered this status change.
 */
class FlakyTestHistoryPolicyMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FlakyTestHistoryPolicyMeta.attributeTypeMap;
    }
}
exports.FlakyTestHistoryPolicyMeta = FlakyTestHistoryPolicyMeta;
/**
 * @ignore
 */
FlakyTestHistoryPolicyMeta.attributeTypeMap = {
    branches: {
        baseName: "branches",
        type: "Array<string>",
    },
    config: {
        baseName: "config",
        type: "FlakyTestHistoryPolicyMetaConfig",
    },
    daysActive: {
        baseName: "days_active",
        type: "number",
        format: "int32",
    },
    daysWithoutFlake: {
        baseName: "days_without_flake",
        type: "number",
        format: "int32",
    },
    failureRate: {
        baseName: "failure_rate",
        type: "number",
        format: "double",
    },
    state: {
        baseName: "state",
        type: "string",
    },
    totalRuns: {
        baseName: "total_runs",
        type: "number",
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
