// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FlakyTestsSearchSort.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PIPELINES_DURATION_LOST_DESCENDING = exports.PIPELINES_DURATION_LOST_ASCENDING = exports.PIPELINES_FAILED_DESCENDING = exports.PIPELINES_FAILED_ASCENDING = exports.FAILURE_RATE_DESCENDING = exports.FAILURE_RATE_ASCENDING = exports.LAST_FLAKED_DESCENDING = exports.LAST_FLAKED_ASCENDING = exports.FIRST_FLAKED_DESCENDING = exports.FIRST_FLAKED_ASCENDING = exports.FQN_DESCENDING = exports.FQN_ASCENDING = void 0;
exports.FQN_ASCENDING = "fqn";
exports.FQN_DESCENDING = "-fqn";
exports.FIRST_FLAKED_ASCENDING = "first_flaked";
exports.FIRST_FLAKED_DESCENDING = "-first_flaked";
exports.LAST_FLAKED_ASCENDING = "last_flaked";
exports.LAST_FLAKED_DESCENDING = "-last_flaked";
exports.FAILURE_RATE_ASCENDING = "failure_rate";
exports.FAILURE_RATE_DESCENDING = "-failure_rate";
exports.PIPELINES_FAILED_ASCENDING = "pipelines_failed";
exports.PIPELINES_FAILED_DESCENDING = "-pipelines_failed";
exports.PIPELINES_DURATION_LOST_ASCENDING = "pipelines_duration_lost";
exports.PIPELINES_DURATION_LOST_DESCENDING = "-pipelines_duration_lost";
