// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FlakyTestHistoryPolicyMetaConfig.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlakyTestHistoryPolicyMetaConfig = void 0;
/**
 * Configuration parameters of the policy that triggered this status change.
 */
class FlakyTestHistoryPolicyMetaConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FlakyTestHistoryPolicyMetaConfig.attributeTypeMap;
    }
}
exports.FlakyTestHistoryPolicyMetaConfig = FlakyTestHistoryPolicyMetaConfig;
/**
 * @ignore
 */
FlakyTestHistoryPolicyMetaConfig.attributeTypeMap = {
    branches: {
        baseName: "branches",
        type: "Array<string>",
    },
    daysActive: {
        baseName: "days_active",
        type: "number",
        format: "int32",
    },
    failureRate: {
        baseName: "failure_rate",
        type: "number",
        format: "double",
    },
    forgetBranches: {
        baseName: "forget_branches",
        type: "Array<string>",
    },
    requiredRuns: {
        baseName: "required_runs",
        type: "number",
        format: "int32",
    },
    state: {
        baseName: "state",
        type: "string",
    },
    testServices: {
        baseName: "test_services",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
