// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FlakyTestHistory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlakyTestHistory = void 0;
/**
 * A single history entry representing a status change for a flaky test.
 */
class FlakyTestHistory {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FlakyTestHistory.attributeTypeMap;
    }
}
exports.FlakyTestHistory = FlakyTestHistory;
/**
 * @ignore
 */
FlakyTestHistory.attributeTypeMap = {
    commitSha: {
        baseName: "commit_sha",
        type: "string",
        required: true,
    },
    policyId: {
        baseName: "policy_id",
        type: "FlakyTestHistoryPolicyId",
    },
    policyMeta: {
        baseName: "policy_meta",
        type: "FlakyTestHistoryPolicyMeta",
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    timestamp: {
        baseName: "timestamp",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
