// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FlakyTestHistoryPolicyId.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UNKNOWN = exports.QUARANTINE_DAYS_ACTIVE = exports.QUARANTINE_BRANCH_FLAKE = exports.QUARANTINE_FAILURE_RATE = exports.DISABLE_DAYS_ACTIVE = exports.DISABLE_BRANCH_FLAKE = exports.DISABLE_FAILURE_RATE = exports.FIXED = exports.MANUAL = void 0;
exports.MANUAL = "ftm_policy.manual";
exports.FIXED = "ftm_policy.fixed";
exports.DISABLE_FAILURE_RATE = "ftm_policy.disable.failure_rate";
exports.DISABLE_BRANCH_FLAKE = "ftm_policy.disable.branch_flake";
exports.DISABLE_DAYS_ACTIVE = "ftm_policy.disable.days_active";
exports.QUARANTINE_FAILURE_RATE = "ftm_policy.quarantine.failure_rate";
exports.QUARANTINE_BRANCH_FLAKE = "ftm_policy.quarantine.branch_flake";
exports.QUARANTINE_DAYS_ACTIVE = "ftm_policy.quarantine.days_active";
exports.UNKNOWN = "unknown";
