// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FleetClusterAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetClusterAttributes = void 0;
/**
 * Attributes of a Kubernetes cluster in the fleet.
 */
class FleetClusterAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FleetClusterAttributes.attributeTypeMap;
    }
}
exports.FleetClusterAttributes = FleetClusterAttributes;
/**
 * @ignore
 */
FleetClusterAttributes.attributeTypeMap = {
    agentVersions: {
        baseName: "agent_versions",
        type: "Array<string>",
    },
    apiKeyNames: {
        baseName: "api_key_names",
        type: "Array<string>",
    },
    apiKeyUuids: {
        baseName: "api_key_uuids",
        type: "Array<string>",
    },
    cloudProviders: {
        baseName: "cloud_providers",
        type: "Array<string>",
    },
    clusterName: {
        baseName: "cluster_name",
        type: "string",
    },
    enabledProducts: {
        baseName: "enabled_products",
        type: "Array<string>",
    },
    envs: {
        baseName: "envs",
        type: "Array<string>",
    },
    firstSeenAt: {
        baseName: "first_seen_at",
        type: "number",
        format: "int64",
    },
    installMethodTool: {
        baseName: "install_method_tool",
        type: "string",
    },
    nodeCount: {
        baseName: "node_count",
        type: "number",
        format: "int64",
    },
    nodeCountByStatus: {
        baseName: "node_count_by_status",
        type: "{ [key: string]: number; }",
    },
    operatingSystems: {
        baseName: "operating_systems",
        type: "Array<string>",
    },
    otelCollectorDistributions: {
        baseName: "otel_collector_distributions",
        type: "Array<string>",
    },
    otelCollectorVersions: {
        baseName: "otel_collector_versions",
        type: "Array<string>",
    },
    podCountByState: {
        baseName: "pod_count_by_state",
        type: "{ [key: string]: number; }",
    },
    services: {
        baseName: "services",
        type: "Array<string>",
    },
    teams: {
        baseName: "teams",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
