// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FleetInstrumentedPodGroupAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetInstrumentedPodGroupAttributes = void 0;
/**
 * Attributes of a group of instrumented pods targeted for SSI injection.
 */
class FleetInstrumentedPodGroupAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FleetInstrumentedPodGroupAttributes.attributeTypeMap;
    }
}
exports.FleetInstrumentedPodGroupAttributes = FleetInstrumentedPodGroupAttributes;
/**
 * @ignore
 */
FleetInstrumentedPodGroupAttributes.attributeTypeMap = {
    appliedTarget: {
        baseName: "applied_target",
        type: "{ [key: string]: any; }",
    },
    appliedTargetName: {
        baseName: "applied_target_name",
        type: "string",
    },
    injectedTags: {
        baseName: "injected_tags",
        type: "Array<string>",
    },
    kubeOwnerrefKind: {
        baseName: "kube_ownerref_kind",
        type: "string",
    },
    kubeOwnerrefName: {
        baseName: "kube_ownerref_name",
        type: "string",
    },
    libInjectionAnnotations: {
        baseName: "lib_injection_annotations",
        type: "Array<string>",
    },
    namespace: {
        baseName: "namespace",
        type: "string",
    },
    podCount: {
        baseName: "pod_count",
        type: "number",
        format: "int64",
    },
    podNames: {
        baseName: "pod_names",
        type: "Array<string>",
    },
    tags: {
        baseName: "tags",
        type: "{ [key: string]: string; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
