// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FleetInstrumentedPodsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetInstrumentedPodsResponseData = void 0;
/**
 * The response data containing the cluster name and instrumented pod groups.
 */
class FleetInstrumentedPodsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FleetInstrumentedPodsResponseData.attributeTypeMap;
    }
}
exports.FleetInstrumentedPodsResponseData = FleetInstrumentedPodsResponseData;
/**
 * @ignore
 */
FleetInstrumentedPodsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "FleetInstrumentedPodsResponseDataAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
