// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FleetTracerAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetTracerAttributes = void 0;
/**
 * Attributes of a fleet tracer representing a service instance reporting telemetry.
 */
class FleetTracerAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FleetTracerAttributes.attributeTypeMap;
    }
}
exports.FleetTracerAttributes = FleetTracerAttributes;
/**
 * @ignore
 */
FleetTracerAttributes.attributeTypeMap = {
    env: {
        baseName: "env",
        type: "string",
    },
    hostname: {
        baseName: "hostname",
        type: "string",
    },
    language: {
        baseName: "language",
        type: "string",
    },
    languageVersion: {
        baseName: "language_version",
        type: "string",
    },
    remoteConfigStatus: {
        baseName: "remote_config_status",
        type: "string",
    },
    runtimeIds: {
        baseName: "runtime_ids",
        type: "Array<string>",
    },
    service: {
        baseName: "service",
        type: "string",
    },
    serviceHostname: {
        baseName: "service_hostname",
        type: "string",
    },
    serviceVersion: {
        baseName: "service_version",
        type: "string",
    },
    tracerVersion: {
        baseName: "tracer_version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
