// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FullPersonalAccessTokenAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FullPersonalAccessTokenAttributes = void 0;
/**
 * Attributes of a full personal access token, including the token key.
 */
class FullPersonalAccessTokenAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FullPersonalAccessTokenAttributes.attributeTypeMap;
    }
}
exports.FullPersonalAccessTokenAttributes = FullPersonalAccessTokenAttributes;
/**
 * @ignore
 */
FullPersonalAccessTokenAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    expiresAt: {
        baseName: "expires_at",
        type: "Date",
        format: "date-time",
    },
    key: {
        baseName: "key",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    publicPortion: {
        baseName: "public_portion",
        type: "string",
    },
    scopes: {
        baseName: "scopes",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
