// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetInvestigationResponseDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetInvestigationResponseDataAttributes = void 0;
/**
 * Attributes of the investigation.
 */
class GetInvestigationResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetInvestigationResponseDataAttributes.attributeTypeMap;
    }
}
exports.GetInvestigationResponseDataAttributes = GetInvestigationResponseDataAttributes;
/**
 * @ignore
 */
GetInvestigationResponseDataAttributes.attributeTypeMap = {
    conclusions: {
        baseName: "conclusions",
        type: "Array<InvestigationConclusion>",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
