// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems = void 0;
/**
 * An argument parameter for a static analysis rule, with a name and description.
 */
class GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems.attributeTypeMap;
    }
}
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems = GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems;
/**
 * @ignore
 */
GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsArgumentsItems.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
