// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData = void 0;
/**
 * The resource identifier and type for a static analysis rule.
 */
class GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData.attributeTypeMap;
    }
}
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData = GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData;
/**
 * @ignore
 */
GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
