// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetMultipleRulesetsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMultipleRulesetsResponseData = void 0;
/**
 * The primary data object in the get-multiple-rulesets response, containing the response attributes and resource type.
 */
class GetMultipleRulesetsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetMultipleRulesetsResponseData.attributeTypeMap;
    }
}
exports.GetMultipleRulesetsResponseData = GetMultipleRulesetsResponseData;
/**
 * @ignore
 */
GetMultipleRulesetsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "GetMultipleRulesetsResponseDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "GetMultipleRulesetsResponseDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
