// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetMultipleRulesetsResponseDataAttributesRulesetsItemsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsData = void 0;
/**
 * The resource identifier and type for a ruleset.
 */
class GetMultipleRulesetsResponseDataAttributesRulesetsItemsData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetMultipleRulesetsResponseDataAttributesRulesetsItemsData.attributeTypeMap;
    }
}
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsData = GetMultipleRulesetsResponseDataAttributesRulesetsItemsData;
/**
 * @ignore
 */
GetMultipleRulesetsResponseDataAttributesRulesetsItemsData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "GetMultipleRulesetsResponseDataAttributesRulesetsItemsDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
