// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems = void 0;
/**
 * A test case associated with a static analysis rule, containing the source code and expected annotation count.
 */
class GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems.attributeTypeMap;
    }
}
exports.GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems = GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems;
/**
 * @ignore
 */
GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItemsTestsItems.attributeTypeMap = {
    annotationCount: {
        baseName: "annotation_count",
        type: "number",
        format: "int64",
    },
    code: {
        baseName: "code",
        type: "string",
    },
    filename: {
        baseName: "filename",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
