// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GlobalIncidentSettingsDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalIncidentSettingsDataResponse = void 0;
/**
 * Data object in the global incident settings response.
 */
class GlobalIncidentSettingsDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GlobalIncidentSettingsDataResponse.attributeTypeMap;
    }
}
exports.GlobalIncidentSettingsDataResponse = GlobalIncidentSettingsDataResponse;
/**
 * @ignore
 */
GlobalIncidentSettingsDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "GlobalIncidentSettingsAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "GlobalIncidentSettingsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
