// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GlobalOrgIdentifier.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalOrgIdentifier = void 0;
/**
 * A unique identifier for an organization including its site.
 */
class GlobalOrgIdentifier {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GlobalOrgIdentifier.attributeTypeMap;
    }
}
exports.GlobalOrgIdentifier = GlobalOrgIdentifier;
/**
 * @ignore
 */
GlobalOrgIdentifier.attributeTypeMap = {
    orgSite: {
        baseName: "org_site",
        type: "string",
        required: true,
    },
    orgUuid: {
        baseName: "org_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
