// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/HistoricalJobResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoricalJobResponseAttributes = void 0;
/**
 * Historical job attributes.
 */
class HistoricalJobResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HistoricalJobResponseAttributes.attributeTypeMap;
    }
}
exports.HistoricalJobResponseAttributes = HistoricalJobResponseAttributes;
/**
 * @ignore
 */
HistoricalJobResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "createdAt",
        type: "string",
    },
    createdByHandle: {
        baseName: "createdByHandle",
        type: "string",
    },
    createdByName: {
        baseName: "createdByName",
        type: "string",
    },
    createdFromRuleId: {
        baseName: "createdFromRuleId",
        type: "string",
    },
    jobDefinition: {
        baseName: "jobDefinition",
        type: "JobDefinition",
    },
    jobName: {
        baseName: "jobName",
        type: "string",
    },
    jobStatus: {
        baseName: "jobStatus",
        type: "string",
    },
    modifiedAt: {
        baseName: "modifiedAt",
        type: "string",
    },
    signalOutput: {
        baseName: "signalOutput",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
