// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/HistoricalJobOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoricalJobOptions = void 0;
/**
 * Job options.
 */
class HistoricalJobOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HistoricalJobOptions.attributeTypeMap;
    }
}
exports.HistoricalJobOptions = HistoricalJobOptions;
/**
 * @ignore
 */
HistoricalJobOptions.attributeTypeMap = {
    anomalyDetectionOptions: {
        baseName: "anomalyDetectionOptions",
        type: "SecurityMonitoringRuleAnomalyDetectionOptions",
    },
    detectionMethod: {
        baseName: "detectionMethod",
        type: "SecurityMonitoringRuleDetectionMethod",
    },
    evaluationWindow: {
        baseName: "evaluationWindow",
        type: "SecurityMonitoringRuleEvaluationWindow",
    },
    impossibleTravelOptions: {
        baseName: "impossibleTravelOptions",
        type: "SecurityMonitoringRuleImpossibleTravelOptions",
    },
    keepAlive: {
        baseName: "keepAlive",
        type: "SecurityMonitoringRuleKeepAlive",
    },
    maxSignalDuration: {
        baseName: "maxSignalDuration",
        type: "SecurityMonitoringRuleMaxSignalDuration",
    },
    newValueOptions: {
        baseName: "newValueOptions",
        type: "SecurityMonitoringRuleNewValueOptions",
    },
    sequenceDetectionOptions: {
        baseName: "sequenceDetectionOptions",
        type: "SecurityMonitoringRuleSequenceDetectionOptions",
    },
    thirdPartyRuleOptions: {
        baseName: "thirdPartyRuleOptions",
        type: "SecurityMonitoringRuleThirdPartyOptions",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
