// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldResponseData = void 0;
/**
 * Data object for an incident user-defined field response.
 */
class IncidentUserDefinedFieldResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldResponseData.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldResponseData = IncidentUserDefinedFieldResponseData;
/**
 * @ignore
 */
IncidentUserDefinedFieldResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "IncidentUserDefinedFieldAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "IncidentUserDefinedFieldRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "IncidentUserDefinedFieldType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
