// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldMetadata.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldMetadata = void 0;
/**
 * Metadata for autocomplete-type user-defined fields, describing how to populate autocomplete options.
 */
class IncidentUserDefinedFieldMetadata {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldMetadata.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldMetadata = IncidentUserDefinedFieldMetadata;
/**
 * @ignore
 */
IncidentUserDefinedFieldMetadata.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "string",
        required: true,
    },
    searchLimitParam: {
        baseName: "search_limit_param",
        type: "string",
        required: true,
    },
    searchParams: {
        baseName: "search_params",
        type: "{ [key: string]: any; }",
        required: true,
    },
    searchQueryParam: {
        baseName: "search_query_param",
        type: "string",
        required: true,
    },
    searchResultPath: {
        baseName: "search_result_path",
        type: "string",
        required: true,
    },
    searchUrl: {
        baseName: "search_url",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
