// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldUpdateData = void 0;
/**
 * Data for updating an incident user-defined field.
 */
class IncidentUserDefinedFieldUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldUpdateData.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldUpdateData = IncidentUserDefinedFieldUpdateData;
/**
 * @ignore
 */
IncidentUserDefinedFieldUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "IncidentUserDefinedFieldAttributesUpdateRequest",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "IncidentUserDefinedFieldType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
