// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldRelationships.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldRelationships = void 0;
/**
 * Relationships of an incident user-defined field.
 */
class IncidentUserDefinedFieldRelationships {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldRelationships.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldRelationships = IncidentUserDefinedFieldRelationships;
/**
 * @ignore
 */
IncidentUserDefinedFieldRelationships.attributeTypeMap = {
    createdByUser: {
        baseName: "created_by_user",
        type: "RelationshipToUser",
        required: true,
    },
    incidentType: {
        baseName: "incident_type",
        type: "RelationshipToIncidentType",
        required: true,
    },
    lastModifiedByUser: {
        baseName: "last_modified_by_user",
        type: "RelationshipToUser",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
