// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldAttributesResponse = void 0;
/**
 * Attributes of an incident user-defined field.
 */
class IncidentUserDefinedFieldAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldAttributesResponse.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldAttributesResponse = IncidentUserDefinedFieldAttributesResponse;
/**
 * @ignore
 */
IncidentUserDefinedFieldAttributesResponse.attributeTypeMap = {
    attachedTo: {
        baseName: "attached_to",
        type: "string",
        required: true,
    },
    category: {
        baseName: "category",
        type: "IncidentUserDefinedFieldCategory",
        required: true,
    },
    collected: {
        baseName: "collected",
        type: "IncidentUserDefinedFieldCollected",
        required: true,
    },
    created: {
        baseName: "created",
        type: "Date",
        required: true,
        format: "date-time",
    },
    defaultValue: {
        baseName: "default_value",
        type: "string",
        required: true,
    },
    deleted: {
        baseName: "deleted",
        type: "Date",
        required: true,
        format: "date-time",
    },
    displayName: {
        baseName: "display_name",
        type: "string",
        required: true,
    },
    metadata: {
        baseName: "metadata",
        type: "IncidentUserDefinedFieldMetadata",
        required: true,
    },
    modified: {
        baseName: "modified",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ordinal: {
        baseName: "ordinal",
        type: "string",
        required: true,
    },
    prerequisite: {
        baseName: "prerequisite",
        type: "string",
        required: true,
    },
    required: {
        baseName: "required",
        type: "boolean",
        required: true,
    },
    reserved: {
        baseName: "reserved",
        type: "boolean",
        required: true,
    },
    tableId: {
        baseName: "table_id",
        type: "number",
        required: true,
        format: "int64",
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "number",
        required: true,
        format: "int32",
    },
    validValues: {
        baseName: "valid_values",
        type: "Array<IncidentUserDefinedFieldValidValue>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
