// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldAttributesCreateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldAttributesCreateRequest = void 0;
/**
 * Attributes for creating an incident user-defined field.
 */
class IncidentUserDefinedFieldAttributesCreateRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldAttributesCreateRequest.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldAttributesCreateRequest = IncidentUserDefinedFieldAttributesCreateRequest;
/**
 * @ignore
 */
IncidentUserDefinedFieldAttributesCreateRequest.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "IncidentUserDefinedFieldCategory",
    },
    collected: {
        baseName: "collected",
        type: "IncidentUserDefinedFieldCollected",
    },
    defaultValue: {
        baseName: "default_value",
        type: "string",
    },
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ordinal: {
        baseName: "ordinal",
        type: "string",
    },
    required: {
        baseName: "required",
        type: "boolean",
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "IncidentUserDefinedFieldFieldType",
        required: true,
    },
    validValues: {
        baseName: "valid_values",
        type: "Array<IncidentUserDefinedFieldValidValue>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
