// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldAttributesUpdateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldAttributesUpdateRequest = void 0;
/**
 * Attributes for updating an incident user-defined field. All fields are optional.
 */
class IncidentUserDefinedFieldAttributesUpdateRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldAttributesUpdateRequest.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldAttributesUpdateRequest = IncidentUserDefinedFieldAttributesUpdateRequest;
/**
 * @ignore
 */
IncidentUserDefinedFieldAttributesUpdateRequest.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "IncidentUserDefinedFieldCategory",
    },
    collected: {
        baseName: "collected",
        type: "IncidentUserDefinedFieldCollected",
    },
    defaultValue: {
        baseName: "default_value",
        type: "string",
    },
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    ordinal: {
        baseName: "ordinal",
        type: "string",
    },
    required: {
        baseName: "required",
        type: "boolean",
    },
    validValues: {
        baseName: "valid_values",
        type: "Array<IncidentUserDefinedFieldValidValue>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
