// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldCreateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldCreateData = void 0;
/**
 * Data for creating an incident user-defined field.
 */
class IncidentUserDefinedFieldCreateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldCreateData.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldCreateData = IncidentUserDefinedFieldCreateData;
/**
 * @ignore
 */
IncidentUserDefinedFieldCreateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "IncidentUserDefinedFieldAttributesCreateRequest",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "IncidentUserDefinedFieldCreateRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "IncidentUserDefinedFieldType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
