// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldValidValue.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldValidValue = void 0;
/**
 * A valid value for an incident user-defined field.
 */
class IncidentUserDefinedFieldValidValue {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldValidValue.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldValidValue = IncidentUserDefinedFieldValidValue;
/**
 * @ignore
 */
IncidentUserDefinedFieldValidValue.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    displayName: {
        baseName: "display_name",
        type: "string",
        required: true,
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
