// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IncidentUserDefinedFieldListMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IncidentUserDefinedFieldListMeta = void 0;
/**
 * Pagination metadata for the user-defined field list response.
 */
class IncidentUserDefinedFieldListMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IncidentUserDefinedFieldListMeta.attributeTypeMap;
    }
}
exports.IncidentUserDefinedFieldListMeta = IncidentUserDefinedFieldListMeta;
/**
 * @ignore
 */
IncidentUserDefinedFieldListMeta.attributeTypeMap = {
    offset: {
        baseName: "offset",
        type: "number",
        format: "int64",
    },
    size: {
        baseName: "size",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
