// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IntegrationOnCallEscalationQueriesItemsTarget.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationOnCallEscalationQueriesItemsTarget = void 0;
/**
 * The target recipient for an On-Call escalation query.
 */
class IntegrationOnCallEscalationQueriesItemsTarget {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IntegrationOnCallEscalationQueriesItemsTarget.attributeTypeMap;
    }
}
exports.IntegrationOnCallEscalationQueriesItemsTarget = IntegrationOnCallEscalationQueriesItemsTarget;
/**
 * @ignore
 */
IntegrationOnCallEscalationQueriesItemsTarget.attributeTypeMap = {
    dynamicTeamPaging: {
        baseName: "dynamic_team_paging",
        type: "boolean",
    },
    teamId: {
        baseName: "team_id",
        type: "string",
    },
    userId: {
        baseName: "user_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
