// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IntegrationOnCallEscalationQueriesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationOnCallEscalationQueriesItems = void 0;
/**
 * An On-Call escalation query entry used to route cases to on-call responders.
 */
class IntegrationOnCallEscalationQueriesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IntegrationOnCallEscalationQueriesItems.attributeTypeMap;
    }
}
exports.IntegrationOnCallEscalationQueriesItems = IntegrationOnCallEscalationQueriesItems;
/**
 * @ignore
 */
IntegrationOnCallEscalationQueriesItems.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "IntegrationOnCallEscalationQueriesItemsTarget",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
