// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/InvestigationConclusion.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvestigationConclusion = void 0;
/**
 * A full explanation of the finding, including root cause analysis and supporting evidence.
 */
class InvestigationConclusion {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return InvestigationConclusion.attributeTypeMap;
    }
}
exports.InvestigationConclusion = InvestigationConclusion;
/**
 * @ignore
 */
InvestigationConclusion.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    summary: {
        baseName: "summary",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
