// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IoCGeoLocation.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoCGeoLocation = void 0;
/**
 * Geographic location information for an IP indicator.
 */
class IoCGeoLocation {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IoCGeoLocation.attributeTypeMap;
    }
}
exports.IoCGeoLocation = IoCGeoLocation;
/**
 * @ignore
 */
IoCGeoLocation.attributeTypeMap = {
    city: {
        baseName: "city",
        type: "string",
    },
    countryCode: {
        baseName: "country_code",
        type: "string",
    },
    countryName: {
        baseName: "country_name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
