// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IoCIndicator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoCIndicator = void 0;
/**
 * An indicator of compromise with threat intelligence data.
 */
class IoCIndicator {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IoCIndicator.attributeTypeMap;
    }
}
exports.IoCIndicator = IoCIndicator;
/**
 * @ignore
 */
IoCIndicator.attributeTypeMap = {
    asGeo: {
        baseName: "as_geo",
        type: "IoCGeoLocation",
    },
    asType: {
        baseName: "as_type",
        type: "string",
    },
    benignSources: {
        baseName: "benign_sources",
        type: "Array<IoCSource>",
    },
    categories: {
        baseName: "categories",
        type: "Array<string>",
    },
    firstSeen: {
        baseName: "first_seen",
        type: "Date",
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    indicator: {
        baseName: "indicator",
        type: "string",
    },
    indicatorType: {
        baseName: "indicator_type",
        type: "string",
    },
    lastSeen: {
        baseName: "last_seen",
        type: "Date",
        format: "date-time",
    },
    logMatches: {
        baseName: "log_matches",
        type: "number",
        format: "int64",
    },
    mAsType: {
        baseName: "m_as_type",
        type: "IoCScoreEffect",
    },
    mPersistence: {
        baseName: "m_persistence",
        type: "IoCScoreEffect",
    },
    mSignal: {
        baseName: "m_signal",
        type: "IoCScoreEffect",
    },
    mSources: {
        baseName: "m_sources",
        type: "IoCScoreEffect",
    },
    maliciousSources: {
        baseName: "malicious_sources",
        type: "Array<IoCSource>",
    },
    maxTrustScore: {
        baseName: "max_trust_score",
        type: "IoCScoreEffect",
    },
    score: {
        baseName: "score",
        type: "number",
        format: "double",
    },
    signalMatches: {
        baseName: "signal_matches",
        type: "number",
        format: "int64",
    },
    signalTier: {
        baseName: "signal_tier",
        type: "number",
        format: "int64",
    },
    suspiciousSources: {
        baseName: "suspicious_sources",
        type: "Array<IoCSource>",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
