// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IoCIndicatorDetailed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoCIndicatorDetailed = void 0;
/**
 * An indicator of compromise with extended context from your environment.
 */
class IoCIndicatorDetailed {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IoCIndicatorDetailed.attributeTypeMap;
    }
}
exports.IoCIndicatorDetailed = IoCIndicatorDetailed;
/**
 * @ignore
 */
IoCIndicatorDetailed.attributeTypeMap = {
    additionalData: {
        baseName: "additional_data",
        type: "{ [key: string]: any; }",
    },
    asCidrBlock: {
        baseName: "as_cidr_block",
        type: "string",
    },
    asGeo: {
        baseName: "as_geo",
        type: "IoCGeoLocation",
    },
    asNumber: {
        baseName: "as_number",
        type: "string",
    },
    asOrganization: {
        baseName: "as_organization",
        type: "string",
    },
    asType: {
        baseName: "as_type",
        type: "string",
    },
    benignSources: {
        baseName: "benign_sources",
        type: "Array<IoCSource>",
    },
    categories: {
        baseName: "categories",
        type: "Array<string>",
    },
    criticalAssets: {
        baseName: "critical_assets",
        type: "Array<string>",
    },
    firstSeen: {
        baseName: "first_seen",
        type: "Date",
        format: "date-time",
    },
    hosts: {
        baseName: "hosts",
        type: "Array<string>",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    indicator: {
        baseName: "indicator",
        type: "string",
    },
    indicatorType: {
        baseName: "indicator_type",
        type: "string",
    },
    lastSeen: {
        baseName: "last_seen",
        type: "Date",
        format: "date-time",
    },
    logMatches: {
        baseName: "log_matches",
        type: "number",
        format: "int64",
    },
    logSources: {
        baseName: "log_sources",
        type: "Array<string>",
    },
    mAsType: {
        baseName: "m_as_type",
        type: "IoCScoreEffect",
    },
    mPersistence: {
        baseName: "m_persistence",
        type: "IoCScoreEffect",
    },
    mSignal: {
        baseName: "m_signal",
        type: "IoCScoreEffect",
    },
    mSources: {
        baseName: "m_sources",
        type: "IoCScoreEffect",
    },
    maliciousSources: {
        baseName: "malicious_sources",
        type: "Array<IoCSource>",
    },
    maxTrustScore: {
        baseName: "max_trust_score",
        type: "IoCScoreEffect",
    },
    score: {
        baseName: "score",
        type: "number",
        format: "double",
    },
    services: {
        baseName: "services",
        type: "Array<string>",
    },
    signalMatches: {
        baseName: "signal_matches",
        type: "number",
        format: "int64",
    },
    signalSeverity: {
        baseName: "signal_severity",
        type: "Array<IoCSignalSeverityCount>",
    },
    signalTier: {
        baseName: "signal_tier",
        type: "number",
        format: "int64",
    },
    suspiciousSources: {
        baseName: "suspicious_sources",
        type: "Array<IoCSource>",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    users: {
        baseName: "users",
        type: "{ [key: string]: Array<string>; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
