// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IoCSignalSeverityCount.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoCSignalSeverityCount = void 0;
/**
 * Count of security signals by severity level.
 */
class IoCSignalSeverityCount {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IoCSignalSeverityCount.attributeTypeMap;
    }
}
exports.IoCSignalSeverityCount = IoCSignalSeverityCount;
/**
 * @ignore
 */
IoCSignalSeverityCount.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        format: "int64",
    },
    severity: {
        baseName: "severity",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
