// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IssueLanguage.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UNKNOWN = exports.TYPESCRIPT = exports.TERRAFORM = exports.SWIFT = exports.SCALA = exports.RUST = exports.RUBY = exports.PYTHON = exports.PHP = exports.PERL = exports.OBJECTIVE_C = exports.KOTLIN = exports.JVM = exports.JAVASCRIPT = exports.JAVA = exports.HCL = exports.HASKELL = exports.GROOVY = exports.GO = exports.ERLANG = exports.ELIXIR = exports.DOT_NET = exports.CLOJURE = exports.C_SHARP = exports.C_PLUS_PLUS = exports.C = exports.BRIGHTSCRIPT = void 0;
exports.BRIGHTSCRIPT = "BRIGHTSCRIPT";
exports.C = "C";
exports.C_PLUS_PLUS = "C_PLUS_PLUS";
exports.C_SHARP = "C_SHARP";
exports.CLOJURE = "CLOJURE";
exports.DOT_NET = "DOT_NET";
exports.ELIXIR = "ELIXIR";
exports.ERLANG = "ERLANG";
exports.GO = "GO";
exports.GROOVY = "GROOVY";
exports.HASKELL = "HASKELL";
exports.HCL = "HCL";
exports.JAVA = "JAVA";
exports.JAVASCRIPT = "JAVASCRIPT";
exports.JVM = "JVM";
exports.KOTLIN = "KOTLIN";
exports.OBJECTIVE_C = "OBJECTIVE_C";
exports.PERL = "PERL";
exports.PHP = "PHP";
exports.PYTHON = "PYTHON";
exports.RUBY = "RUBY";
exports.RUST = "RUST";
exports.SCALA = "SCALA";
exports.SWIFT = "SWIFT";
exports.TERRAFORM = "TERRAFORM";
exports.TYPESCRIPT = "TYPESCRIPT";
exports.UNKNOWN = "UNKNOWN";
