// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ListInvestigationsResponseDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListInvestigationsResponseDataAttributes = void 0;
/**
 * Attributes of an investigation list item.
 */
class ListInvestigationsResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ListInvestigationsResponseDataAttributes.attributeTypeMap;
    }
}
exports.ListInvestigationsResponseDataAttributes = ListInvestigationsResponseDataAttributes;
/**
 * @ignore
 */
ListInvestigationsResponseDataAttributes.attributeTypeMap = {
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
