// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigVertexAIOptions.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigVertexAIOptions = void 0;
/**
 * Google Vertex AI-specific options for LLM provider configuration.
 */
class LLMObsCustomEvalConfigVertexAIOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigVertexAIOptions.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigVertexAIOptions = LLMObsCustomEvalConfigVertexAIOptions;
/**
 * @ignore
 */
LLMObsCustomEvalConfigVertexAIOptions.attributeTypeMap = {
    location: {
        baseName: "location",
        type: "string",
    },
    project: {
        baseName: "project",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
