// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigAssessmentCriteria.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigAssessmentCriteria = void 0;
/**
 * Criteria used to assess the pass/fail result of a custom evaluator.
 */
class LLMObsCustomEvalConfigAssessmentCriteria {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigAssessmentCriteria.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigAssessmentCriteria = LLMObsCustomEvalConfigAssessmentCriteria;
/**
 * @ignore
 */
LLMObsCustomEvalConfigAssessmentCriteria.attributeTypeMap = {
    maxThreshold: {
        baseName: "max_threshold",
        type: "number",
        format: "double",
    },
    minThreshold: {
        baseName: "min_threshold",
        type: "number",
        format: "double",
    },
    passValues: {
        baseName: "pass_values",
        type: "Array<string>",
    },
    passWhen: {
        baseName: "pass_when",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
