// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigTarget.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigTarget = void 0;
/**
 * Target application configuration for a custom evaluator.
 */
class LLMObsCustomEvalConfigTarget {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigTarget.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigTarget = LLMObsCustomEvalConfigTarget;
/**
 * @ignore
 */
LLMObsCustomEvalConfigTarget.attributeTypeMap = {
    applicationName: {
        baseName: "application_name",
        type: "string",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    evalScope: {
        baseName: "eval_scope",
        type: "LLMObsCustomEvalConfigEvalScope",
    },
    filter: {
        baseName: "filter",
        type: "string",
    },
    rootSpansOnly: {
        baseName: "root_spans_only",
        type: "boolean",
    },
    samplingPercentage: {
        baseName: "sampling_percentage",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
