// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationItem.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationItem = void 0;
/**
 * A single annotation on an interaction.
 */
class LLMObsAnnotationItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationItem.attributeTypeMap;
    }
}
exports.LLMObsAnnotationItem = LLMObsAnnotationItem;
/**
 * @ignore
 */
LLMObsAnnotationItem.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    interactionId: {
        baseName: "interaction_id",
        type: "string",
        required: true,
    },
    labelValues: {
        baseName: "label_values",
        type: "{ [key: string]: any; }",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedBy: {
        baseName: "modified_by",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
