// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigInferenceParams.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigInferenceParams = void 0;
/**
 * LLM inference parameters for a custom evaluator.
 */
class LLMObsCustomEvalConfigInferenceParams {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigInferenceParams.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigInferenceParams = LLMObsCustomEvalConfigInferenceParams;
/**
 * @ignore
 */
LLMObsCustomEvalConfigInferenceParams.attributeTypeMap = {
    frequencyPenalty: {
        baseName: "frequency_penalty",
        type: "number",
        format: "double",
    },
    maxTokens: {
        baseName: "max_tokens",
        type: "number",
        format: "int64",
    },
    presencePenalty: {
        baseName: "presence_penalty",
        type: "number",
        format: "double",
    },
    temperature: {
        baseName: "temperature",
        type: "number",
        format: "double",
    },
    topK: {
        baseName: "top_k",
        type: "number",
        format: "int64",
    },
    topP: {
        baseName: "top_p",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
