// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest = void 0;
/**
 * Attributes for deleting interactions from an annotation queue.
 */
class LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest = LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest;
/**
 * @ignore
 */
LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest.attributeTypeMap = {
    interactionIds: {
        baseName: "interaction_ids",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
