// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueInteractionsDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueInteractionsDataResponse = void 0;
/**
 * Data object for the interaction addition response.
 */
class LLMObsAnnotationQueueInteractionsDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueInteractionsDataResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueInteractionsDataResponse = LLMObsAnnotationQueueInteractionsDataResponse;
/**
 * @ignore
 */
LLMObsAnnotationQueueInteractionsDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsAnnotationQueueInteractionsDataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotationQueueInteractionsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
