// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueDataResponse = void 0;
/**
 * Data object for an LLM Observability annotation queue.
 */
class LLMObsAnnotationQueueDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueDataResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueDataResponse = LLMObsAnnotationQueueDataResponse;
/**
 * @ignore
 */
LLMObsAnnotationQueueDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsAnnotationQueueDataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotationQueueType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
