// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigBedrockOptions.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigBedrockOptions = void 0;
/**
 * AWS Bedrock-specific options for LLM provider configuration.
 */
class LLMObsCustomEvalConfigBedrockOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigBedrockOptions.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigBedrockOptions = LLMObsCustomEvalConfigBedrockOptions;
/**
 * @ignore
 */
LLMObsCustomEvalConfigBedrockOptions.attributeTypeMap = {
    region: {
        baseName: "region",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
