// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotatedInteractionItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotatedInteractionItem = void 0;
/**
 * An interaction with its associated annotations.
 */
class LLMObsAnnotatedInteractionItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotatedInteractionItem.attributeTypeMap;
    }
}
exports.LLMObsAnnotatedInteractionItem = LLMObsAnnotatedInteractionItem;
/**
 * @ignore
 */
LLMObsAnnotatedInteractionItem.attributeTypeMap = {
    annotations: {
        baseName: "annotations",
        type: "Array<LLMObsAnnotationItem>",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
