// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueInteractionResponseItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueInteractionResponseItem = void 0;
/**
 * A single interaction result.
 */
class LLMObsAnnotationQueueInteractionResponseItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueInteractionResponseItem.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueInteractionResponseItem = LLMObsAnnotationQueueInteractionResponseItem;
/**
 * @ignore
 */
LLMObsAnnotationQueueInteractionResponseItem.attributeTypeMap = {
    alreadyExisted: {
        baseName: "already_existed",
        type: "boolean",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
