// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigUpdateData = void 0;
/**
 * Data object for creating or updating a custom LLM Observability evaluator configuration.
 */
class LLMObsCustomEvalConfigUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigUpdateData.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigUpdateData = LLMObsCustomEvalConfigUpdateData;
/**
 * @ignore
 */
LLMObsCustomEvalConfigUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsCustomEvalConfigUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "LLMObsCustomEvalConfigType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
