// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigPromptToolResult.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigPromptToolResult = void 0;
/**
 * A tool call result within a prompt message.
 */
class LLMObsCustomEvalConfigPromptToolResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigPromptToolResult.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigPromptToolResult = LLMObsCustomEvalConfigPromptToolResult;
/**
 * @ignore
 */
LLMObsCustomEvalConfigPromptToolResult.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    result: {
        baseName: "result",
        type: "string",
    },
    toolId: {
        baseName: "tool_id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
