// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotatedInteractionsDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotatedInteractionsDataResponse = void 0;
/**
 * Data object for annotated interactions.
 */
class LLMObsAnnotatedInteractionsDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotatedInteractionsDataResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotatedInteractionsDataResponse = LLMObsAnnotatedInteractionsDataResponse;
/**
 * @ignore
 */
LLMObsAnnotatedInteractionsDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsAnnotatedInteractionsDataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotatedInteractionsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
