// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigLLMJudgeConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigLLMJudgeConfig = void 0;
/**
 * LLM judge configuration for a custom evaluator.
 */
class LLMObsCustomEvalConfigLLMJudgeConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigLLMJudgeConfig.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigLLMJudgeConfig = LLMObsCustomEvalConfigLLMJudgeConfig;
/**
 * @ignore
 */
LLMObsCustomEvalConfigLLMJudgeConfig.attributeTypeMap = {
    assessmentCriteria: {
        baseName: "assessment_criteria",
        type: "LLMObsCustomEvalConfigAssessmentCriteria",
    },
    inferenceParams: {
        baseName: "inference_params",
        type: "LLMObsCustomEvalConfigInferenceParams",
        required: true,
    },
    lastUsedLibraryPromptTemplateName: {
        baseName: "last_used_library_prompt_template_name",
        type: "string",
    },
    modifiedLibraryPromptTemplate: {
        baseName: "modified_library_prompt_template",
        type: "boolean",
    },
    outputSchema: {
        baseName: "output_schema",
        type: "{ [key: string]: any; }",
    },
    parsingType: {
        baseName: "parsing_type",
        type: "LLMObsCustomEvalConfigParsingType",
    },
    promptTemplate: {
        baseName: "prompt_template",
        type: "Array<LLMObsCustomEvalConfigPromptMessage>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
