// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigAttributes = void 0;
/**
 * Attributes of a custom LLM Observability evaluator configuration.
 */
class LLMObsCustomEvalConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigAttributes.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigAttributes = LLMObsCustomEvalConfigAttributes;
/**
 * @ignore
 */
LLMObsCustomEvalConfigAttributes.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "LLMObsCustomEvalConfigUser",
    },
    evalName: {
        baseName: "eval_name",
        type: "string",
        required: true,
    },
    lastUpdatedBy: {
        baseName: "last_updated_by",
        type: "LLMObsCustomEvalConfigUser",
    },
    llmJudgeConfig: {
        baseName: "llm_judge_config",
        type: "LLMObsCustomEvalConfigLLMJudgeConfig",
    },
    llmProvider: {
        baseName: "llm_provider",
        type: "LLMObsCustomEvalConfigLLMProvider",
    },
    target: {
        baseName: "target",
        type: "LLMObsCustomEvalConfigTarget",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
