// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueInteractionsDataRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueInteractionsDataRequest = void 0;
/**
 * Data object for adding interactions to an annotation queue.
 */
class LLMObsAnnotationQueueInteractionsDataRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueInteractionsDataRequest.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueInteractionsDataRequest = LLMObsAnnotationQueueInteractionsDataRequest;
/**
 * @ignore
 */
LLMObsAnnotationQueueInteractionsDataRequest.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsAnnotationQueueInteractionsDataAttributesRequest",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotationQueueInteractionsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
