// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigPromptToolCall.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigPromptToolCall = void 0;
/**
 * A tool call within a prompt message.
 */
class LLMObsCustomEvalConfigPromptToolCall {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigPromptToolCall.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigPromptToolCall = LLMObsCustomEvalConfigPromptToolCall;
/**
 * @ignore
 */
LLMObsCustomEvalConfigPromptToolCall.attributeTypeMap = {
    arguments: {
        baseName: "arguments",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
