// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDeleteAnnotationQueueInteractionsDataRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDeleteAnnotationQueueInteractionsDataRequest = void 0;
/**
 * Data object for deleting interactions from an annotation queue.
 */
class LLMObsDeleteAnnotationQueueInteractionsDataRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDeleteAnnotationQueueInteractionsDataRequest.attributeTypeMap;
    }
}
exports.LLMObsDeleteAnnotationQueueInteractionsDataRequest = LLMObsDeleteAnnotationQueueInteractionsDataRequest;
/**
 * @ignore
 */
LLMObsDeleteAnnotationQueueInteractionsDataRequest.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsDeleteAnnotationQueueInteractionsDataAttributesRequest",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotationQueueInteractionsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
