// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigUpdateAttributes = void 0;
/**
 * Attributes for creating or updating a custom LLM Observability evaluator configuration.
 */
class LLMObsCustomEvalConfigUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigUpdateAttributes.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigUpdateAttributes = LLMObsCustomEvalConfigUpdateAttributes;
/**
 * @ignore
 */
LLMObsCustomEvalConfigUpdateAttributes.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "string",
    },
    evalName: {
        baseName: "eval_name",
        type: "string",
    },
    llmJudgeConfig: {
        baseName: "llm_judge_config",
        type: "LLMObsCustomEvalConfigLLMJudgeConfig",
    },
    llmProvider: {
        baseName: "llm_provider",
        type: "LLMObsCustomEvalConfigLLMProvider",
    },
    target: {
        baseName: "target",
        type: "LLMObsCustomEvalConfigTarget",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
