// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigPromptContentValue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigPromptContentValue = void 0;
/**
 * Value of a prompt message content block.
 */
class LLMObsCustomEvalConfigPromptContentValue {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigPromptContentValue.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigPromptContentValue = LLMObsCustomEvalConfigPromptContentValue;
/**
 * @ignore
 */
LLMObsCustomEvalConfigPromptContentValue.attributeTypeMap = {
    text: {
        baseName: "text",
        type: "string",
    },
    toolCall: {
        baseName: "tool_call",
        type: "LLMObsCustomEvalConfigPromptToolCall",
    },
    toolCallResult: {
        baseName: "tool_call_result",
        type: "LLMObsCustomEvalConfigPromptToolResult",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
