// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsCustomEvalConfigLLMProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsCustomEvalConfigLLMProvider = void 0;
/**
 * LLM provider configuration for a custom evaluator.
 */
class LLMObsCustomEvalConfigLLMProvider {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsCustomEvalConfigLLMProvider.attributeTypeMap;
    }
}
exports.LLMObsCustomEvalConfigLLMProvider = LLMObsCustomEvalConfigLLMProvider;
/**
 * @ignore
 */
LLMObsCustomEvalConfigLLMProvider.attributeTypeMap = {
    bedrock: {
        baseName: "bedrock",
        type: "LLMObsCustomEvalConfigBedrockOptions",
    },
    integrationAccountId: {
        baseName: "integration_account_id",
        type: "string",
    },
    integrationProvider: {
        baseName: "integration_provider",
        type: "LLMObsCustomEvalConfigIntegrationProvider",
    },
    modelName: {
        baseName: "model_name",
        type: "string",
    },
    vertexAi: {
        baseName: "vertex_ai",
        type: "LLMObsCustomEvalConfigVertexAIOptions",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
