// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MaintenanceDataRelationshipsLastModifiedByUserData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceDataRelationshipsLastModifiedByUserData = void 0;
/**
 * The data object identifying the Datadog user who last modified the maintenance.
 */
class MaintenanceDataRelationshipsLastModifiedByUserData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MaintenanceDataRelationshipsLastModifiedByUserData.attributeTypeMap;
    }
}
exports.MaintenanceDataRelationshipsLastModifiedByUserData = MaintenanceDataRelationshipsLastModifiedByUserData;
/**
 * @ignore
 */
MaintenanceDataRelationshipsLastModifiedByUserData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "StatusPagesUserType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
