// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MicrosoftTeamsWorkflowsWebhookHandleResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MicrosoftTeamsWorkflowsWebhookHandleResponseData = void 0;
/**
 * Workflows Webhook handle data from a response.
 */
class MicrosoftTeamsWorkflowsWebhookHandleResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MicrosoftTeamsWorkflowsWebhookHandleResponseData.attributeTypeMap;
    }
}
exports.MicrosoftTeamsWorkflowsWebhookHandleResponseData = MicrosoftTeamsWorkflowsWebhookHandleResponseData;
/**
 * @ignore
 */
MicrosoftTeamsWorkflowsWebhookHandleResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "MicrosoftTeamsWorkflowsWebhookResponseAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "MicrosoftTeamsWorkflowsWebhookHandleType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
