// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MonitorAlertTriggerAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorAlertTriggerAttributes = void 0;
/**
 * Attributes for a monitor alert trigger.
 */
class MonitorAlertTriggerAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorAlertTriggerAttributes.attributeTypeMap;
    }
}
exports.MonitorAlertTriggerAttributes = MonitorAlertTriggerAttributes;
/**
 * @ignore
 */
MonitorAlertTriggerAttributes.attributeTypeMap = {
    eventId: {
        baseName: "event_id",
        type: "string",
        required: true,
    },
    eventTs: {
        baseName: "event_ts",
        type: "number",
        required: true,
        format: "int64",
    },
    monitorId: {
        baseName: "monitor_id",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
