// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineAmazonS3GenericCompressionZstd.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineAmazonS3GenericCompressionZstd = void 0;
/**
 * Zstd compression.
 */
class ObservabilityPipelineAmazonS3GenericCompressionZstd {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineAmazonS3GenericCompressionZstd.attributeTypeMap;
    }
}
exports.ObservabilityPipelineAmazonS3GenericCompressionZstd = ObservabilityPipelineAmazonS3GenericCompressionZstd;
/**
 * @ignore
 */
ObservabilityPipelineAmazonS3GenericCompressionZstd.attributeTypeMap = {
    algorithm: {
        baseName: "algorithm",
        type: "ObservabilityPipelineAmazonS3GenericCompressionZstdType",
        required: true,
    },
    level: {
        baseName: "level",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
