// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineOcsfMappingLibrary.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PALO_ALTO_NETWORKS_FIREWALL_TRAFFIC = exports.OKTA_SYSTEM_LOG_AUTHENTICATION = exports.MICROSOFT_365_DEFENDER_USERLOGGEDIN = exports.MICROSOFT_365_DEFENDER_INCIDENT = exports.GOOGLE_WORKSPACE_ADMIN_AUDIT_ADDPRIVILEGE = exports.GITHUB_AUDIT_LOG_API_ACTIVITY = exports.GCP_CLOUD_AUDIT_UPDATESINK = exports.GCP_CLOUD_AUDIT_SETIAMPOLICY = exports.GCP_CLOUD_AUDIT_CREATESINK = exports.GCP_CLOUD_AUDIT_CREATEBUCKET = exports.CLOUDTRAIL_ACCOUNT_CHANGE = void 0;
exports.CLOUDTRAIL_ACCOUNT_CHANGE = "CloudTrail Account Change";
exports.GCP_CLOUD_AUDIT_CREATEBUCKET = "GCP Cloud Audit CreateBucket";
exports.GCP_CLOUD_AUDIT_CREATESINK = "GCP Cloud Audit CreateSink";
exports.GCP_CLOUD_AUDIT_SETIAMPOLICY = "GCP Cloud Audit SetIamPolicy";
exports.GCP_CLOUD_AUDIT_UPDATESINK = "GCP Cloud Audit UpdateSink";
exports.GITHUB_AUDIT_LOG_API_ACTIVITY = "Github Audit Log API Activity";
exports.GOOGLE_WORKSPACE_ADMIN_AUDIT_ADDPRIVILEGE = "Google Workspace Admin Audit addPrivilege";
exports.MICROSOFT_365_DEFENDER_INCIDENT = "Microsoft 365 Defender Incident";
exports.MICROSOFT_365_DEFENDER_USERLOGGEDIN = "Microsoft 365 Defender UserLoggedIn";
exports.OKTA_SYSTEM_LOG_AUTHENTICATION = "Okta System Log Authentication";
exports.PALO_ALTO_NETWORKS_FIREWALL_TRAFFIC = "Palo Alto Networks Firewall Traffic";
