// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineEnrichmentTableFieldVrlLookup.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineEnrichmentTableFieldVrlLookup = void 0;
/**
 * Evaluates a VRL expression to produce the lookup key.
 */
class ObservabilityPipelineEnrichmentTableFieldVrlLookup {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineEnrichmentTableFieldVrlLookup.attributeTypeMap;
    }
}
exports.ObservabilityPipelineEnrichmentTableFieldVrlLookup = ObservabilityPipelineEnrichmentTableFieldVrlLookup;
/**
 * @ignore
 */
ObservabilityPipelineEnrichmentTableFieldVrlLookup.attributeTypeMap = {
    vrl: {
        baseName: "vrl",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
