// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineEnrichmentTableFieldEventLookup.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineEnrichmentTableFieldEventLookup = void 0;
/**
 * Looks up a value from a field path in the log event.
 */
class ObservabilityPipelineEnrichmentTableFieldEventLookup {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineEnrichmentTableFieldEventLookup.attributeTypeMap;
    }
}
exports.ObservabilityPipelineEnrichmentTableFieldEventLookup = ObservabilityPipelineEnrichmentTableFieldEventLookup;
/**
 * @ignore
 */
ObservabilityPipelineEnrichmentTableFieldEventLookup.attributeTypeMap = {
    event: {
        baseName: "event",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
