// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineAmazonS3GenericDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineAmazonS3GenericDestination = void 0;
/**
 * The `amazon_s3_generic` destination sends your logs to an Amazon S3 bucket.
 *
 * **Supported pipeline types:** logs
 */
class ObservabilityPipelineAmazonS3GenericDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineAmazonS3GenericDestination.attributeTypeMap;
    }
}
exports.ObservabilityPipelineAmazonS3GenericDestination = ObservabilityPipelineAmazonS3GenericDestination;
/**
 * @ignore
 */
ObservabilityPipelineAmazonS3GenericDestination.attributeTypeMap = {
    auth: {
        baseName: "auth",
        type: "ObservabilityPipelineAwsAuth",
    },
    batchSettings: {
        baseName: "batch_settings",
        type: "ObservabilityPipelineAmazonS3GenericBatchSettings",
    },
    bucket: {
        baseName: "bucket",
        type: "string",
        required: true,
    },
    compression: {
        baseName: "compression",
        type: "ObservabilityPipelineAmazonS3GenericCompression",
        required: true,
    },
    encoding: {
        baseName: "encoding",
        type: "ObservabilityPipelineAmazonS3GenericEncoding",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    keyPrefix: {
        baseName: "key_prefix",
        type: "string",
    },
    region: {
        baseName: "region",
        type: "string",
        required: true,
    },
    storageClass: {
        baseName: "storage_class",
        type: "ObservabilityPipelineAmazonS3DestinationStorageClass",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineAmazonS3GenericDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
