// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineElasticsearchDestinationCompression.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineElasticsearchDestinationCompression = void 0;
/**
 * Compression configuration for the Elasticsearch destination.
 */
class ObservabilityPipelineElasticsearchDestinationCompression {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineElasticsearchDestinationCompression.attributeTypeMap;
    }
}
exports.ObservabilityPipelineElasticsearchDestinationCompression = ObservabilityPipelineElasticsearchDestinationCompression;
/**
 * @ignore
 */
ObservabilityPipelineElasticsearchDestinationCompression.attributeTypeMap = {
    algorithm: {
        baseName: "algorithm",
        type: "ObservabilityPipelineElasticsearchDestinationCompressionAlgorithm",
        required: true,
    },
    level: {
        baseName: "level",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
