// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineEnrichmentTableFieldSecretLookup.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineEnrichmentTableFieldSecretLookup = void 0;
/**
 * Looks up a value stored as a pipeline secret.
 */
class ObservabilityPipelineEnrichmentTableFieldSecretLookup {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineEnrichmentTableFieldSecretLookup.attributeTypeMap;
    }
}
exports.ObservabilityPipelineEnrichmentTableFieldSecretLookup = ObservabilityPipelineEnrichmentTableFieldSecretLookup;
/**
 * @ignore
 */
ObservabilityPipelineEnrichmentTableFieldSecretLookup.attributeTypeMap = {
    secret: {
        baseName: "secret",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
