// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineAmazonS3GenericBatchSettings.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineAmazonS3GenericBatchSettings = void 0;
/**
 * Event batching settings
 */
class ObservabilityPipelineAmazonS3GenericBatchSettings {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineAmazonS3GenericBatchSettings.attributeTypeMap;
    }
}
exports.ObservabilityPipelineAmazonS3GenericBatchSettings = ObservabilityPipelineAmazonS3GenericBatchSettings;
/**
 * @ignore
 */
ObservabilityPipelineAmazonS3GenericBatchSettings.attributeTypeMap = {
    batchSize: {
        baseName: "batch_size",
        type: "number",
        format: "int64",
    },
    timeoutSecs: {
        baseName: "timeout_secs",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
