// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineAmazonS3GenericCompressionGzip.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineAmazonS3GenericCompressionGzip = void 0;
/**
 * Gzip compression.
 */
class ObservabilityPipelineAmazonS3GenericCompressionGzip {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineAmazonS3GenericCompressionGzip.attributeTypeMap;
    }
}
exports.ObservabilityPipelineAmazonS3GenericCompressionGzip = ObservabilityPipelineAmazonS3GenericCompressionGzip;
/**
 * @ignore
 */
ObservabilityPipelineAmazonS3GenericCompressionGzip.attributeTypeMap = {
    algorithm: {
        baseName: "algorithm",
        type: "ObservabilityPipelineAmazonS3GenericCompressionGzipType",
        required: true,
    },
    level: {
        baseName: "level",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
