// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyAttributes = void 0;
/**
 * Attributes of an org group policy.
 */
class OrgGroupPolicyAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyAttributes.attributeTypeMap;
    }
}
exports.OrgGroupPolicyAttributes = OrgGroupPolicyAttributes;
/**
 * @ignore
 */
OrgGroupPolicyAttributes.attributeTypeMap = {
    content: {
        baseName: "content",
        type: "{ [key: string]: any; }",
    },
    enforcementTier: {
        baseName: "enforcement_tier",
        type: "OrgGroupPolicyEnforcementTier",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    policyName: {
        baseName: "policy_name",
        type: "string",
        required: true,
    },
    policyType: {
        baseName: "policy_type",
        type: "OrgGroupPolicyPolicyType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
