// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupMembershipBulkUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupMembershipBulkUpdateData = void 0;
/**
 * Data for bulk updating org group memberships.
 */
class OrgGroupMembershipBulkUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupMembershipBulkUpdateData.attributeTypeMap;
    }
}
exports.OrgGroupMembershipBulkUpdateData = OrgGroupMembershipBulkUpdateData;
/**
 * @ignore
 */
OrgGroupMembershipBulkUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OrgGroupMembershipBulkUpdateAttributes",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "OrgGroupMembershipBulkUpdateRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "OrgGroupMembershipBulkUpdateType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
