// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupMembershipData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupMembershipData = void 0;
/**
 * An org group membership resource.
 */
class OrgGroupMembershipData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupMembershipData.attributeTypeMap;
    }
}
exports.OrgGroupMembershipData = OrgGroupMembershipData;
/**
 * @ignore
 */
OrgGroupMembershipData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OrgGroupMembershipAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    relationships: {
        baseName: "relationships",
        type: "OrgGroupMembershipRelationships",
    },
    type: {
        baseName: "type",
        type: "OrgGroupMembershipType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
