// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyOverrideUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyOverrideUpdateData = void 0;
/**
 * Data for updating a policy override.
 */
class OrgGroupPolicyOverrideUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyOverrideUpdateData.attributeTypeMap;
    }
}
exports.OrgGroupPolicyOverrideUpdateData = OrgGroupPolicyOverrideUpdateData;
/**
 * @ignore
 */
OrgGroupPolicyOverrideUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OrgGroupPolicyOverrideUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "OrgGroupPolicyOverrideType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
