// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyOverrideAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyOverrideAttributes = void 0;
/**
 * Attributes of an org group policy override.
 */
class OrgGroupPolicyOverrideAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyOverrideAttributes.attributeTypeMap;
    }
}
exports.OrgGroupPolicyOverrideAttributes = OrgGroupPolicyOverrideAttributes;
/**
 * @ignore
 */
OrgGroupPolicyOverrideAttributes.attributeTypeMap = {
    content: {
        baseName: "content",
        type: "{ [key: string]: any; }",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    orgSite: {
        baseName: "org_site",
        type: "string",
        required: true,
    },
    orgUuid: {
        baseName: "org_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
