// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyConfigAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyConfigAttributes = void 0;
/**
 * Attributes of an org group policy config.
 */
class OrgGroupPolicyConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyConfigAttributes.attributeTypeMap;
    }
}
exports.OrgGroupPolicyConfigAttributes = OrgGroupPolicyConfigAttributes;
/**
 * @ignore
 */
OrgGroupPolicyConfigAttributes.attributeTypeMap = {
    allowedValues: {
        baseName: "allowed_values",
        type: "Array<string>",
        required: true,
    },
    defaultValue: {
        baseName: "default_value",
        type: "any",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    valueType: {
        baseName: "value_type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
