// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyOverrideCreateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyOverrideCreateData = void 0;
/**
 * Data for creating an org group policy override.
 */
class OrgGroupPolicyOverrideCreateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyOverrideCreateData.attributeTypeMap;
    }
}
exports.OrgGroupPolicyOverrideCreateData = OrgGroupPolicyOverrideCreateData;
/**
 * @ignore
 */
OrgGroupPolicyOverrideCreateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OrgGroupPolicyOverrideCreateAttributes",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "OrgGroupPolicyOverrideCreateRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "OrgGroupPolicyOverrideType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
