// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyOverrideUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyOverrideUpdateAttributes = void 0;
/**
 * Attributes for updating a policy override. The `org_uuid` and `org_site` fields must match the existing override and cannot be changed.
 */
class OrgGroupPolicyOverrideUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyOverrideUpdateAttributes.attributeTypeMap;
    }
}
exports.OrgGroupPolicyOverrideUpdateAttributes = OrgGroupPolicyOverrideUpdateAttributes;
/**
 * @ignore
 */
OrgGroupPolicyOverrideUpdateAttributes.attributeTypeMap = {
    orgSite: {
        baseName: "org_site",
        type: "string",
        required: true,
    },
    orgUuid: {
        baseName: "org_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
