// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyOverrideData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyOverrideData = void 0;
/**
 * An org group policy override resource.
 */
class OrgGroupPolicyOverrideData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyOverrideData.attributeTypeMap;
    }
}
exports.OrgGroupPolicyOverrideData = OrgGroupPolicyOverrideData;
/**
 * @ignore
 */
OrgGroupPolicyOverrideData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OrgGroupPolicyOverrideAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    relationships: {
        baseName: "relationships",
        type: "OrgGroupPolicyOverrideRelationships",
    },
    type: {
        baseName: "type",
        type: "OrgGroupPolicyOverrideType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
