// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyData = void 0;
/**
 * An org group policy resource.
 */
class OrgGroupPolicyData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyData.attributeTypeMap;
    }
}
exports.OrgGroupPolicyData = OrgGroupPolicyData;
/**
 * @ignore
 */
OrgGroupPolicyData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OrgGroupPolicyAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    relationships: {
        baseName: "relationships",
        type: "OrgGroupPolicyRelationships",
    },
    type: {
        baseName: "type",
        type: "OrgGroupPolicyType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
