// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyCreateAttributes = void 0;
/**
 * Attributes for creating an org group policy. If `policy_type` or `enforcement_tier` are not provided, they default to `org_config` and `DEFAULT` respectively.
 */
class OrgGroupPolicyCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyCreateAttributes.attributeTypeMap;
    }
}
exports.OrgGroupPolicyCreateAttributes = OrgGroupPolicyCreateAttributes;
/**
 * @ignore
 */
OrgGroupPolicyCreateAttributes.attributeTypeMap = {
    content: {
        baseName: "content",
        type: "{ [key: string]: any; }",
        required: true,
    },
    enforcementTier: {
        baseName: "enforcement_tier",
        type: "OrgGroupPolicyEnforcementTier",
    },
    policyName: {
        baseName: "policy_name",
        type: "string",
        required: true,
    },
    policyType: {
        baseName: "policy_type",
        type: "OrgGroupPolicyPolicyType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
