// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupPolicyOverrideCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupPolicyOverrideCreateAttributes = void 0;
/**
 * Attributes for creating a policy override.
 */
class OrgGroupPolicyOverrideCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupPolicyOverrideCreateAttributes.attributeTypeMap;
    }
}
exports.OrgGroupPolicyOverrideCreateAttributes = OrgGroupPolicyOverrideCreateAttributes;
/**
 * @ignore
 */
OrgGroupPolicyOverrideCreateAttributes.attributeTypeMap = {
    orgSite: {
        baseName: "org_site",
        type: "string",
        required: true,
    },
    orgUuid: {
        baseName: "org_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
