// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgGroupAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgGroupAttributes = void 0;
/**
 * Attributes of an org group.
 */
class OrgGroupAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgGroupAttributes.attributeTypeMap;
    }
}
exports.OrgGroupAttributes = OrgGroupAttributes;
/**
 * @ignore
 */
OrgGroupAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ownerOrgSite: {
        baseName: "owner_org_site",
        type: "string",
        required: true,
    },
    ownerOrgUuid: {
        baseName: "owner_org_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
