// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/PaginatedResponseMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaginatedResponseMeta = void 0;
/**
 * Metadata for scores response.
 */
class PaginatedResponseMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PaginatedResponseMeta.attributeTypeMap;
    }
}
exports.PaginatedResponseMeta = PaginatedResponseMeta;
/**
 * @ignore
 */
PaginatedResponseMeta.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    limit: {
        baseName: "limit",
        type: "number",
        required: true,
        format: "int64",
    },
    offset: {
        baseName: "offset",
        type: "number",
        required: true,
        format: "int64",
    },
    total: {
        baseName: "total",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
