// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/PatchDegradationRequestDataAttributesComponentsAffectedItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchDegradationRequestDataAttributesComponentsAffectedItems = void 0;
/**
 * A component affected by a degradation.
 */
class PatchDegradationRequestDataAttributesComponentsAffectedItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PatchDegradationRequestDataAttributesComponentsAffectedItems.attributeTypeMap;
    }
}
exports.PatchDegradationRequestDataAttributesComponentsAffectedItems = PatchDegradationRequestDataAttributesComponentsAffectedItems;
/**
 * @ignore
 */
PatchDegradationRequestDataAttributesComponentsAffectedItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "StatusPagesComponentDataAttributesStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
