// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/PersonalAccessTokenCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PersonalAccessTokenCreateAttributes = void 0;
/**
 * Attributes used to create a personal access token.
 */
class PersonalAccessTokenCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PersonalAccessTokenCreateAttributes.attributeTypeMap;
    }
}
exports.PersonalAccessTokenCreateAttributes = PersonalAccessTokenCreateAttributes;
/**
 * @ignore
 */
PersonalAccessTokenCreateAttributes.attributeTypeMap = {
    expiresAt: {
        baseName: "expires_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    scopes: {
        baseName: "scopes",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
