// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/PostmortemTemplateAttributesResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostmortemTemplateAttributesResponse = void 0;
/**
 * Attributes of a postmortem template returned in a response.
 */
class PostmortemTemplateAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PostmortemTemplateAttributesResponse.attributeTypeMap;
    }
}
exports.PostmortemTemplateAttributesResponse = PostmortemTemplateAttributesResponse;
/**
 * @ignore
 */
PostmortemTemplateAttributesResponse.attributeTypeMap = {
    createdAt: {
        baseName: "createdAt",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modifiedAt",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
