// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/PreviewEntityResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreviewEntityResponseData = void 0;
/**
 * Entity data returned in a preview response, including attributes, relationships, and type.
 */
class PreviewEntityResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PreviewEntityResponseData.attributeTypeMap;
    }
}
exports.PreviewEntityResponseData = PreviewEntityResponseData;
/**
 * @ignore
 */
PreviewEntityResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "EntityResponseDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    relationships: {
        baseName: "relationships",
        type: "EntityResponseDataRelationships",
    },
    type: {
        baseName: "type",
        type: "EntityResponseDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
