// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ProductAnalyticsServerSideEventItemEvent.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsServerSideEventItemEvent = void 0;
/**
 * Fields used for the event.
 */
class ProductAnalyticsServerSideEventItemEvent {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsServerSideEventItemEvent.attributeTypeMap;
    }
}
exports.ProductAnalyticsServerSideEventItemEvent = ProductAnalyticsServerSideEventItemEvent;
/**
 * @ignore
 */
ProductAnalyticsServerSideEventItemEvent.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
