// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ProductAnalyticsServerSideEventItemAccount.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductAnalyticsServerSideEventItemAccount = void 0;
/**
 * The account linked to your event.
 */
class ProductAnalyticsServerSideEventItemAccount {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ProductAnalyticsServerSideEventItemAccount.attributeTypeMap;
    }
}
exports.ProductAnalyticsServerSideEventItemAccount = ProductAnalyticsServerSideEventItemAccount;
/**
 * @ignore
 */
ProductAnalyticsServerSideEventItemAccount.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
