// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/QueryEventFilteredUsersRequestDataAttributesEventQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryEventFilteredUsersRequestDataAttributesEventQuery = void 0;
/**
 * Event platform query used to filter users based on their event activity within a specified time window.
 */
class QueryEventFilteredUsersRequestDataAttributesEventQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return QueryEventFilteredUsersRequestDataAttributesEventQuery.attributeTypeMap;
    }
}
exports.QueryEventFilteredUsersRequestDataAttributesEventQuery = QueryEventFilteredUsersRequestDataAttributesEventQuery;
/**
 * @ignore
 */
QueryEventFilteredUsersRequestDataAttributesEventQuery.attributeTypeMap = {
    query: {
        baseName: "query",
        type: "string",
    },
    timeFrame: {
        baseName: "time_frame",
        type: "QueryEventFilteredUsersRequestDataAttributesEventQueryTimeFrame",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
