// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/QueryEventFilteredUsersRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryEventFilteredUsersRequestData = void 0;
/**
 * The data object containing the resource type and attributes for querying event-filtered users.
 */
class QueryEventFilteredUsersRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return QueryEventFilteredUsersRequestData.attributeTypeMap;
    }
}
exports.QueryEventFilteredUsersRequestData = QueryEventFilteredUsersRequestData;
/**
 * @ignore
 */
QueryEventFilteredUsersRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "QueryEventFilteredUsersRequestDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "QueryEventFilteredUsersRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
