// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/QueryEventFilteredUsersRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryEventFilteredUsersRequestDataAttributes = void 0;
/**
 * Attributes for filtering users by both user properties and event platform activity.
 */
class QueryEventFilteredUsersRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return QueryEventFilteredUsersRequestDataAttributes.attributeTypeMap;
    }
}
exports.QueryEventFilteredUsersRequestDataAttributes = QueryEventFilteredUsersRequestDataAttributes;
/**
 * @ignore
 */
QueryEventFilteredUsersRequestDataAttributes.attributeTypeMap = {
    eventQuery: {
        baseName: "event_query",
        type: "QueryEventFilteredUsersRequestDataAttributesEventQuery",
    },
    includeRowCount: {
        baseName: "include_row_count",
        type: "boolean",
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    selectColumns: {
        baseName: "select_columns",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
