// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/QueryUsersRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryUsersRequestDataAttributes = void 0;
/**
 * Attributes for filtering and shaping the user query results.
 */
class QueryUsersRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return QueryUsersRequestDataAttributes.attributeTypeMap;
    }
}
exports.QueryUsersRequestDataAttributes = QueryUsersRequestDataAttributes;
/**
 * @ignore
 */
QueryUsersRequestDataAttributes.attributeTypeMap = {
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    selectColumns: {
        baseName: "select_columns",
        type: "Array<string>",
    },
    sort: {
        baseName: "sort",
        type: "QueryUsersRequestDataAttributesSort",
    },
    wildcardSearchTerm: {
        baseName: "wildcard_search_term",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
