// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RelationType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RELATIONTYPEIMPLEMENTS = exports.RELATIONTYPEIMPLEMENTEDBY = exports.RELATIONTYPEOTHEROWNEDBY = exports.RELATIONTYPEOTHEROWNS = exports.RELATIONTYPEHASPART = exports.RELATIONTYPEPARTSOF = exports.RELATIONTYPEDEPENDENCYOF = exports.RELATIONTYPEDEPENDSON = exports.RELATIONTYPEOWNEDBY = exports.RELATIONTYPEOWNS = void 0;
exports.RELATIONTYPEOWNS = "RelationTypeOwns";
exports.RELATIONTYPEOWNEDBY = "RelationTypeOwnedBy";
exports.RELATIONTYPEDEPENDSON = "RelationTypeDependsOn";
exports.RELATIONTYPEDEPENDENCYOF = "RelationTypeDependencyOf";
exports.RELATIONTYPEPARTSOF = "RelationTypePartsOf";
exports.RELATIONTYPEHASPART = "RelationTypeHasPart";
exports.RELATIONTYPEOTHEROWNS = "RelationTypeOtherOwns";
exports.RELATIONTYPEOTHEROWNEDBY = "RelationTypeOtherOwnedBy";
exports.RELATIONTYPEIMPLEMENTEDBY = "RelationTypeImplementedBy";
exports.RELATIONTYPEIMPLEMENTS = "RelationTypeImplements";
