// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ResolveVulnerableSymbolsRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolveVulnerableSymbolsRequestData = void 0;
/**
 * The data object in a request to resolve vulnerable symbols, containing the package PURLs and request type.
 */
class ResolveVulnerableSymbolsRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ResolveVulnerableSymbolsRequestData.attributeTypeMap;
    }
}
exports.ResolveVulnerableSymbolsRequestData = ResolveVulnerableSymbolsRequestData;
/**
 * @ignore
 */
ResolveVulnerableSymbolsRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ResolveVulnerableSymbolsRequestDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "ResolveVulnerableSymbolsRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
