// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ResolveVulnerableSymbolsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolveVulnerableSymbolsResponseData = void 0;
/**
 * The data object in a response for resolving vulnerable symbols, containing the result attributes and response type.
 */
class ResolveVulnerableSymbolsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ResolveVulnerableSymbolsResponseData.attributeTypeMap;
    }
}
exports.ResolveVulnerableSymbolsResponseData = ResolveVulnerableSymbolsResponseData;
/**
 * @ignore
 */
ResolveVulnerableSymbolsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ResolveVulnerableSymbolsResponseDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "ResolveVulnerableSymbolsResponseDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
