// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols = void 0;
/**
 * A symbol identified as vulnerable within a dependency, including its name, type, and value.
 */
class ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols.attributeTypeMap;
    }
}
exports.ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols = ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols;
/**
 * @ignore
 */
ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
