// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ResolveVulnerableSymbolsResponseResultsVulnerableSymbols.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolveVulnerableSymbolsResponseResultsVulnerableSymbols = void 0;
/**
 * A collection of vulnerable symbols associated with a specific security advisory.
 */
class ResolveVulnerableSymbolsResponseResultsVulnerableSymbols {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ResolveVulnerableSymbolsResponseResultsVulnerableSymbols.attributeTypeMap;
    }
}
exports.ResolveVulnerableSymbolsResponseResultsVulnerableSymbols = ResolveVulnerableSymbolsResponseResultsVulnerableSymbols;
/**
 * @ignore
 */
ResolveVulnerableSymbolsResponseResultsVulnerableSymbols.attributeTypeMap = {
    advisoryId: {
        baseName: "advisory_id",
        type: "string",
    },
    symbols: {
        baseName: "symbols",
        type: "Array<ResolveVulnerableSymbolsResponseResultsVulnerableSymbolsSymbols>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
