// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RoleAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleAttributes = void 0;
/**
 * Attributes of the role.
 */
class RoleAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RoleAttributes.attributeTypeMap;
    }
}
exports.RoleAttributes = RoleAttributes;
/**
 * @ignore
 */
RoleAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    receivesPermissionsFrom: {
        baseName: "receives_permissions_from",
        type: "Array<string>",
    },
    userCount: {
        baseName: "user_count",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
