// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RulesValidateQueryRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RulesValidateQueryRequestDataAttributes = void 0;
/**
 * The definition of `RulesValidateQueryRequestDataAttributes` object.
 */
class RulesValidateQueryRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RulesValidateQueryRequestDataAttributes.attributeTypeMap;
    }
}
exports.RulesValidateQueryRequestDataAttributes = RulesValidateQueryRequestDataAttributes;
/**
 * @ignore
 */
RulesValidateQueryRequestDataAttributes.attributeTypeMap = {
    query: {
        baseName: "Query",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
