// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RuleAttributesRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleAttributesRequest = void 0;
/**
 * Attributes for creating or updating a rule. Server-managed fields (created_at, modified_at, custom) are excluded.
 */
class RuleAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RuleAttributesRequest.attributeTypeMap;
    }
}
exports.RuleAttributesRequest = RuleAttributesRequest;
/**
 * @ignore
 */
RuleAttributesRequest.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    level: {
        baseName: "level",
        type: "number",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    owner: {
        baseName: "owner",
        type: "string",
    },
    scopeQuery: {
        baseName: "scope_query",
        type: "string",
    },
    scorecardName: {
        baseName: "scorecard_name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
