// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RuleTypesItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SECRET_VULNERABILITY = exports.SAST_VULNERABILITY = exports.IAC_MISCONFIGURATION = exports.HOST_VULNERABILITY = exports.API_SECURITY = exports.MISCONFIGURATION = exports.IDENTITY_RISK = exports.CONTAINER_IMAGE_VULNERABILITY = exports.ATTACK_PATH = exports.APPLICATION_LIBRARY_VULNERABILITY = exports.APPLICATION_CODE_VULNERABILITY = exports.INFRASTRUCTURE_CONFIGURATION = exports.CLOUD_CONFIGURATION = exports.SIGNAL_CORRELATION = exports.WORKLOAD_SECURITY = exports.LOG_DETECTION = exports.APPLICATION_SECURITY = void 0;
exports.APPLICATION_SECURITY = "application_security";
exports.LOG_DETECTION = "log_detection";
exports.WORKLOAD_SECURITY = "workload_security";
exports.SIGNAL_CORRELATION = "signal_correlation";
exports.CLOUD_CONFIGURATION = "cloud_configuration";
exports.INFRASTRUCTURE_CONFIGURATION = "infrastructure_configuration";
exports.APPLICATION_CODE_VULNERABILITY = "application_code_vulnerability";
exports.APPLICATION_LIBRARY_VULNERABILITY = "application_library_vulnerability";
exports.ATTACK_PATH = "attack_path";
exports.CONTAINER_IMAGE_VULNERABILITY = "container_image_vulnerability";
exports.IDENTITY_RISK = "identity_risk";
exports.MISCONFIGURATION = "misconfiguration";
exports.API_SECURITY = "api_security";
exports.HOST_VULNERABILITY = "host_vulnerability";
exports.IAC_MISCONFIGURATION = "iac_misconfiguration";
exports.SAST_VULNERABILITY = "sast_vulnerability";
exports.SECRET_VULNERABILITY = "secret_vulnerability";
