// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumCrossProductSamplingCreate.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumCrossProductSamplingCreate = void 0;
/**
 * The configuration for cross-product retention filters.
 */
class RumCrossProductSamplingCreate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumCrossProductSamplingCreate.attributeTypeMap;
    }
}
exports.RumCrossProductSamplingCreate = RumCrossProductSamplingCreate;
/**
 * @ignore
 */
RumCrossProductSamplingCreate.attributeTypeMap = {
    traceEnabled: {
        baseName: "trace_enabled",
        type: "boolean",
    },
    traceSampleRate: {
        baseName: "trace_sample_rate",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
