// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumCrossProductSamplingUpdate.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumCrossProductSamplingUpdate = void 0;
/**
 * The configuration for cross-product retention filters. All fields are optional for partial updates.
 */
class RumCrossProductSamplingUpdate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumCrossProductSamplingUpdate.attributeTypeMap;
    }
}
exports.RumCrossProductSamplingUpdate = RumCrossProductSamplingUpdate;
/**
 * @ignore
 */
RumCrossProductSamplingUpdate.attributeTypeMap = {
    traceEnabled: {
        baseName: "trace_enabled",
        type: "boolean",
    },
    traceSampleRate: {
        baseName: "trace_sample_rate",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
