// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumRetentionFilterUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumRetentionFilterUpdateAttributes = void 0;
/**
 * The object describing attributes of a RUM retention filter to update.
 */
class RumRetentionFilterUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumRetentionFilterUpdateAttributes.attributeTypeMap;
    }
}
exports.RumRetentionFilterUpdateAttributes = RumRetentionFilterUpdateAttributes;
/**
 * @ignore
 */
RumRetentionFilterUpdateAttributes.attributeTypeMap = {
    crossProductSampling: {
        baseName: "cross_product_sampling",
        type: "RumCrossProductSamplingUpdate",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    eventType: {
        baseName: "event_type",
        type: "RumRetentionFilterEventType",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    sampleRate: {
        baseName: "sample_rate",
        type: "number",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
