// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SAMLAssertionAttributeAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLAssertionAttributeAttributes = void 0;
/**
 * Key/Value pair of attributes used in SAML assertion attributes.
 */
class SAMLAssertionAttributeAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SAMLAssertionAttributeAttributes.attributeTypeMap;
    }
}
exports.SAMLAssertionAttributeAttributes = SAMLAssertionAttributeAttributes;
/**
 * @ignore
 */
SAMLAssertionAttributeAttributes.attributeTypeMap = {
    attributeKey: {
        baseName: "attribute_key",
        type: "string",
    },
    attributeValue: {
        baseName: "attribute_value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
