// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems = void 0;
/**
 * A key-value property describing a reachable symbol within a dependency.
 */
class ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems.attributeTypeMap;
    }
}
exports.ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems = ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems;
/**
 * @ignore
 */
ScaRequestDataAttributesDependenciesItemsReachableSymbolPropertiesItems.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
