// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScaRequestDataAttributesVulnerabilitiesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScaRequestDataAttributesVulnerabilitiesItems = void 0;
/**
 * A vulnerability entry from the Software Bill of Materials (SBOM), describing a known security issue and the components it affects.
 */
class ScaRequestDataAttributesVulnerabilitiesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScaRequestDataAttributesVulnerabilitiesItems.attributeTypeMap;
    }
}
exports.ScaRequestDataAttributesVulnerabilitiesItems = ScaRequestDataAttributesVulnerabilitiesItems;
/**
 * @ignore
 */
ScaRequestDataAttributesVulnerabilitiesItems.attributeTypeMap = {
    affects: {
        baseName: "affects",
        type: "Array<ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems>",
    },
    bomRef: {
        baseName: "bom_ref",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
