// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition = void 0;
/**
 * A specific position (line and column) within a source file.
 */
class ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition.attributeTypeMap;
    }
}
exports.ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition = ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition;
/**
 * @ignore
 */
ScaRequestDataAttributesDependenciesItemsLocationsItemsPosition.attributeTypeMap = {
    col: {
        baseName: "col",
        type: "number",
        format: "int32",
    },
    line: {
        baseName: "line",
        type: "number",
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
