// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems = void 0;
/**
 * A reference to a component affected by a vulnerability.
 */
class ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems.attributeTypeMap;
    }
}
exports.ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems = ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems;
/**
 * @ignore
 */
ScaRequestDataAttributesVulnerabilitiesItemsAffectsItems.attributeTypeMap = {
    ref: {
        baseName: "ref",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
