// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScheduleRequestDataAttributesLayersItemsMembersItemsUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleRequestDataAttributesLayersItemsMembersItemsUser = void 0;
/**
 * Identifies the user participating in this layer as a single object with an `id`.
 */
class ScheduleRequestDataAttributesLayersItemsMembersItemsUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScheduleRequestDataAttributesLayersItemsMembersItemsUser.attributeTypeMap;
    }
}
exports.ScheduleRequestDataAttributesLayersItemsMembersItemsUser = ScheduleRequestDataAttributesLayersItemsMembersItemsUser;
/**
 * @ignore
 */
ScheduleRequestDataAttributesLayersItemsMembersItemsUser.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
