// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScorecardListResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScorecardListResponseAttributes = void 0;
/**
 * Scorecard attributes.
 */
class ScorecardListResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScorecardListResponseAttributes.attributeTypeMap;
    }
}
exports.ScorecardListResponseAttributes = ScorecardListResponseAttributes;
/**
 * @ignore
 */
ScorecardListResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
