// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SeatUserDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeatUserDataAttributes = void 0;
/**
 * Attributes of a user assigned to a seat, including their email, name, and assignment timestamp.
 */
class SeatUserDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SeatUserDataAttributes.attributeTypeMap;
    }
}
exports.SeatUserDataAttributes = SeatUserDataAttributes;
/**
 * @ignore
 */
SeatUserDataAttributes.attributeTypeMap = {
    assignedAt: {
        baseName: "assigned_at",
        type: "Date",
        format: "date-time",
    },
    email: {
        baseName: "email",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
