// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringTerraformExportAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringTerraformExportAttributes = void 0;
/**
 * Attributes of the Terraform export response.
 */
class SecurityMonitoringTerraformExportAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringTerraformExportAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringTerraformExportAttributes = SecurityMonitoringTerraformExportAttributes;
/**
 * @ignore
 */
SecurityMonitoringTerraformExportAttributes.attributeTypeMap = {
    output: {
        baseName: "output",
        type: "string",
    },
    resourceId: {
        baseName: "resource_id",
        type: "string",
        required: true,
    },
    typeName: {
        baseName: "type_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
