// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecureEmbedCreateRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecureEmbedCreateRequestAttributes = void 0;
/**
 * Attributes for creating a secure embed shared dashboard.
 */
class SecureEmbedCreateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecureEmbedCreateRequestAttributes.attributeTypeMap;
    }
}
exports.SecureEmbedCreateRequestAttributes = SecureEmbedCreateRequestAttributes;
/**
 * @ignore
 */
SecureEmbedCreateRequestAttributes.attributeTypeMap = {
    globalTime: {
        baseName: "global_time",
        type: "SecureEmbedGlobalTime",
        required: true,
    },
    globalTimeSelectable: {
        baseName: "global_time_selectable",
        type: "boolean",
        required: true,
    },
    selectableTemplateVars: {
        baseName: "selectable_template_vars",
        type: "Array<SecureEmbedSelectableTemplateVariable>",
        required: true,
    },
    status: {
        baseName: "status",
        type: "SecureEmbedStatus",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    viewingPreferences: {
        baseName: "viewing_preferences",
        type: "SecureEmbedViewingPreferences",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
