// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalSuggestedActionAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalSuggestedActionAttributes = void 0;
/**
 * Attributes of a suggested action for a security signal. The available fields depend on the action type.
 */
class SecurityMonitoringSignalSuggestedActionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalSuggestedActionAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalSuggestedActionAttributes = SecurityMonitoringSignalSuggestedActionAttributes;
/**
 * @ignore
 */
SecurityMonitoringSignalSuggestedActionAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    queryFilter: {
        baseName: "query_filter",
        type: "string",
    },
    templateVariables: {
        baseName: "template_variables",
        type: "{ [key: string]: Array<string>; }",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
