// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalsBulkAssigneeUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalsBulkAssigneeUpdateAttributes = void 0;
/**
 * Attributes describing the new assignees for a bulk signal update.
 */
class SecurityMonitoringSignalsBulkAssigneeUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalsBulkAssigneeUpdateAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalsBulkAssigneeUpdateAttributes = SecurityMonitoringSignalsBulkAssigneeUpdateAttributes;
/**
 * @ignore
 */
SecurityMonitoringSignalsBulkAssigneeUpdateAttributes.attributeTypeMap = {
    assignee: {
        baseName: "assignee",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
