// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringTerraformConvertData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringTerraformConvertData = void 0;
/**
 * The convert request data object.
 */
class SecurityMonitoringTerraformConvertData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringTerraformConvertData.attributeTypeMap;
    }
}
exports.SecurityMonitoringTerraformConvertData = SecurityMonitoringTerraformConvertData;
/**
 * @ignore
 */
SecurityMonitoringTerraformConvertData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringTerraformConvertAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
