// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalsBulkUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalsBulkUpdateData = void 0;
/**
 * Data for updating a single security signal in a bulk update operation.
 */
class SecurityMonitoringSignalsBulkUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalsBulkUpdateData.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalsBulkUpdateData = SecurityMonitoringSignalsBulkUpdateData;
/**
 * @ignore
 */
SecurityMonitoringSignalsBulkUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringSignalUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringSignalType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
