// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringThirdPartyRuleCase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringThirdPartyRuleCase = void 0;
/**
 * Case when signal is generated by a third party rule.
 */
class SecurityMonitoringThirdPartyRuleCase {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringThirdPartyRuleCase.attributeTypeMap;
    }
}
exports.SecurityMonitoringThirdPartyRuleCase = SecurityMonitoringThirdPartyRuleCase;
/**
 * @ignore
 */
SecurityMonitoringThirdPartyRuleCase.attributeTypeMap = {
    customStatus: {
        baseName: "customStatus",
        type: "SecurityMonitoringRuleSeverity",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    notifications: {
        baseName: "notifications",
        type: "Array<string>",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "SecurityMonitoringRuleSeverity",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
