// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalsBulkTriageUpdateResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalsBulkTriageUpdateResponse = void 0;
/**
 * Response for a bulk triage update of security signals.
 */
class SecurityMonitoringSignalsBulkTriageUpdateResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalsBulkTriageUpdateResponse.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalsBulkTriageUpdateResponse = SecurityMonitoringSignalsBulkTriageUpdateResponse;
/**
 * @ignore
 */
SecurityMonitoringSignalsBulkTriageUpdateResponse.attributeTypeMap = {
    result: {
        baseName: "result",
        type: "SecurityMonitoringSignalsBulkTriageUpdateResult",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
