// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecureEmbedSelectableTemplateVariable.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecureEmbedSelectableTemplateVariable = void 0;
/**
 * A template variable that viewers can modify on the secure embed shared dashboard.
 */
class SecureEmbedSelectableTemplateVariable {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecureEmbedSelectableTemplateVariable.attributeTypeMap;
    }
}
exports.SecureEmbedSelectableTemplateVariable = SecureEmbedSelectableTemplateVariable;
/**
 * @ignore
 */
SecureEmbedSelectableTemplateVariable.attributeTypeMap = {
    defaultValues: {
        baseName: "default_values",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    prefix: {
        baseName: "prefix",
        type: "string",
    },
    visibleTags: {
        baseName: "visible_tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
