// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalsBulkTriageUpdateResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalsBulkTriageUpdateResult = void 0;
/**
 * The result payload of a bulk signal triage update.
 */
class SecurityMonitoringSignalsBulkTriageUpdateResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalsBulkTriageUpdateResult.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalsBulkTriageUpdateResult = SecurityMonitoringSignalsBulkTriageUpdateResult;
/**
 * @ignore
 */
SecurityMonitoringSignalsBulkTriageUpdateResult.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    events: {
        baseName: "events",
        type: "Array<SecurityMonitoringSignalsBulkTriageEvent>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
