// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecureEmbedCreateResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecureEmbedCreateResponseAttributes = void 0;
/**
 * Attributes of a newly created secure embed shared dashboard.
 */
class SecureEmbedCreateResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecureEmbedCreateResponseAttributes.attributeTypeMap;
    }
}
exports.SecureEmbedCreateResponseAttributes = SecureEmbedCreateResponseAttributes;
/**
 * @ignore
 */
SecureEmbedCreateResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "string",
    },
    credential: {
        baseName: "credential",
        type: "string",
    },
    dashboardId: {
        baseName: "dashboard_id",
        type: "string",
    },
    globalTime: {
        baseName: "global_time",
        type: "SecureEmbedGlobalTime",
    },
    globalTimeSelectable: {
        baseName: "global_time_selectable",
        type: "boolean",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    selectableTemplateVars: {
        baseName: "selectable_template_vars",
        type: "Array<SecureEmbedSelectableTemplateVariable>",
    },
    shareType: {
        baseName: "share_type",
        type: "SecureEmbedShareType",
    },
    status: {
        baseName: "status",
        type: "SecureEmbedStatus",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    token: {
        baseName: "token",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    viewingPreferences: {
        baseName: "viewing_preferences",
        type: "SecureEmbedViewingPreferences",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
