// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringRuleSort.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SOURCE_DESCENDING = exports.HIGHEST_SEVERITY_DESCENDING = exports.TYPE_DESCENDING = exports.ENABLED_DESCENDING = exports.UPDATE_DATE_DESCENDING = exports.CREATION_DATE_DESCENDING = exports.NAME_DESCENDING = exports.SOURCE = exports.HIGHEST_SEVERITY = exports.TYPE = exports.ENABLED = exports.UPDATE_DATE = exports.CREATION_DATE = exports.NAME = void 0;
exports.NAME = "name";
exports.CREATION_DATE = "creation_date";
exports.UPDATE_DATE = "update_date";
exports.ENABLED = "enabled";
exports.TYPE = "type";
exports.HIGHEST_SEVERITY = "highest_severity";
exports.SOURCE = "source";
exports.NAME_DESCENDING = "-name";
exports.CREATION_DATE_DESCENDING = "-creation_date";
exports.UPDATE_DATE_DESCENDING = "-update_date";
exports.ENABLED_DESCENDING = "-enabled";
exports.TYPE_DESCENDING = "-type";
exports.HIGHEST_SEVERITY_DESCENDING = "-highest_severity";
exports.SOURCE_DESCENDING = "-source";
