// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityFilterExclusionFilterResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityFilterExclusionFilterResponse = void 0;
/**
 * A single exclusion filter.
 */
class SecurityFilterExclusionFilterResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityFilterExclusionFilterResponse.attributeTypeMap;
    }
}
exports.SecurityFilterExclusionFilterResponse = SecurityFilterExclusionFilterResponse;
/**
 * @ignore
 */
SecurityFilterExclusionFilterResponse.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
