// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalUpdateAttributes = void 0;
/**
 * Attributes for updating the triage state or assignee of a security signal.
 */
class SecurityMonitoringSignalUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalUpdateAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalUpdateAttributes = SecurityMonitoringSignalUpdateAttributes;
/**
 * @ignore
 */
SecurityMonitoringSignalUpdateAttributes.attributeTypeMap = {
    archiveComment: {
        baseName: "archive_comment",
        type: "string",
    },
    archiveReason: {
        baseName: "archive_reason",
        type: "SecurityMonitoringSignalArchiveReason",
    },
    assignee: {
        baseName: "assignee",
        type: "SecurityMonitoringTriageUser",
    },
    state: {
        baseName: "state",
        type: "SecurityMonitoringSignalState",
    },
    version: {
        baseName: "version",
        type: "number",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
