// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSuppressionSort.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ENABLED_DESCENDING = exports.CREATION_DATE_DESCENDING = exports.UPDATE_DATE_DESCENDING = exports.EXPIRATION_DATE_DESCENDING = exports.START_DATE_DESCENDING = exports.NAME_DESCENDING = exports.ENABLED = exports.UPDATE_DATE = exports.EXPIRATION_DATE = exports.START_DATE = exports.NAME = void 0;
exports.NAME = "name";
exports.START_DATE = "start_date";
exports.EXPIRATION_DATE = "expiration_date";
exports.UPDATE_DATE = "update_date";
exports.ENABLED = "enabled";
exports.NAME_DESCENDING = "-name";
exports.START_DATE_DESCENDING = "-start_date";
exports.EXPIRATION_DATE_DESCENDING = "-expiration_date";
exports.UPDATE_DATE_DESCENDING = "-update_date";
exports.CREATION_DATE_DESCENDING = "-creation_date";
exports.ENABLED_DESCENDING = "-enabled";
