// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringTerraformExportData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringTerraformExportData = void 0;
/**
 * The Terraform export data object.
 */
class SecurityMonitoringTerraformExportData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringTerraformExportData.attributeTypeMap;
    }
}
exports.SecurityMonitoringTerraformExportData = SecurityMonitoringTerraformExportData;
/**
 * @ignore
 */
SecurityMonitoringTerraformExportData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringTerraformExportAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
