// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalSuggestedAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalSuggestedAction = void 0;
/**
 * A suggested action for a security signal.
 */
class SecurityMonitoringSignalSuggestedAction {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalSuggestedAction.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalSuggestedAction = SecurityMonitoringSignalSuggestedAction;
/**
 * @ignore
 */
SecurityMonitoringSignalSuggestedAction.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringSignalSuggestedActionAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringSignalSuggestedActionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
