// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringTerraformBulkExportAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringTerraformBulkExportAttributes = void 0;
/**
 * Attributes for the bulk export request.
 */
class SecurityMonitoringTerraformBulkExportAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringTerraformBulkExportAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringTerraformBulkExportAttributes = SecurityMonitoringTerraformBulkExportAttributes;
/**
 * @ignore
 */
SecurityMonitoringTerraformBulkExportAttributes.attributeTypeMap = {
    resourceIds: {
        baseName: "resource_ids",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
