// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalsBulkAssigneeUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalsBulkAssigneeUpdateData = void 0;
/**
 * Data for updating the assignees for multiple security signals.
 */
class SecurityMonitoringSignalsBulkAssigneeUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalsBulkAssigneeUpdateData.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalsBulkAssigneeUpdateData = SecurityMonitoringSignalsBulkAssigneeUpdateData;
/**
 * @ignore
 */
SecurityMonitoringSignalsBulkAssigneeUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringSignalsBulkAssigneeUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringSignalType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
