// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecureEmbedUpdateRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecureEmbedUpdateRequestAttributes = void 0;
/**
 * Attributes for updating a secure embed shared dashboard. All fields are optional.
 */
class SecureEmbedUpdateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecureEmbedUpdateRequestAttributes.attributeTypeMap;
    }
}
exports.SecureEmbedUpdateRequestAttributes = SecureEmbedUpdateRequestAttributes;
/**
 * @ignore
 */
SecureEmbedUpdateRequestAttributes.attributeTypeMap = {
    globalTime: {
        baseName: "global_time",
        type: "SecureEmbedGlobalTime",
    },
    globalTimeSelectable: {
        baseName: "global_time_selectable",
        type: "boolean",
    },
    selectableTemplateVars: {
        baseName: "selectable_template_vars",
        type: "Array<SecureEmbedSelectableTemplateVariable>",
    },
    status: {
        baseName: "status",
        type: "SecureEmbedStatus",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    viewingPreferences: {
        baseName: "viewing_preferences",
        type: "SecureEmbedViewingPreferences",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
