// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringThirdPartyRuleCaseCreate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringThirdPartyRuleCaseCreate = void 0;
/**
 * Case when a signal is generated by a third party rule.
 */
class SecurityMonitoringThirdPartyRuleCaseCreate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringThirdPartyRuleCaseCreate.attributeTypeMap;
    }
}
exports.SecurityMonitoringThirdPartyRuleCaseCreate = SecurityMonitoringThirdPartyRuleCaseCreate;
/**
 * @ignore
 */
SecurityMonitoringThirdPartyRuleCaseCreate.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    notifications: {
        baseName: "notifications",
        type: "Array<string>",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "SecurityMonitoringRuleSeverity",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
