// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringSignalsBulkStateUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringSignalsBulkStateUpdateData = void 0;
/**
 * Data for updating the state for multiple security signals.
 */
class SecurityMonitoringSignalsBulkStateUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringSignalsBulkStateUpdateData.attributeTypeMap;
    }
}
exports.SecurityMonitoringSignalsBulkStateUpdateData = SecurityMonitoringSignalsBulkStateUpdateData;
/**
 * @ignore
 */
SecurityMonitoringSignalsBulkStateUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringSignalStateUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringSignalType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
