// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ServiceAccountAccessTokenCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAccountAccessTokenCreateAttributes = void 0;
/**
 * Attributes used to create a service account access token.
 */
class ServiceAccountAccessTokenCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ServiceAccountAccessTokenCreateAttributes.attributeTypeMap;
    }
}
exports.ServiceAccountAccessTokenCreateAttributes = ServiceAccountAccessTokenCreateAttributes;
/**
 * @ignore
 */
ServiceAccountAccessTokenCreateAttributes.attributeTypeMap = {
    expiresAt: {
        baseName: "expires_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    scopes: {
        baseName: "scopes",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
