// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SloQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SloQuery = void 0;
/**
 * A query for SLO status, error budget, and burn rate metrics.
 */
class SloQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SloQuery.attributeTypeMap;
    }
}
exports.SloQuery = SloQuery;
/**
 * @ignore
 */
SloQuery.attributeTypeMap = {
    additionalQueryFilters: {
        baseName: "additional_query_filters",
        type: "string",
    },
    dataSource: {
        baseName: "data_source",
        type: "SloDataSource",
        required: true,
    },
    groupMode: {
        baseName: "group_mode",
        type: "SlosGroupMode",
    },
    measure: {
        baseName: "measure",
        type: "SlosMeasure",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    sloId: {
        baseName: "slo_id",
        type: "string",
        required: true,
    },
    sloQueryType: {
        baseName: "slo_query_type",
        type: "SlosQueryType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
