// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SlosMeasure.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BAD_MINUTES = exports.GOOD_MINUTES = exports.SLO_STATUS_HISTORY = exports.BURN_RATE = exports.ERROR_BUDGET_BURNDOWN = exports.ERROR_BUDGET_REMAINING_HISTORY = exports.ERROR_BUDGET_REMAINING = exports.SLO_STATUS = exports.BAD_EVENTS = exports.GOOD_EVENTS = void 0;
exports.GOOD_EVENTS = "good_events";
exports.BAD_EVENTS = "bad_events";
exports.SLO_STATUS = "slo_status";
exports.ERROR_BUDGET_REMAINING = "error_budget_remaining";
exports.ERROR_BUDGET_REMAINING_HISTORY = "error_budget_remaining_history";
exports.ERROR_BUDGET_BURNDOWN = "error_budget_burndown";
exports.BURN_RATE = "burn_rate";
exports.SLO_STATUS_HISTORY = "slo_status_history";
exports.GOOD_MINUTES = "good_minutes";
exports.BAD_MINUTES = "bad_minutes";
