// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatusPagesComponentGroupRelationshipsStatusPageData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusPagesComponentGroupRelationshipsStatusPageData = void 0;
/**
 * The data object identifying the status page the component group belongs to.
 */
class StatusPagesComponentGroupRelationshipsStatusPageData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatusPagesComponentGroupRelationshipsStatusPageData.attributeTypeMap;
    }
}
exports.StatusPagesComponentGroupRelationshipsStatusPageData = StatusPagesComponentGroupRelationshipsStatusPageData;
/**
 * @ignore
 */
StatusPagesComponentGroupRelationshipsStatusPageData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "StatusPageDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
