// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatusPageAsIncludedRelationshipsLastModifiedByUserData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusPageAsIncludedRelationshipsLastModifiedByUserData = void 0;
/**
 * The data object identifying the Datadog user who last modified the status page.
 */
class StatusPageAsIncludedRelationshipsLastModifiedByUserData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatusPageAsIncludedRelationshipsLastModifiedByUserData.attributeTypeMap;
    }
}
exports.StatusPageAsIncludedRelationshipsLastModifiedByUserData = StatusPageAsIncludedRelationshipsLastModifiedByUserData;
/**
 * @ignore
 */
StatusPageAsIncludedRelationshipsLastModifiedByUserData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "StatusPagesUserType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
