// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatusPagesComponentDataRelationshipsLastModifiedByUserData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusPagesComponentDataRelationshipsLastModifiedByUserData = void 0;
/**
 * The data object identifying the Datadog user who last modified the component.
 */
class StatusPagesComponentDataRelationshipsLastModifiedByUserData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatusPagesComponentDataRelationshipsLastModifiedByUserData.attributeTypeMap;
    }
}
exports.StatusPagesComponentDataRelationshipsLastModifiedByUserData = StatusPagesComponentDataRelationshipsLastModifiedByUserData;
/**
 * @ignore
 */
StatusPagesComponentDataRelationshipsLastModifiedByUserData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "StatusPagesUserType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
