// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatusPagesComponentData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusPagesComponentData = void 0;
/**
 * The data object for a component.
 */
class StatusPagesComponentData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatusPagesComponentData.attributeTypeMap;
    }
}
exports.StatusPagesComponentData = StatusPagesComponentData;
/**
 * @ignore
 */
StatusPagesComponentData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "StatusPagesComponentDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    relationships: {
        baseName: "relationships",
        type: "StatusPagesComponentDataRelationships",
    },
    type: {
        baseName: "type",
        type: "StatusPagesComponentGroupType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
