// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultCertificateValidity.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultCertificateValidity = void 0;
/**
 * Validity window of a certificate.
 */
class SyntheticsTestResultCertificateValidity {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultCertificateValidity.attributeTypeMap;
    }
}
exports.SyntheticsTestResultCertificateValidity = SyntheticsTestResultCertificateValidity;
/**
 * @ignore
 */
SyntheticsTestResultCertificateValidity.attributeTypeMap = {
    from: {
        baseName: "from",
        type: "number",
        format: "int64",
    },
    to: {
        baseName: "to",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
