// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDeviceBrowser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDeviceBrowser = void 0;
/**
 * Browser information for the device used to run the test.
 */
class SyntheticsTestResultDeviceBrowser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDeviceBrowser.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDeviceBrowser = SyntheticsTestResultDeviceBrowser;
/**
 * @ignore
 */
SyntheticsTestResultDeviceBrowser.attributeTypeMap = {
    type: {
        baseName: "type",
        type: "string",
    },
    userAgent: {
        baseName: "user_agent",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
