// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsApiMultistepSubtestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsApiMultistepSubtestAttributes = void 0;
/**
 * Attributes of a Synthetic API multistep subtest.
 */
class SyntheticsApiMultistepSubtestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsApiMultistepSubtestAttributes.attributeTypeMap;
    }
}
exports.SyntheticsApiMultistepSubtestAttributes = SyntheticsApiMultistepSubtestAttributes;
/**
 * @ignore
 */
SyntheticsApiMultistepSubtestAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    publicId: {
        baseName: "public_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
