// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultRouter.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultRouter = void 0;
/**
 * A router along the traceroute path.
 */
class SyntheticsTestResultRouter {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultRouter.attributeTypeMap;
    }
}
exports.SyntheticsTestResultRouter = SyntheticsTestResultRouter;
/**
 * @ignore
 */
SyntheticsTestResultRouter.attributeTypeMap = {
    ip: {
        baseName: "ip",
        type: "string",
    },
    resolvedHost: {
        baseName: "resolved_host",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
