// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultNetstats.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultNetstats = void 0;
/**
 * Aggregated network statistics from the test execution.
 */
class SyntheticsTestResultNetstats {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultNetstats.attributeTypeMap;
    }
}
exports.SyntheticsTestResultNetstats = SyntheticsTestResultNetstats;
/**
 * @ignore
 */
SyntheticsTestResultNetstats.attributeTypeMap = {
    hops: {
        baseName: "hops",
        type: "SyntheticsTestResultNetstatsHops",
    },
    jitter: {
        baseName: "jitter",
        type: "number",
        format: "double",
    },
    latency: {
        baseName: "latency",
        type: "SyntheticsTestResultNetworkLatency",
    },
    packetLossPercentage: {
        baseName: "packet_loss_percentage",
        type: "number",
        format: "double",
    },
    packetsReceived: {
        baseName: "packets_received",
        type: "number",
        format: "int64",
    },
    packetsSent: {
        baseName: "packets_sent",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
