// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultHealthCheck.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultHealthCheck = void 0;
/**
 * Health check information returned from a gRPC health check call.
 */
class SyntheticsTestResultHealthCheck {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultHealthCheck.attributeTypeMap;
    }
}
exports.SyntheticsTestResultHealthCheck = SyntheticsTestResultHealthCheck;
/**
 * @ignore
 */
SyntheticsTestResultHealthCheck.attributeTypeMap = {
    message: {
        baseName: "message",
        type: "{ [key: string]: string; }",
    },
    status: {
        baseName: "status",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
