// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultCIPipeline.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultCIPipeline = void 0;
/**
 * Details of the CI pipeline.
 */
class SyntheticsTestResultCIPipeline {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultCIPipeline.attributeTypeMap;
    }
}
exports.SyntheticsTestResultCIPipeline = SyntheticsTestResultCIPipeline;
/**
 * @ignore
 */
SyntheticsTestResultCIPipeline.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    number: {
        baseName: "number",
        type: "number",
        format: "int64",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
