// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDnsResolution.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDnsResolution = void 0;
/**
 * DNS resolution details recorded during the test execution.
 */
class SyntheticsTestResultDnsResolution {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDnsResolution.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDnsResolution = SyntheticsTestResultDnsResolution;
/**
 * @ignore
 */
SyntheticsTestResultDnsResolution.attributeTypeMap = {
    attempts: {
        baseName: "attempts",
        type: "Array<{ [key: string]: string; }>",
    },
    resolvedIp: {
        baseName: "resolved_ip",
        type: "string",
    },
    resolvedPort: {
        baseName: "resolved_port",
        type: "string",
    },
    server: {
        baseName: "server",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
