// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultVariable.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultVariable = void 0;
/**
 * A variable used or extracted during a test.
 */
class SyntheticsTestResultVariable {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultVariable.attributeTypeMap;
    }
}
exports.SyntheticsTestResultVariable = SyntheticsTestResultVariable;
/**
 * @ignore
 */
SyntheticsTestResultVariable.attributeTypeMap = {
    err: {
        baseName: "err",
        type: "string",
    },
    errorMessage: {
        baseName: "error_message",
        type: "string",
    },
    example: {
        baseName: "example",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    pattern: {
        baseName: "pattern",
        type: "string",
    },
    secure: {
        baseName: "secure",
        type: "boolean",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    val: {
        baseName: "val",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
