// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultStep.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultStep = void 0;
/**
 * A step result from a browser, mobile, or multistep API test.
 */
class SyntheticsTestResultStep {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultStep.attributeTypeMap;
    }
}
exports.SyntheticsTestResultStep = SyntheticsTestResultStep;
/**
 * @ignore
 */
SyntheticsTestResultStep.attributeTypeMap = {
    allowFailure: {
        baseName: "allow_failure",
        type: "boolean",
    },
    apiTest: {
        baseName: "api_test",
        type: "{ [key: string]: any; }",
    },
    assertionResult: {
        baseName: "assertion_result",
        type: "SyntheticsTestResultStepAssertionResult",
    },
    assertions: {
        baseName: "assertions",
        type: "Array<SyntheticsTestResultAssertionResult>",
    },
    blockedRequestsUrls: {
        baseName: "blocked_requests_urls",
        type: "Array<string>",
    },
    bounds: {
        baseName: "bounds",
        type: "SyntheticsTestResultBounds",
    },
    browserErrors: {
        baseName: "browser_errors",
        type: "Array<SyntheticsTestResultBrowserError>",
    },
    bucketKeys: {
        baseName: "bucket_keys",
        type: "SyntheticsTestResultBucketKeys",
    },
    cdnResources: {
        baseName: "cdn_resources",
        type: "Array<SyntheticsTestResultCdnResource>",
    },
    clickType: {
        baseName: "click_type",
        type: "string",
    },
    compressedJsonDescriptor: {
        baseName: "compressed_json_descriptor",
        type: "string",
    },
    config: {
        baseName: "config",
        type: "{ [key: string]: any; }",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    duration: {
        baseName: "duration",
        type: "number",
        format: "double",
    },
    elementDescription: {
        baseName: "element_description",
        type: "string",
    },
    elementUpdates: {
        baseName: "element_updates",
        type: "SyntheticsTestResultStepElementUpdates",
    },
    extractedValue: {
        baseName: "extracted_value",
        type: "SyntheticsTestResultVariable",
    },
    failure: {
        baseName: "failure",
        type: "SyntheticsTestResultFailure",
    },
    httpResults: {
        baseName: "http_results",
        type: "Array<SyntheticsTestResultAssertionResult>",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    isCritical: {
        baseName: "is_critical",
        type: "boolean",
    },
    javascriptCustomAssertionCode: {
        baseName: "javascript_custom_assertion_code",
        type: "boolean",
    },
    locateElementDuration: {
        baseName: "locate_element_duration",
        type: "number",
        format: "double",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    request: {
        baseName: "request",
        type: "SyntheticsTestResultRequestInfo",
    },
    response: {
        baseName: "response",
        type: "SyntheticsTestResultResponseInfo",
    },
    retries: {
        baseName: "retries",
        type: "Array<SyntheticsTestResultStep>",
    },
    retryCount: {
        baseName: "retry_count",
        type: "number",
        format: "int64",
    },
    rumContext: {
        baseName: "rum_context",
        type: "SyntheticsTestResultRumContext",
    },
    startedAt: {
        baseName: "started_at",
        type: "number",
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "string",
    },
    subStep: {
        baseName: "sub_step",
        type: "SyntheticsTestResultSubStep",
    },
    subTest: {
        baseName: "sub_test",
        type: "SyntheticsTestResultSubTest",
    },
    subtype: {
        baseName: "subtype",
        type: "string",
    },
    tabs: {
        baseName: "tabs",
        type: "Array<SyntheticsTestResultTab>",
    },
    timings: {
        baseName: "timings",
        type: "{ [key: string]: any; }",
    },
    tunnel: {
        baseName: "tunnel",
        type: "boolean",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "any",
    },
    variables: {
        baseName: "variables",
        type: "SyntheticsTestResultVariables",
    },
    vitalsMetrics: {
        baseName: "vitals_metrics",
        type: "Array<SyntheticsTestResultVitalsMetrics>",
    },
    warnings: {
        baseName: "warnings",
        type: "Array<SyntheticsTestResultWarning>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
