// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultStepsInfo.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultStepsInfo = void 0;
/**
 * Step execution summary for a Synthetic test result.
 */
class SyntheticsTestResultStepsInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultStepsInfo.attributeTypeMap;
    }
}
exports.SyntheticsTestResultStepsInfo = SyntheticsTestResultStepsInfo;
/**
 * @ignore
 */
SyntheticsTestResultStepsInfo.attributeTypeMap = {
    completed: {
        baseName: "completed",
        type: "number",
        format: "int64",
    },
    errors: {
        baseName: "errors",
        type: "number",
        format: "int64",
    },
    total: {
        baseName: "total",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
