// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultIncludedItem.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultIncludedItem = void 0;
/**
 * An included related resource.
 */
class SyntheticsTestResultIncludedItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultIncludedItem.attributeTypeMap;
    }
}
exports.SyntheticsTestResultIncludedItem = SyntheticsTestResultIncludedItem;
/**
 * @ignore
 */
SyntheticsTestResultIncludedItem.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "{ [key: string]: any; }",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
