// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultFailure.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultFailure = void 0;
/**
 * Details about the failure of a Synthetic test.
 */
class SyntheticsTestResultFailure {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultFailure.attributeTypeMap;
    }
}
exports.SyntheticsTestResultFailure = SyntheticsTestResultFailure;
/**
 * @ignore
 */
SyntheticsTestResultFailure.attributeTypeMap = {
    code: {
        baseName: "code",
        type: "string",
    },
    internalCode: {
        baseName: "internal_code",
        type: "string",
    },
    internalMessage: {
        baseName: "internal_message",
        type: "string",
    },
    message: {
        baseName: "message",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
