// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultOCSPUpdates.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultOCSPUpdates = void 0;
/**
 * OCSP response update timestamps.
 */
class SyntheticsTestResultOCSPUpdates {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultOCSPUpdates.attributeTypeMap;
    }
}
exports.SyntheticsTestResultOCSPUpdates = SyntheticsTestResultOCSPUpdates;
/**
 * @ignore
 */
SyntheticsTestResultOCSPUpdates.attributeTypeMap = {
    nextUpdate: {
        baseName: "next_update",
        type: "number",
        format: "int64",
    },
    producedAt: {
        baseName: "produced_at",
        type: "number",
        format: "int64",
    },
    thisUpdate: {
        baseName: "this_update",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
