// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultTracerouteHop.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultTracerouteHop = void 0;
/**
 * A network probe result, used for traceroute hops and ping summaries.
 */
class SyntheticsTestResultTracerouteHop {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultTracerouteHop.attributeTypeMap;
    }
}
exports.SyntheticsTestResultTracerouteHop = SyntheticsTestResultTracerouteHop;
/**
 * @ignore
 */
SyntheticsTestResultTracerouteHop.attributeTypeMap = {
    host: {
        baseName: "host",
        type: "string",
    },
    latency: {
        baseName: "latency",
        type: "SyntheticsTestResultNetworkLatency",
    },
    packetLossPercentage: {
        baseName: "packet_loss_percentage",
        type: "number",
        format: "double",
    },
    packetSize: {
        baseName: "packet_size",
        type: "number",
        format: "int64",
    },
    packetsReceived: {
        baseName: "packets_received",
        type: "number",
        format: "int64",
    },
    packetsSent: {
        baseName: "packets_sent",
        type: "number",
        format: "int64",
    },
    resolvedIp: {
        baseName: "resolved_ip",
        type: "string",
    },
    routers: {
        baseName: "routers",
        type: "Array<SyntheticsTestResultRouter>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
