// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileAbortMultipartUploadRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileAbortMultipartUploadRequest = void 0;
/**
 * Request body for aborting a multipart file upload.
 */
class SyntheticsTestFileAbortMultipartUploadRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileAbortMultipartUploadRequest.attributeTypeMap;
    }
}
exports.SyntheticsTestFileAbortMultipartUploadRequest = SyntheticsTestFileAbortMultipartUploadRequest;
/**
 * @ignore
 */
SyntheticsTestFileAbortMultipartUploadRequest.attributeTypeMap = {
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    uploadId: {
        baseName: "uploadId",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
