// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultAttributes = void 0;
/**
 * Attributes of a Synthetic test result.
 */
class SyntheticsTestResultAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultAttributes.attributeTypeMap;
    }
}
exports.SyntheticsTestResultAttributes = SyntheticsTestResultAttributes;
/**
 * @ignore
 */
SyntheticsTestResultAttributes.attributeTypeMap = {
    batch: {
        baseName: "batch",
        type: "SyntheticsTestResultBatch",
    },
    ci: {
        baseName: "ci",
        type: "SyntheticsTestResultCI",
    },
    device: {
        baseName: "device",
        type: "SyntheticsTestResultDevice",
    },
    git: {
        baseName: "git",
        type: "SyntheticsTestResultGit",
    },
    location: {
        baseName: "location",
        type: "SyntheticsTestResultLocation",
    },
    result: {
        baseName: "result",
        type: "SyntheticsTestResultDetail",
    },
    testSubType: {
        baseName: "test_sub_type",
        type: "SyntheticsTestSubType",
    },
    testType: {
        baseName: "test_type",
        type: "SyntheticsTestType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
