// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultNetpathDestination.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultNetpathDestination = void 0;
/**
 * Destination endpoint of a network path measurement.
 */
class SyntheticsTestResultNetpathDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultNetpathDestination.attributeTypeMap;
    }
}
exports.SyntheticsTestResultNetpathDestination = SyntheticsTestResultNetpathDestination;
/**
 * @ignore
 */
SyntheticsTestResultNetpathDestination.attributeTypeMap = {
    hostname: {
        baseName: "hostname",
        type: "string",
    },
    ipAddress: {
        baseName: "ip_address",
        type: "string",
    },
    port: {
        baseName: "port",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
