// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultTrace.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultTrace = void 0;
/**
 * Trace identifiers associated with a Synthetic test result.
 */
class SyntheticsTestResultTrace {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultTrace.attributeTypeMap;
    }
}
exports.SyntheticsTestResultTrace = SyntheticsTestResultTrace;
/**
 * @ignore
 */
SyntheticsTestResultTrace.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    otelId: {
        baseName: "otel_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
