// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileCompleteMultipartUploadPart.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileCompleteMultipartUploadPart = void 0;
/**
 * A completed part of a multipart upload.
 */
class SyntheticsTestFileCompleteMultipartUploadPart {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileCompleteMultipartUploadPart.attributeTypeMap;
    }
}
exports.SyntheticsTestFileCompleteMultipartUploadPart = SyntheticsTestFileCompleteMultipartUploadPart;
/**
 * @ignore
 */
SyntheticsTestFileCompleteMultipartUploadPart.attributeTypeMap = {
    eTag: {
        baseName: "ETag",
        type: "string",
        required: true,
    },
    partNumber: {
        baseName: "PartNumber",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
