// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultTab.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultTab = void 0;
/**
 * Information about a browser tab involved in a step.
 */
class SyntheticsTestResultTab {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultTab.attributeTypeMap;
    }
}
exports.SyntheticsTestResultTab = SyntheticsTestResultTab;
/**
 * @ignore
 */
SyntheticsTestResultTab.attributeTypeMap = {
    focused: {
        baseName: "focused",
        type: "boolean",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
