// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestParentSuiteAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestParentSuiteAttributes = void 0;
/**
 * Object containing details about a parent suite of a Synthetic test.
 */
class SyntheticsTestParentSuiteAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestParentSuiteAttributes.attributeTypeMap;
    }
}
exports.SyntheticsTestParentSuiteAttributes = SyntheticsTestParentSuiteAttributes;
/**
 * @ignore
 */
SyntheticsTestParentSuiteAttributes.attributeTypeMap = {
    childName: {
        baseName: "child_name",
        type: "string",
    },
    childPublicId: {
        baseName: "child_public_id",
        type: "string",
    },
    monitorId: {
        baseName: "monitor_id",
        type: "number",
        format: "int64",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    overallState: {
        baseName: "overall_state",
        type: "number",
        format: "int64",
    },
    overallStateModified: {
        baseName: "overall_state_modified",
        type: "string",
    },
    publicId: {
        baseName: "public_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
