// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsFastTestResultDetail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsFastTestResultDetail = void 0;
/**
 * Detailed result data for the fast test run. The exact shape of nested fields
 * (`request`, `response`, `assertions`, etc.) depends on the test subtype.
 */
class SyntheticsFastTestResultDetail {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsFastTestResultDetail.attributeTypeMap;
    }
}
exports.SyntheticsFastTestResultDetail = SyntheticsFastTestResultDetail;
/**
 * @ignore
 */
SyntheticsFastTestResultDetail.attributeTypeMap = {
    assertions: {
        baseName: "assertions",
        type: "Array<SyntheticsTestResultAssertionResult>",
    },
    callType: {
        baseName: "call_type",
        type: "string",
    },
    cert: {
        baseName: "cert",
        type: "SyntheticsTestResultCertificate",
    },
    duration: {
        baseName: "duration",
        type: "number",
        format: "double",
    },
    failure: {
        baseName: "failure",
        type: "SyntheticsTestResultFailure",
    },
    finishedAt: {
        baseName: "finished_at",
        type: "number",
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    isFastRetry: {
        baseName: "is_fast_retry",
        type: "boolean",
    },
    request: {
        baseName: "request",
        type: "SyntheticsTestResultRequestInfo",
    },
    resolvedIp: {
        baseName: "resolved_ip",
        type: "string",
    },
    response: {
        baseName: "response",
        type: "SyntheticsTestResultResponseInfo",
    },
    runType: {
        baseName: "run_type",
        type: "SyntheticsTestResultRunType",
    },
    startedAt: {
        baseName: "started_at",
        type: "number",
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "string",
    },
    steps: {
        baseName: "steps",
        type: "Array<SyntheticsTestResultStep>",
    },
    timings: {
        baseName: "timings",
        type: "{ [key: string]: any; }",
    },
    traceroute: {
        baseName: "traceroute",
        type: "Array<SyntheticsTestResultTracerouteHop>",
    },
    triggeredAt: {
        baseName: "triggered_at",
        type: "number",
        format: "int64",
    },
    tunnel: {
        baseName: "tunnel",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
