// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultSummaryData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultSummaryData = void 0;
/**
 * Wrapper object for a Synthetic test result summary.
 */
class SyntheticsTestResultSummaryData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultSummaryData.attributeTypeMap;
    }
}
exports.SyntheticsTestResultSummaryData = SyntheticsTestResultSummaryData;
/**
 * @ignore
 */
SyntheticsTestResultSummaryData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SyntheticsTestResultSummaryAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    relationships: {
        baseName: "relationships",
        type: "SyntheticsTestResultRelationships",
    },
    type: {
        baseName: "type",
        type: "SyntheticsTestResultSummaryType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
