// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultVitalsMetrics.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultVitalsMetrics = void 0;
/**
 * Web vitals metrics captured during a browser test step.
 */
class SyntheticsTestResultVitalsMetrics {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultVitalsMetrics.attributeTypeMap;
    }
}
exports.SyntheticsTestResultVitalsMetrics = SyntheticsTestResultVitalsMetrics;
/**
 * @ignore
 */
SyntheticsTestResultVitalsMetrics.attributeTypeMap = {
    cls: {
        baseName: "cls",
        type: "number",
        format: "double",
    },
    fcp: {
        baseName: "fcp",
        type: "number",
        format: "double",
    },
    inp: {
        baseName: "inp",
        type: "number",
        format: "double",
    },
    lcp: {
        baseName: "lcp",
        type: "number",
        format: "double",
    },
    ttfb: {
        baseName: "ttfb",
        type: "number",
        format: "double",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
