// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileCompleteMultipartUploadRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileCompleteMultipartUploadRequest = void 0;
/**
 * Request body for completing a multipart file upload.
 */
class SyntheticsTestFileCompleteMultipartUploadRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileCompleteMultipartUploadRequest.attributeTypeMap;
    }
}
exports.SyntheticsTestFileCompleteMultipartUploadRequest = SyntheticsTestFileCompleteMultipartUploadRequest;
/**
 * @ignore
 */
SyntheticsTestFileCompleteMultipartUploadRequest.attributeTypeMap = {
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    parts: {
        baseName: "parts",
        type: "Array<SyntheticsTestFileCompleteMultipartUploadPart>",
        required: true,
    },
    uploadId: {
        baseName: "uploadId",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
