// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileMultipartPresignedUrlsPart.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileMultipartPresignedUrlsPart = void 0;
/**
 * A part descriptor for initiating a multipart upload.
 */
class SyntheticsTestFileMultipartPresignedUrlsPart {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileMultipartPresignedUrlsPart.attributeTypeMap;
    }
}
exports.SyntheticsTestFileMultipartPresignedUrlsPart = SyntheticsTestFileMultipartPresignedUrlsPart;
/**
 * @ignore
 */
SyntheticsTestFileMultipartPresignedUrlsPart.attributeTypeMap = {
    md5: {
        baseName: "md5",
        type: "string",
        required: true,
    },
    partNumber: {
        baseName: "partNumber",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
