// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionDiffPatchDiff.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionDiffPatchDiff = void 0;
/**
 * Object describing a single text diff operation.
 */
class SyntheticsTestVersionDiffPatchDiff {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionDiffPatchDiff.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionDiffPatchDiff = SyntheticsTestVersionDiffPatchDiff;
/**
 * @ignore
 */
SyntheticsTestVersionDiffPatchDiff.attributeTypeMap = {
    changeText: {
        baseName: "change_text",
        type: "string",
    },
    operation: {
        baseName: "operation",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
