// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultOCSPCertificate.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultOCSPCertificate = void 0;
/**
 * Certificate details returned in an OCSP response.
 */
class SyntheticsTestResultOCSPCertificate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultOCSPCertificate.attributeTypeMap;
    }
}
exports.SyntheticsTestResultOCSPCertificate = SyntheticsTestResultOCSPCertificate;
/**
 * @ignore
 */
SyntheticsTestResultOCSPCertificate.attributeTypeMap = {
    revocationReason: {
        baseName: "revocation_reason",
        type: "string",
    },
    revocationTime: {
        baseName: "revocation_time",
        type: "number",
        format: "int64",
    },
    serialNumber: {
        baseName: "serial_number",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
