// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionChangeAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionChangeAttributes = void 0;
/**
 * Attributes of a version change record.
 */
class SyntheticsTestVersionChangeAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionChangeAttributes.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionChangeAttributes = SyntheticsTestVersionChangeAttributes;
/**
 * @ignore
 */
SyntheticsTestVersionChangeAttributes.attributeTypeMap = {
    authorUuid: {
        baseName: "author_uuid",
        type: "string",
    },
    changeMetadata: {
        baseName: "change_metadata",
        type: "Array<SyntheticsTestVersionChangeMetadataItem>",
    },
    versionNumber: {
        baseName: "version_number",
        type: "number",
        format: "int64",
    },
    versionPayloadCreatedAt: {
        baseName: "version_payload_created_at",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
