// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultNetpathHop.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultNetpathHop = void 0;
/**
 * A single hop along a network path.
 */
class SyntheticsTestResultNetpathHop {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultNetpathHop.attributeTypeMap;
    }
}
exports.SyntheticsTestResultNetpathHop = SyntheticsTestResultNetpathHop;
/**
 * @ignore
 */
SyntheticsTestResultNetpathHop.attributeTypeMap = {
    hostname: {
        baseName: "hostname",
        type: "string",
    },
    ipAddress: {
        baseName: "ip_address",
        type: "string",
    },
    reachable: {
        baseName: "reachable",
        type: "boolean",
    },
    rtt: {
        baseName: "rtt",
        type: "number",
        format: "double",
    },
    ttl: {
        baseName: "ttl",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
