// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultSummaryAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultSummaryAttributes = void 0;
/**
 * Attributes of a Synthetic test result summary.
 */
class SyntheticsTestResultSummaryAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultSummaryAttributes.attributeTypeMap;
    }
}
exports.SyntheticsTestResultSummaryAttributes = SyntheticsTestResultSummaryAttributes;
/**
 * @ignore
 */
SyntheticsTestResultSummaryAttributes.attributeTypeMap = {
    device: {
        baseName: "device",
        type: "SyntheticsTestResultDevice",
    },
    executionInfo: {
        baseName: "execution_info",
        type: "SyntheticsTestResultExecutionInfo",
    },
    finishedAt: {
        baseName: "finished_at",
        type: "number",
        format: "int64",
    },
    location: {
        baseName: "location",
        type: "SyntheticsTestResultLocation",
    },
    runType: {
        baseName: "run_type",
        type: "SyntheticsTestResultRunType",
    },
    startedAt: {
        baseName: "started_at",
        type: "number",
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "SyntheticsTestResultStatus",
    },
    stepsInfo: {
        baseName: "steps_info",
        type: "SyntheticsTestResultStepsInfo",
    },
    testSubType: {
        baseName: "test_sub_type",
        type: "SyntheticsTestSubType",
    },
    testType: {
        baseName: "test_type",
        type: "SyntheticsTestType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
