// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionAuthor.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionAuthor = void 0;
/**
 * Object describing the author of a test version.
 */
class SyntheticsTestVersionAuthor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionAuthor.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionAuthor = SyntheticsTestVersionAuthor;
/**
 * @ignore
 */
SyntheticsTestVersionAuthor.attributeTypeMap = {
    email: {
        baseName: "email",
        type: "string",
    },
    handle: {
        baseName: "handle",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
