// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultStepElementUpdates.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultStepElementUpdates = void 0;
/**
 * Element locator updates produced during a step.
 */
class SyntheticsTestResultStepElementUpdates {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultStepElementUpdates.attributeTypeMap;
    }
}
exports.SyntheticsTestResultStepElementUpdates = SyntheticsTestResultStepElementUpdates;
/**
 * @ignore
 */
SyntheticsTestResultStepElementUpdates.attributeTypeMap = {
    multiLocator: {
        baseName: "multi_locator",
        type: "{ [key: string]: string; }",
    },
    targetOuterHtml: {
        baseName: "target_outer_html",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
