// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultAssertionResult.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultAssertionResult = void 0;
/**
 * An individual assertion result from a Synthetic test.
 */
class SyntheticsTestResultAssertionResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultAssertionResult.attributeTypeMap;
    }
}
exports.SyntheticsTestResultAssertionResult = SyntheticsTestResultAssertionResult;
/**
 * @ignore
 */
SyntheticsTestResultAssertionResult.attributeTypeMap = {
    actual: {
        baseName: "actual",
        type: "any",
    },
    errorMessage: {
        baseName: "error_message",
        type: "string",
    },
    expected: {
        baseName: "expected",
        type: "any",
    },
    operator: {
        baseName: "operator",
        type: "string",
    },
    property: {
        baseName: "property",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "any",
    },
    targetPath: {
        baseName: "target_path",
        type: "string",
    },
    targetPathOperator: {
        baseName: "target_path_operator",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    valid: {
        baseName: "valid",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
