// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultBounds.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultBounds = void 0;
/**
 * Bounding box of an element on the page.
 */
class SyntheticsTestResultBounds {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultBounds.attributeTypeMap;
    }
}
exports.SyntheticsTestResultBounds = SyntheticsTestResultBounds;
/**
 * @ignore
 */
SyntheticsTestResultBounds.attributeTypeMap = {
    height: {
        baseName: "height",
        type: "number",
        format: "int64",
    },
    width: {
        baseName: "width",
        type: "number",
        format: "int64",
    },
    x: {
        baseName: "x",
        type: "number",
        format: "int64",
    },
    y: {
        baseName: "y",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
