// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultRedirect.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultRedirect = void 0;
/**
 * A redirect hop encountered while performing the request.
 */
class SyntheticsTestResultRedirect {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultRedirect.attributeTypeMap;
    }
}
exports.SyntheticsTestResultRedirect = SyntheticsTestResultRedirect;
/**
 * @ignore
 */
SyntheticsTestResultRedirect.attributeTypeMap = {
    location: {
        baseName: "location",
        type: "string",
    },
    statusCode: {
        baseName: "status_code",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
