// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileMultipartPresignedUrlsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileMultipartPresignedUrlsResponse = void 0;
/**
 * Response containing presigned URLs for multipart file upload and the bucket key.
 */
class SyntheticsTestFileMultipartPresignedUrlsResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileMultipartPresignedUrlsResponse.attributeTypeMap;
    }
}
exports.SyntheticsTestFileMultipartPresignedUrlsResponse = SyntheticsTestFileMultipartPresignedUrlsResponse;
/**
 * @ignore
 */
SyntheticsTestFileMultipartPresignedUrlsResponse.attributeTypeMap = {
    bucketKey: {
        baseName: "bucketKey",
        type: "string",
    },
    multipartPresignedUrlsParams: {
        baseName: "multipart_presigned_urls_params",
        type: "SyntheticsTestFileMultipartPresignedUrlsParams",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
