// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDevicePlatform.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDevicePlatform = void 0;
/**
 * Platform information for the device used to run the test.
 */
class SyntheticsTestResultDevicePlatform {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDevicePlatform.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDevicePlatform = SyntheticsTestResultDevicePlatform;
/**
 * @ignore
 */
SyntheticsTestResultDevicePlatform.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
