// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultRumContext.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultRumContext = void 0;
/**
 * RUM application context associated with a step or sub-test.
 */
class SyntheticsTestResultRumContext {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultRumContext.attributeTypeMap;
    }
}
exports.SyntheticsTestResultRumContext = SyntheticsTestResultRumContext;
/**
 * @ignore
 */
SyntheticsTestResultRumContext.attributeTypeMap = {
    applicationId: {
        baseName: "application_id",
        type: "string",
    },
    sessionId: {
        baseName: "session_id",
        type: "string",
    },
    viewId: {
        baseName: "view_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
