// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultCdnResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultCdnResource = void 0;
/**
 * A CDN resource encountered while executing a browser step.
 */
class SyntheticsTestResultCdnResource {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultCdnResource.attributeTypeMap;
    }
}
exports.SyntheticsTestResultCdnResource = SyntheticsTestResultCdnResource;
/**
 * @ignore
 */
SyntheticsTestResultCdnResource.attributeTypeMap = {
    cdn: {
        baseName: "cdn",
        type: "SyntheticsTestResultCdnProviderInfo",
    },
    resolvedIp: {
        baseName: "resolved_ip",
        type: "string",
    },
    timestamp: {
        baseName: "timestamp",
        type: "number",
        format: "int64",
    },
    timings: {
        baseName: "timings",
        type: "{ [key: string]: any; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
