// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultFileRef.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultFileRef = void 0;
/**
 * Reference to a file attached to a Synthetic test request.
 */
class SyntheticsTestResultFileRef {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultFileRef.attributeTypeMap;
    }
}
exports.SyntheticsTestResultFileRef = SyntheticsTestResultFileRef;
/**
 * @ignore
 */
SyntheticsTestResultFileRef.attributeTypeMap = {
    bucketKey: {
        baseName: "bucket_key",
        type: "string",
    },
    encoding: {
        baseName: "encoding",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    size: {
        baseName: "size",
        type: "number",
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
