// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultTurn.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultTurn = void 0;
/**
 * A turn in a goal-based browser test, grouping steps and reasoning.
 */
class SyntheticsTestResultTurn {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultTurn.attributeTypeMap;
    }
}
exports.SyntheticsTestResultTurn = SyntheticsTestResultTurn;
/**
 * @ignore
 */
SyntheticsTestResultTurn.attributeTypeMap = {
    bucketKeys: {
        baseName: "bucket_keys",
        type: "SyntheticsTestResultBucketKeys",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    reasoning: {
        baseName: "reasoning",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "string",
    },
    steps: {
        baseName: "steps",
        type: "Array<SyntheticsTestResultTurnStep>",
    },
    turnFinishedAt: {
        baseName: "turn_finished_at",
        type: "number",
        format: "int64",
    },
    turnStartedAt: {
        baseName: "turn_started_at",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
