// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileMultipartPresignedUrlsParams.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileMultipartPresignedUrlsParams = void 0;
/**
 * Presigned URL parameters returned for a multipart upload.
 */
class SyntheticsTestFileMultipartPresignedUrlsParams {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileMultipartPresignedUrlsParams.attributeTypeMap;
    }
}
exports.SyntheticsTestFileMultipartPresignedUrlsParams = SyntheticsTestFileMultipartPresignedUrlsParams;
/**
 * @ignore
 */
SyntheticsTestFileMultipartPresignedUrlsParams.attributeTypeMap = {
    key: {
        baseName: "key",
        type: "string",
    },
    uploadId: {
        baseName: "upload_id",
        type: "string",
    },
    urls: {
        baseName: "urls",
        type: "{ [key: string]: string; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
