// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDnsRecord.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDnsRecord = void 0;
/**
 * A DNS record returned in a DNS test response.
 */
class SyntheticsTestResultDnsRecord {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDnsRecord.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDnsRecord = SyntheticsTestResultDnsRecord;
/**
 * @ignore
 */
SyntheticsTestResultDnsRecord.attributeTypeMap = {
    type: {
        baseName: "type",
        type: "string",
    },
    values: {
        baseName: "values",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
