// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultGitCommit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultGitCommit = void 0;
/**
 * Details of the Git commit associated with the test result.
 */
class SyntheticsTestResultGitCommit {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultGitCommit.attributeTypeMap;
    }
}
exports.SyntheticsTestResultGitCommit = SyntheticsTestResultGitCommit;
/**
 * @ignore
 */
SyntheticsTestResultGitCommit.attributeTypeMap = {
    author: {
        baseName: "author",
        type: "SyntheticsTestResultGitUser",
    },
    committer: {
        baseName: "committer",
        type: "SyntheticsTestResultGitUser",
    },
    message: {
        baseName: "message",
        type: "string",
    },
    sha: {
        baseName: "sha",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
