// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultNetstatsHops.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultNetstatsHops = void 0;
/**
 * Statistics about the number of hops for a network test.
 */
class SyntheticsTestResultNetstatsHops {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultNetstatsHops.attributeTypeMap;
    }
}
exports.SyntheticsTestResultNetstatsHops = SyntheticsTestResultNetstatsHops;
/**
 * @ignore
 */
SyntheticsTestResultNetstatsHops.attributeTypeMap = {
    avg: {
        baseName: "avg",
        type: "number",
        format: "double",
    },
    max: {
        baseName: "max",
        type: "number",
        format: "int64",
    },
    min: {
        baseName: "min",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
