// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileDownloadResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileDownloadResponse = void 0;
/**
 * Response containing a presigned URL for downloading a test file.
 */
class SyntheticsTestFileDownloadResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileDownloadResponse.attributeTypeMap;
    }
}
exports.SyntheticsTestFileDownloadResponse = SyntheticsTestFileDownloadResponse;
/**
 * @ignore
 */
SyntheticsTestFileDownloadResponse.attributeTypeMap = {
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
