// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDuration.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDuration = void 0;
/**
 * Total duration of a Synthetic test execution.
 */
class SyntheticsTestResultDuration {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDuration.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDuration = SyntheticsTestResultDuration;
/**
 * @ignore
 */
SyntheticsTestResultDuration.attributeTypeMap = {
    hasDuration: {
        baseName: "has_duration",
        type: "boolean",
    },
    value: {
        baseName: "value",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
