// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultExecutionInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultExecutionInfo = void 0;
/**
 * Execution details for a Synthetic test result.
 */
class SyntheticsTestResultExecutionInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultExecutionInfo.attributeTypeMap;
    }
}
exports.SyntheticsTestResultExecutionInfo = SyntheticsTestResultExecutionInfo;
/**
 * @ignore
 */
SyntheticsTestResultExecutionInfo.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "SyntheticsTestResultDuration",
    },
    errorMessage: {
        baseName: "error_message",
        type: "string",
    },
    isFastRetry: {
        baseName: "is_fast_retry",
        type: "boolean",
    },
    timings: {
        baseName: "timings",
        type: "{ [key: string]: any; }",
    },
    tunnel: {
        baseName: "tunnel",
        type: "boolean",
    },
    unhealthy: {
        baseName: "unhealthy",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
