// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDevice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDevice = void 0;
/**
 * Device information for the test result (browser and mobile tests).
 */
class SyntheticsTestResultDevice {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDevice.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDevice = SyntheticsTestResultDevice;
/**
 * @ignore
 */
SyntheticsTestResultDevice.attributeTypeMap = {
    browser: {
        baseName: "browser",
        type: "SyntheticsTestResultDeviceBrowser",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    platform: {
        baseName: "platform",
        type: "SyntheticsTestResultDevicePlatform",
    },
    resolution: {
        baseName: "resolution",
        type: "SyntheticsTestResultDeviceResolution",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
