// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionDiffPatches.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionDiffPatches = void 0;
/**
 * Object describing a patch in the diff.
 */
class SyntheticsTestVersionDiffPatches {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionDiffPatches.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionDiffPatches = SyntheticsTestVersionDiffPatches;
/**
 * @ignore
 */
SyntheticsTestVersionDiffPatches.attributeTypeMap = {
    diffs: {
        baseName: "diffs",
        type: "Array<SyntheticsTestVersionDiffPatchDiff>",
    },
    length1: {
        baseName: "length1",
        type: "number",
        format: "int64",
    },
    length2: {
        baseName: "length2",
        type: "number",
        format: "int64",
    },
    start1: {
        baseName: "start1",
        type: "number",
        format: "int64",
    },
    start2: {
        baseName: "start2",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
