// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultCdnCacheStatus.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultCdnCacheStatus = void 0;
/**
 * Cache status reported by the CDN for the response.
 */
class SyntheticsTestResultCdnCacheStatus {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultCdnCacheStatus.attributeTypeMap;
    }
}
exports.SyntheticsTestResultCdnCacheStatus = SyntheticsTestResultCdnCacheStatus;
/**
 * @ignore
 */
SyntheticsTestResultCdnCacheStatus.attributeTypeMap = {
    cached: {
        baseName: "cached",
        type: "boolean",
    },
    status: {
        baseName: "status",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
