// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionActionMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionActionMetadata = void 0;
/**
 * Object containing metadata about a change action.
 */
class SyntheticsTestVersionActionMetadata {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionActionMetadata.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionActionMetadata = SyntheticsTestVersionActionMetadata;
/**
 * @ignore
 */
SyntheticsTestVersionActionMetadata.attributeTypeMap = {
    afterValue: {
        baseName: "after_value",
        type: "any",
    },
    beforeValue: {
        baseName: "before_value",
        type: "any",
    },
    diffPatches: {
        baseName: "diff_patches",
        type: "Array<SyntheticsTestVersionDiffPatches>",
    },
    propertyPath: {
        baseName: "property_path",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
