// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultResponseInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultResponseInfo = void 0;
/**
 * Details of the response received during the test execution.
 */
class SyntheticsTestResultResponseInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultResponseInfo.attributeTypeMap;
    }
}
exports.SyntheticsTestResultResponseInfo = SyntheticsTestResultResponseInfo;
/**
 * @ignore
 */
SyntheticsTestResultResponseInfo.attributeTypeMap = {
    body: {
        baseName: "body",
        type: "string",
    },
    bodyCompressed: {
        baseName: "body_compressed",
        type: "string",
    },
    bodyHashes: {
        baseName: "body_hashes",
        type: "string",
    },
    bodySize: {
        baseName: "body_size",
        type: "number",
        format: "int64",
    },
    cacheHeaders: {
        baseName: "cache_headers",
        type: "{ [key: string]: string; }",
    },
    cdn: {
        baseName: "cdn",
        type: "SyntheticsTestResultCdnProviderInfo",
    },
    close: {
        baseName: "close",
        type: "SyntheticsTestResultWebSocketClose",
    },
    compressedMessage: {
        baseName: "compressed_message",
        type: "string",
    },
    headers: {
        baseName: "headers",
        type: "{ [key: string]: any; }",
    },
    healthcheck: {
        baseName: "healthcheck",
        type: "SyntheticsTestResultHealthCheck",
    },
    httpVersion: {
        baseName: "http_version",
        type: "string",
    },
    isBodyTruncated: {
        baseName: "is_body_truncated",
        type: "boolean",
    },
    isMessageTruncated: {
        baseName: "is_message_truncated",
        type: "boolean",
    },
    message: {
        baseName: "message",
        type: "string",
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: string; }",
    },
    records: {
        baseName: "records",
        type: "Array<SyntheticsTestResultDnsRecord>",
    },
    redirects: {
        baseName: "redirects",
        type: "Array<SyntheticsTestResultRedirect>",
    },
    statusCode: {
        baseName: "status_code",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
