// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDetail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDetail = void 0;
/**
 * Full result details for a Synthetic test execution.
 */
class SyntheticsTestResultDetail {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDetail.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDetail = SyntheticsTestResultDetail;
/**
 * @ignore
 */
SyntheticsTestResultDetail.attributeTypeMap = {
    assertions: {
        baseName: "assertions",
        type: "Array<SyntheticsTestResultAssertionResult>",
    },
    bucketKeys: {
        baseName: "bucket_keys",
        type: "SyntheticsTestResultBucketKeys",
    },
    callType: {
        baseName: "call_type",
        type: "string",
    },
    cert: {
        baseName: "cert",
        type: "SyntheticsTestResultCertificate",
    },
    compressedJsonDescriptor: {
        baseName: "compressed_json_descriptor",
        type: "string",
    },
    compressedSteps: {
        baseName: "compressed_steps",
        type: "string",
    },
    connectionOutcome: {
        baseName: "connection_outcome",
        type: "string",
    },
    dnsResolution: {
        baseName: "dns_resolution",
        type: "SyntheticsTestResultDnsResolution",
    },
    duration: {
        baseName: "duration",
        type: "number",
        format: "double",
    },
    exitedOnStepSuccess: {
        baseName: "exited_on_step_success",
        type: "boolean",
    },
    failure: {
        baseName: "failure",
        type: "SyntheticsTestResultFailure",
    },
    finishedAt: {
        baseName: "finished_at",
        type: "number",
        format: "int64",
    },
    handshake: {
        baseName: "handshake",
        type: "SyntheticsTestResultHandshake",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    initialId: {
        baseName: "initial_id",
        type: "string",
    },
    isFastRetry: {
        baseName: "is_fast_retry",
        type: "boolean",
    },
    isLastRetry: {
        baseName: "is_last_retry",
        type: "boolean",
    },
    netpath: {
        baseName: "netpath",
        type: "SyntheticsTestResultNetpath",
    },
    netstats: {
        baseName: "netstats",
        type: "SyntheticsTestResultNetstats",
    },
    ocsp: {
        baseName: "ocsp",
        type: "SyntheticsTestResultOCSPResponse",
    },
    ping: {
        baseName: "ping",
        type: "SyntheticsTestResultTracerouteHop",
    },
    receivedEmailCount: {
        baseName: "received_email_count",
        type: "number",
        format: "int64",
    },
    receivedMessage: {
        baseName: "received_message",
        type: "string",
    },
    request: {
        baseName: "request",
        type: "SyntheticsTestResultRequestInfo",
    },
    resolvedIp: {
        baseName: "resolved_ip",
        type: "string",
    },
    response: {
        baseName: "response",
        type: "SyntheticsTestResultResponseInfo",
    },
    runType: {
        baseName: "run_type",
        type: "SyntheticsTestResultRunType",
    },
    sentMessage: {
        baseName: "sent_message",
        type: "string",
    },
    startUrl: {
        baseName: "start_url",
        type: "string",
    },
    startedAt: {
        baseName: "started_at",
        type: "number",
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "SyntheticsTestResultStatus",
    },
    steps: {
        baseName: "steps",
        type: "Array<SyntheticsTestResultStep>",
    },
    timeToInteractive: {
        baseName: "time_to_interactive",
        type: "number",
        format: "int64",
    },
    timings: {
        baseName: "timings",
        type: "{ [key: string]: any; }",
    },
    trace: {
        baseName: "trace",
        type: "SyntheticsTestResultTrace",
    },
    traceroute: {
        baseName: "traceroute",
        type: "Array<SyntheticsTestResultTracerouteHop>",
    },
    triggeredAt: {
        baseName: "triggered_at",
        type: "number",
        format: "int64",
    },
    tunnel: {
        baseName: "tunnel",
        type: "boolean",
    },
    turns: {
        baseName: "turns",
        type: "Array<SyntheticsTestResultTurn>",
    },
    unhealthy: {
        baseName: "unhealthy",
        type: "boolean",
    },
    variables: {
        baseName: "variables",
        type: "SyntheticsTestResultVariables",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
