// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionHistoryMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionHistoryMeta = void 0;
/**
 * Pagination metadata for a version history response.
 */
class SyntheticsTestVersionHistoryMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionHistoryMeta.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionHistoryMeta = SyntheticsTestVersionHistoryMeta;
/**
 * @ignore
 */
SyntheticsTestVersionHistoryMeta.attributeTypeMap = {
    nextLastVersionNumber: {
        baseName: "next_last_version_number",
        type: "number",
        format: "int64",
    },
    retentionPeriodInDays: {
        baseName: "retention_period_in_days",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
