// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestFileMultipartPresignedUrlsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestFileMultipartPresignedUrlsRequest = void 0;
/**
 * Request body for getting presigned URLs for a multipart file upload.
 */
class SyntheticsTestFileMultipartPresignedUrlsRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestFileMultipartPresignedUrlsRequest.attributeTypeMap;
    }
}
exports.SyntheticsTestFileMultipartPresignedUrlsRequest = SyntheticsTestFileMultipartPresignedUrlsRequest;
/**
 * @ignore
 */
SyntheticsTestFileMultipartPresignedUrlsRequest.attributeTypeMap = {
    bucketKeyPrefix: {
        baseName: "bucketKeyPrefix",
        type: "SyntheticsTestFileMultipartPresignedUrlsRequestBucketKeyPrefix",
        required: true,
    },
    parts: {
        baseName: "parts",
        type: "Array<SyntheticsTestFileMultipartPresignedUrlsPart>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
