// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestVersionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestVersionAttributes = void 0;
/**
 * Attributes of a specific Synthetic test version.
 */
class SyntheticsTestVersionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestVersionAttributes.attributeTypeMap;
    }
}
exports.SyntheticsTestVersionAttributes = SyntheticsTestVersionAttributes;
/**
 * @ignore
 */
SyntheticsTestVersionAttributes.attributeTypeMap = {
    author: {
        baseName: "author",
        type: "SyntheticsTestVersionAuthor",
    },
    changeMetadata: {
        baseName: "change_metadata",
        type: "Array<SyntheticsTestVersionChangeMetadataItem>",
    },
    payload: {
        baseName: "payload",
        type: "{ [key: string]: any; }",
    },
    versionPayloadCreatedAt: {
        baseName: "version_payload_created_at",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
