// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsFastTestResultAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsFastTestResultAttributes = void 0;
/**
 * Attributes of the fast test result.
 */
class SyntheticsFastTestResultAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsFastTestResultAttributes.attributeTypeMap;
    }
}
exports.SyntheticsFastTestResultAttributes = SyntheticsFastTestResultAttributes;
/**
 * @ignore
 */
SyntheticsFastTestResultAttributes.attributeTypeMap = {
    device: {
        baseName: "device",
        type: "SyntheticsTestResultDevice",
    },
    location: {
        baseName: "location",
        type: "SyntheticsTestResultLocation",
    },
    result: {
        baseName: "result",
        type: "SyntheticsFastTestResultDetail",
    },
    testSubType: {
        baseName: "test_sub_type",
        type: "SyntheticsFastTestSubType",
    },
    testType: {
        baseName: "test_type",
        type: "SyntheticsFastTestType",
    },
    testVersion: {
        baseName: "test_version",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
