// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultGitUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultGitUser = void 0;
/**
 * A Git user (author or committer).
 */
class SyntheticsTestResultGitUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultGitUser.attributeTypeMap;
    }
}
exports.SyntheticsTestResultGitUser = SyntheticsTestResultGitUser;
/**
 * @ignore
 */
SyntheticsTestResultGitUser.attributeTypeMap = {
    date: {
        baseName: "date",
        type: "string",
    },
    email: {
        baseName: "email",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
