// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultWebSocketClose.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultWebSocketClose = void 0;
/**
 * WebSocket close frame information for WebSocket test responses.
 */
class SyntheticsTestResultWebSocketClose {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultWebSocketClose.attributeTypeMap;
    }
}
exports.SyntheticsTestResultWebSocketClose = SyntheticsTestResultWebSocketClose;
/**
 * @ignore
 */
SyntheticsTestResultWebSocketClose.attributeTypeMap = {
    reason: {
        baseName: "reason",
        type: "string",
    },
    statusCode: {
        baseName: "status_code",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
