// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultStepAssertionResult.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultStepAssertionResult = void 0;
/**
 * Assertion result for a browser or mobile step.
 */
class SyntheticsTestResultStepAssertionResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultStepAssertionResult.attributeTypeMap;
    }
}
exports.SyntheticsTestResultStepAssertionResult = SyntheticsTestResultStepAssertionResult;
/**
 * @ignore
 */
SyntheticsTestResultStepAssertionResult.attributeTypeMap = {
    actual: {
        baseName: "actual",
        type: "any",
    },
    checkType: {
        baseName: "check_type",
        type: "string",
    },
    expected: {
        baseName: "expected",
        type: "any",
    },
    hasSecureVariables: {
        baseName: "has_secure_variables",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
