// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultDeviceResolution.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultDeviceResolution = void 0;
/**
 * Screen resolution of the device used to run the test.
 */
class SyntheticsTestResultDeviceResolution {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultDeviceResolution.attributeTypeMap;
    }
}
exports.SyntheticsTestResultDeviceResolution = SyntheticsTestResultDeviceResolution;
/**
 * @ignore
 */
SyntheticsTestResultDeviceResolution.attributeTypeMap = {
    height: {
        baseName: "height",
        type: "number",
        format: "int64",
    },
    pixelRatio: {
        baseName: "pixel_ratio",
        type: "number",
        format: "double",
    },
    width: {
        baseName: "width",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
