// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultBucketKeys.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultBucketKeys = void 0;
/**
 * Storage bucket keys for artifacts produced during a step or test.
 */
class SyntheticsTestResultBucketKeys {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultBucketKeys.attributeTypeMap;
    }
}
exports.SyntheticsTestResultBucketKeys = SyntheticsTestResultBucketKeys;
/**
 * @ignore
 */
SyntheticsTestResultBucketKeys.attributeTypeMap = {
    afterStepScreenshot: {
        baseName: "after_step_screenshot",
        type: "string",
    },
    afterTurnScreenshot: {
        baseName: "after_turn_screenshot",
        type: "string",
    },
    artifacts: {
        baseName: "artifacts",
        type: "string",
    },
    beforeStepScreenshot: {
        baseName: "before_step_screenshot",
        type: "string",
    },
    beforeTurnScreenshot: {
        baseName: "before_turn_screenshot",
        type: "string",
    },
    crashReport: {
        baseName: "crash_report",
        type: "string",
    },
    deviceLogs: {
        baseName: "device_logs",
        type: "string",
    },
    emailMessages: {
        baseName: "email_messages",
        type: "Array<string>",
    },
    screenshot: {
        baseName: "screenshot",
        type: "string",
    },
    snapshot: {
        baseName: "snapshot",
        type: "string",
    },
    source: {
        baseName: "source",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
