// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultNetpath.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultNetpath = void 0;
/**
 * Network Path test result capturing the path between source and destination.
 */
class SyntheticsTestResultNetpath {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultNetpath.attributeTypeMap;
    }
}
exports.SyntheticsTestResultNetpath = SyntheticsTestResultNetpath;
/**
 * @ignore
 */
SyntheticsTestResultNetpath.attributeTypeMap = {
    destination: {
        baseName: "destination",
        type: "SyntheticsTestResultNetpathDestination",
    },
    hops: {
        baseName: "hops",
        type: "Array<SyntheticsTestResultNetpathHop>",
    },
    origin: {
        baseName: "origin",
        type: "string",
    },
    pathtraceId: {
        baseName: "pathtrace_id",
        type: "string",
    },
    protocol: {
        baseName: "protocol",
        type: "string",
    },
    source: {
        baseName: "source",
        type: "SyntheticsTestResultNetpathEndpoint",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    timestamp: {
        baseName: "timestamp",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
