// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultCertificate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultCertificate = void 0;
/**
 * SSL/TLS certificate information returned from an SSL test.
 */
class SyntheticsTestResultCertificate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultCertificate.attributeTypeMap;
    }
}
exports.SyntheticsTestResultCertificate = SyntheticsTestResultCertificate;
/**
 * @ignore
 */
SyntheticsTestResultCertificate.attributeTypeMap = {
    cipher: {
        baseName: "cipher",
        type: "string",
    },
    exponent: {
        baseName: "exponent",
        type: "number",
        format: "int64",
    },
    extKeyUsage: {
        baseName: "ext_key_usage",
        type: "Array<string>",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
    },
    fingerprint256: {
        baseName: "fingerprint256",
        type: "string",
    },
    issuer: {
        baseName: "issuer",
        type: "{ [key: string]: string; }",
    },
    modulus: {
        baseName: "modulus",
        type: "string",
    },
    protocol: {
        baseName: "protocol",
        type: "string",
    },
    serialNumber: {
        baseName: "serial_number",
        type: "string",
    },
    subject: {
        baseName: "subject",
        type: "{ [key: string]: string; }",
    },
    tlsVersion: {
        baseName: "tls_version",
        type: "number",
        format: "double",
    },
    valid: {
        baseName: "valid",
        type: "SyntheticsTestResultCertificateValidity",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
