// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultNetworkLatency.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultNetworkLatency = void 0;
/**
 * Latency statistics for a network probe.
 */
class SyntheticsTestResultNetworkLatency {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultNetworkLatency.attributeTypeMap;
    }
}
exports.SyntheticsTestResultNetworkLatency = SyntheticsTestResultNetworkLatency;
/**
 * @ignore
 */
SyntheticsTestResultNetworkLatency.attributeTypeMap = {
    avg: {
        baseName: "avg",
        type: "number",
        format: "double",
    },
    max: {
        baseName: "max",
        type: "number",
        format: "double",
    },
    min: {
        baseName: "min",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
