// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultRelationshipTestData.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultRelationshipTestData = void 0;
/**
 * Data for the test relationship.
 */
class SyntheticsTestResultRelationshipTestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultRelationshipTestData.attributeTypeMap;
    }
}
exports.SyntheticsTestResultRelationshipTestData = SyntheticsTestResultRelationshipTestData;
/**
 * @ignore
 */
SyntheticsTestResultRelationshipTestData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
