// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultRequestInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultRequestInfo = void 0;
/**
 * Details of the outgoing request made during the test execution.
 */
class SyntheticsTestResultRequestInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultRequestInfo.attributeTypeMap;
    }
}
exports.SyntheticsTestResultRequestInfo = SyntheticsTestResultRequestInfo;
/**
 * @ignore
 */
SyntheticsTestResultRequestInfo.attributeTypeMap = {
    allowInsecure: {
        baseName: "allow_insecure",
        type: "boolean",
    },
    body: {
        baseName: "body",
        type: "string",
    },
    callType: {
        baseName: "call_type",
        type: "string",
    },
    destinationService: {
        baseName: "destination_service",
        type: "string",
    },
    dnsServer: {
        baseName: "dns_server",
        type: "string",
    },
    dnsServerPort: {
        baseName: "dns_server_port",
        type: "number",
        format: "int64",
    },
    e2eQueries: {
        baseName: "e2e_queries",
        type: "number",
        format: "int64",
    },
    files: {
        baseName: "files",
        type: "Array<SyntheticsTestResultFileRef>",
    },
    headers: {
        baseName: "headers",
        type: "{ [key: string]: any; }",
    },
    host: {
        baseName: "host",
        type: "string",
    },
    maxTtl: {
        baseName: "max_ttl",
        type: "number",
        format: "int64",
    },
    message: {
        baseName: "message",
        type: "string",
    },
    method: {
        baseName: "method",
        type: "string",
    },
    noSavingResponseBody: {
        baseName: "no_saving_response_body",
        type: "boolean",
    },
    port: {
        baseName: "port",
        type: "any",
    },
    service: {
        baseName: "service",
        type: "string",
    },
    sourceService: {
        baseName: "source_service",
        type: "string",
    },
    timeout: {
        baseName: "timeout",
        type: "number",
        format: "int64",
    },
    toolName: {
        baseName: "tool_name",
        type: "string",
    },
    tracerouteQueries: {
        baseName: "traceroute_queries",
        type: "number",
        format: "int64",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
