// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsTestResultBrowserError.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsTestResultBrowserError = void 0;
/**
 * A browser error captured during a browser test step.
 */
class SyntheticsTestResultBrowserError {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsTestResultBrowserError.attributeTypeMap;
    }
}
exports.SyntheticsTestResultBrowserError = SyntheticsTestResultBrowserError;
/**
 * @ignore
 */
SyntheticsTestResultBrowserError.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    method: {
        baseName: "method",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "number",
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "{ [key: string]: any; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
