// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TargetingRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetingRule = void 0;
/**
 * Targeting rule details.
 */
class TargetingRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TargetingRule.attributeTypeMap;
    }
}
exports.TargetingRule = TargetingRule;
/**
 * @ignore
 */
TargetingRule.attributeTypeMap = {
    conditions: {
        baseName: "conditions",
        type: "Array<Condition>",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
