// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TeamsField.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.USER_TEAM_PERMISSIONS = exports.TEAM_LINKS = exports.LINK_COUNT = exports.USER_COUNT = exports.MODIFIED_AT = exports.CREATED_AT = exports.HIDDEN_MODULES = exports.VISIBLE_MODULES = exports.BANNER = exports.AVATAR = exports.DESCRIPTION = exports.SUMMARY = exports.HANDLE = exports.NAME = exports.ID = void 0;
exports.ID = "id";
exports.NAME = "name";
exports.HANDLE = "handle";
exports.SUMMARY = "summary";
exports.DESCRIPTION = "description";
exports.AVATAR = "avatar";
exports.BANNER = "banner";
exports.VISIBLE_MODULES = "visible_modules";
exports.HIDDEN_MODULES = "hidden_modules";
exports.CREATED_AT = "created_at";
exports.MODIFIED_AT = "modified_at";
exports.USER_COUNT = "user_count";
exports.LINK_COUNT = "link_count";
exports.TEAM_LINKS = "team_links";
exports.USER_TEAM_PERMISSIONS = "user_team_permissions";
