// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TeamSyncAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamSyncAttributes = void 0;
/**
 * Team sync attributes.
 */
class TeamSyncAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TeamSyncAttributes.attributeTypeMap;
    }
}
exports.TeamSyncAttributes = TeamSyncAttributes;
/**
 * @ignore
 */
TeamSyncAttributes.attributeTypeMap = {
    frequency: {
        baseName: "frequency",
        type: "TeamSyncAttributesFrequency",
    },
    selectionState: {
        baseName: "selection_state",
        type: "Array<TeamSyncSelectionStateItem>",
    },
    source: {
        baseName: "source",
        type: "TeamSyncAttributesSource",
        required: true,
    },
    syncMembership: {
        baseName: "sync_membership",
        type: "boolean",
    },
    type: {
        baseName: "type",
        type: "TeamSyncAttributesType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
