// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesDisabled.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesDisabled = void 0;
/**
 * Configuration for the disabled Flaky Tests Management policy.
 */
class TestOptimizationFlakyTestsManagementPoliciesDisabled {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesDisabled.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesDisabled = TestOptimizationFlakyTestsManagementPoliciesDisabled;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesDisabled.attributeTypeMap = {
    autoDisableRule: {
        baseName: "auto_disable_rule",
        type: "TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule",
    },
    branchRule: {
        baseName: "branch_rule",
        type: "TestOptimizationFlakyTestsManagementPoliciesBranchRule",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    failureRateRule: {
        baseName: "failure_rate_rule",
        type: "TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
