// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesBranchRule.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesBranchRule = void 0;
/**
 * Branch filtering rule for a Flaky Tests Management policy.
 */
class TestOptimizationFlakyTestsManagementPoliciesBranchRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesBranchRule.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesBranchRule = TestOptimizationFlakyTestsManagementPoliciesBranchRule;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesBranchRule.attributeTypeMap = {
    branches: {
        baseName: "branches",
        type: "Array<string>",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    excludedBranches: {
        baseName: "excluded_branches",
        type: "Array<string>",
    },
    excludedTestServices: {
        baseName: "excluded_test_services",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
