// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationDeleteServiceSettingsRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationDeleteServiceSettingsRequestAttributes = void 0;
/**
 * Attributes for deleting Test Optimization service settings.
 */
class TestOptimizationDeleteServiceSettingsRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationDeleteServiceSettingsRequestAttributes.attributeTypeMap;
    }
}
exports.TestOptimizationDeleteServiceSettingsRequestAttributes = TestOptimizationDeleteServiceSettingsRequestAttributes;
/**
 * @ignore
 */
TestOptimizationDeleteServiceSettingsRequestAttributes.attributeTypeMap = {
    env: {
        baseName: "env",
        type: "string",
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
        required: true,
    },
    serviceName: {
        baseName: "service_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
