// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationGetServiceSettingsRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationGetServiceSettingsRequestData = void 0;
/**
 * Data object for get service settings request.
 */
class TestOptimizationGetServiceSettingsRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationGetServiceSettingsRequestData.attributeTypeMap;
    }
}
exports.TestOptimizationGetServiceSettingsRequestData = TestOptimizationGetServiceSettingsRequestData;
/**
 * @ignore
 */
TestOptimizationGetServiceSettingsRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TestOptimizationGetServiceSettingsRequestAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "TestOptimizationGetServiceSettingsRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
