// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesAttemptToFix.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesAttemptToFix = void 0;
/**
 * Configuration for the attempt-to-fix Flaky Tests Management policy.
 */
class TestOptimizationFlakyTestsManagementPoliciesAttemptToFix {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesAttemptToFix.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesAttemptToFix = TestOptimizationFlakyTestsManagementPoliciesAttemptToFix;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesAttemptToFix.attributeTypeMap = {
    retries: {
        baseName: "retries",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
