// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationServiceSettingsAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationServiceSettingsAttributes = void 0;
/**
 * Attributes for Test Optimization service settings.
 */
class TestOptimizationServiceSettingsAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationServiceSettingsAttributes.attributeTypeMap;
    }
}
exports.TestOptimizationServiceSettingsAttributes = TestOptimizationServiceSettingsAttributes;
/**
 * @ignore
 */
TestOptimizationServiceSettingsAttributes.attributeTypeMap = {
    autoTestRetriesEnabled: {
        baseName: "auto_test_retries_enabled",
        type: "boolean",
    },
    codeCoverageEnabled: {
        baseName: "code_coverage_enabled",
        type: "boolean",
    },
    earlyFlakeDetectionEnabled: {
        baseName: "early_flake_detection_enabled",
        type: "boolean",
    },
    env: {
        baseName: "env",
        type: "string",
    },
    failedTestReplayEnabled: {
        baseName: "failed_test_replay_enabled",
        type: "boolean",
    },
    prCommentsEnabled: {
        baseName: "pr_comments_enabled",
        type: "boolean",
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
    },
    serviceName: {
        baseName: "service_name",
        type: "string",
    },
    testImpactAnalysisEnabled: {
        baseName: "test_impact_analysis_enabled",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
