// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesAttributes = void 0;
/**
 * Attributes of the Flaky Tests Management policies for a repository.
 */
class TestOptimizationFlakyTestsManagementPoliciesAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesAttributes.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesAttributes = TestOptimizationFlakyTestsManagementPoliciesAttributes;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesAttributes.attributeTypeMap = {
    attemptToFix: {
        baseName: "attempt_to_fix",
        type: "TestOptimizationFlakyTestsManagementPoliciesAttemptToFix",
    },
    disabled: {
        baseName: "disabled",
        type: "TestOptimizationFlakyTestsManagementPoliciesDisabled",
    },
    quarantined: {
        baseName: "quarantined",
        type: "TestOptimizationFlakyTestsManagementPoliciesQuarantined",
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
