// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationDeleteServiceSettingsRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationDeleteServiceSettingsRequestData = void 0;
/**
 * Data object for delete service settings request.
 */
class TestOptimizationDeleteServiceSettingsRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationDeleteServiceSettingsRequestData.attributeTypeMap;
    }
}
exports.TestOptimizationDeleteServiceSettingsRequestData = TestOptimizationDeleteServiceSettingsRequestData;
/**
 * @ignore
 */
TestOptimizationDeleteServiceSettingsRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TestOptimizationDeleteServiceSettingsRequestAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "TestOptimizationDeleteServiceSettingsRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
