// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationUpdateServiceSettingsRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationUpdateServiceSettingsRequestData = void 0;
/**
 * Data object for update service settings request.
 */
class TestOptimizationUpdateServiceSettingsRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationUpdateServiceSettingsRequestData.attributeTypeMap;
    }
}
exports.TestOptimizationUpdateServiceSettingsRequestData = TestOptimizationUpdateServiceSettingsRequestData;
/**
 * @ignore
 */
TestOptimizationUpdateServiceSettingsRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TestOptimizationUpdateServiceSettingsRequestAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "TestOptimizationUpdateServiceSettingsRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
