// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesQuarantined.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesQuarantined = void 0;
/**
 * Configuration for the quarantined Flaky Tests Management policy.
 */
class TestOptimizationFlakyTestsManagementPoliciesQuarantined {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesQuarantined.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesQuarantined = TestOptimizationFlakyTestsManagementPoliciesQuarantined;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesQuarantined.attributeTypeMap = {
    autoQuarantineRule: {
        baseName: "auto_quarantine_rule",
        type: "TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule",
    },
    branchRule: {
        baseName: "branch_rule",
        type: "TestOptimizationFlakyTestsManagementPoliciesBranchRule",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    failureRateRule: {
        baseName: "failure_rate_rule",
        type: "TestOptimizationFlakyTestsManagementPoliciesQuarantinedFailureRateRule",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
