// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes = void 0;
/**
 * Attributes for requesting Flaky Tests Management policies.
 */
class TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes = TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes.attributeTypeMap = {
    repositoryId: {
        baseName: "repository_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
