// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData = void 0;
/**
 * Data object for update Flaky Tests Management policies request.
 */
class TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData = TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesUpdateRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "TestOptimizationUpdateFlakyTestsManagementPoliciesRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
