// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesData = void 0;
/**
 * Data object for Flaky Tests Management policies response.
 */
class TestOptimizationFlakyTestsManagementPoliciesData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesData.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesData = TestOptimizationFlakyTestsManagementPoliciesData;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TestOptimizationFlakyTestsManagementPoliciesAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "TestOptimizationFlakyTestsManagementPoliciesType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
