// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes = void 0;
/**
 * Attributes for updating Flaky Tests Management policies.
 * Only provided policy blocks are updated; omitted blocks are left unchanged.
 */
class TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes = TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesUpdateRequestAttributes.attributeTypeMap = {
    attemptToFix: {
        baseName: "attempt_to_fix",
        type: "TestOptimizationFlakyTestsManagementPoliciesAttemptToFix",
    },
    disabled: {
        baseName: "disabled",
        type: "TestOptimizationFlakyTestsManagementPoliciesDisabled",
    },
    quarantined: {
        baseName: "quarantined",
        type: "TestOptimizationFlakyTestsManagementPoliciesQuarantined",
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
