// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule = void 0;
/**
 * Automatic quarantine triggering rule based on a time window.
 */
class TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule = TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesAutoQuarantineRule.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    windowSeconds: {
        baseName: "window_seconds",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
