// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule = void 0;
/**
 * Failure-rate-based rule for the disabled policy.
 */
class TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule = TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesDisabledFailureRateRule.attributeTypeMap = {
    branches: {
        baseName: "branches",
        type: "Array<string>",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    minRuns: {
        baseName: "min_runs",
        type: "number",
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "TestOptimizationFlakyTestsManagementPoliciesDisabledStatus",
    },
    threshold: {
        baseName: "threshold",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
