// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesGetRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesGetRequestData = void 0;
/**
 * Data object for get Flaky Tests Management policies request.
 */
class TestOptimizationFlakyTestsManagementPoliciesGetRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesGetRequestData.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesGetRequestData = TestOptimizationFlakyTestsManagementPoliciesGetRequestData;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesGetRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TestOptimizationFlakyTestsManagementPoliciesGetRequestAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "TestOptimizationGetFlakyTestsManagementPoliciesRequestDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
