// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule = void 0;
/**
 * Automatic disable triggering rule based on a time window and test status.
 */
class TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule.attributeTypeMap;
    }
}
exports.TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule = TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule;
/**
 * @ignore
 */
TestOptimizationFlakyTestsManagementPoliciesAutoDisableRule.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    status: {
        baseName: "status",
        type: "TestOptimizationFlakyTestsManagementPoliciesDisabledStatus",
    },
    windowSeconds: {
        baseName: "window_seconds",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
