// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TriggerInvestigationResponseDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerInvestigationResponseDataAttributes = void 0;
/**
 * Attributes for the trigger investigation response.
 */
class TriggerInvestigationResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TriggerInvestigationResponseDataAttributes.attributeTypeMap;
    }
}
exports.TriggerInvestigationResponseDataAttributes = TriggerInvestigationResponseDataAttributes;
/**
 * @ignore
 */
TriggerInvestigationResponseDataAttributes.attributeTypeMap = {
    investigationId: {
        baseName: "investigation_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
