// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TriggerInvestigationResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerInvestigationResponseData = void 0;
/**
 * Data for the trigger investigation response.
 */
class TriggerInvestigationResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TriggerInvestigationResponseData.attributeTypeMap;
    }
}
exports.TriggerInvestigationResponseData = TriggerInvestigationResponseData;
/**
 * @ignore
 */
TriggerInvestigationResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TriggerInvestigationResponseDataAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "TriggerInvestigationResponseType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
