// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpdateCampaignRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateCampaignRequestAttributes = void 0;
/**
 * Attributes for updating a campaign.
 */
class UpdateCampaignRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpdateCampaignRequestAttributes.attributeTypeMap;
    }
}
exports.UpdateCampaignRequestAttributes = UpdateCampaignRequestAttributes;
/**
 * @ignore
 */
UpdateCampaignRequestAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    dueDate: {
        baseName: "due_date",
        type: "Date",
        format: "date-time",
    },
    entityScope: {
        baseName: "entity_scope",
        type: "string",
    },
    guidance: {
        baseName: "guidance",
        type: "string",
    },
    key: {
        baseName: "key",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ownerId: {
        baseName: "owner_id",
        type: "string",
        required: true,
    },
    ruleIds: {
        baseName: "rule_ids",
        type: "Array<string>",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
