// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpdateTenancyConfigData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTenancyConfigData = void 0;
/**
 * The data object for updating an existing OCI tenancy integration configuration, including the tenancy ID, type, and updated attributes.
 */
class UpdateTenancyConfigData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpdateTenancyConfigData.attributeTypeMap;
    }
}
exports.UpdateTenancyConfigData = UpdateTenancyConfigData;
/**
 * @ignore
 */
UpdateTenancyConfigData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "UpdateTenancyConfigDataAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "UpdateTenancyConfigDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
