// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpsertAllocationRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertAllocationRequest = void 0;
/**
 * Request to create or update a targeting rule (allocation) for a feature flag environment.
 */
class UpsertAllocationRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpsertAllocationRequest.attributeTypeMap;
    }
}
exports.UpsertAllocationRequest = UpsertAllocationRequest;
/**
 * @ignore
 */
UpsertAllocationRequest.attributeTypeMap = {
    experimentId: {
        baseName: "experiment_id",
        type: "string",
    },
    exposureSchedule: {
        baseName: "exposure_schedule",
        type: "ExposureScheduleRequest",
    },
    guardrailMetrics: {
        baseName: "guardrail_metrics",
        type: "Array<GuardrailMetricRequest>",
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    targetingRules: {
        baseName: "targeting_rules",
        type: "Array<TargetingRuleRequest>",
    },
    type: {
        baseName: "type",
        type: "AllocationType",
        required: true,
    },
    variantWeights: {
        baseName: "variant_weights",
        type: "Array<VariantWeightRequest>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
