// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UserUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserUpdateAttributes = void 0;
/**
 * Attributes of the edited user.
 */
class UserUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UserUpdateAttributes.attributeTypeMap;
    }
}
exports.UserUpdateAttributes = UserUpdateAttributes;
/**
 * @ignore
 */
UserUpdateAttributes.attributeTypeMap = {
    disabled: {
        baseName: "disabled",
        type: "boolean",
    },
    email: {
        baseName: "email",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
