// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/VariantWeightRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VariantWeightRequest = void 0;
/**
 * Variant weight request payload.
 */
class VariantWeightRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return VariantWeightRequest.attributeTypeMap;
    }
}
exports.VariantWeightRequest = VariantWeightRequest;
/**
 * @ignore
 */
VariantWeightRequest.attributeTypeMap = {
    value: {
        baseName: "value",
        type: "number",
        required: true,
        format: "double",
    },
    variantId: {
        baseName: "variant_id",
        type: "string",
        format: "uuid",
    },
    variantKey: {
        baseName: "variant_key",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
