// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/VariantWeight.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VariantWeight = void 0;
/**
 * Variant weight details.
 */
class VariantWeight {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return VariantWeight.attributeTypeMap;
    }
}
exports.VariantWeight = VariantWeight;
/**
 * @ignore
 */
VariantWeight.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        format: "date-time",
    },
    value: {
        baseName: "value",
        type: "number",
        required: true,
        format: "double",
    },
    variant: {
        baseName: "variant",
        type: "Variant",
    },
    variantId: {
        baseName: "variant_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
