// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WidgetType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TREEMAP = exports.TOPLIST = exports.TIMESERIES = exports.SUNBURST = exports.SANKEY = exports.RETENTION_CURVE = exports.QUERY_VALUE = exports.QUERY_TABLE = exports.LIST_STREAM = exports.GEOMAP = exports.FUNNEL = exports.COHORT = exports.CLOUD_COST_SUMMARY = exports.CHANGE = exports.BAR_CHART = void 0;
exports.BAR_CHART = "bar_chart";
exports.CHANGE = "change";
exports.CLOUD_COST_SUMMARY = "cloud_cost_summary";
exports.COHORT = "cohort";
exports.FUNNEL = "funnel";
exports.GEOMAP = "geomap";
exports.LIST_STREAM = "list_stream";
exports.QUERY_TABLE = "query_table";
exports.QUERY_VALUE = "query_value";
exports.RETENTION_CURVE = "retention_curve";
exports.SANKEY = "sankey";
exports.SUNBURST = "sunburst";
exports.TIMESERIES = "timeseries";
exports.TOPLIST = "toplist";
exports.TREEMAP = "treemap";
