// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WidgetSearchMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidgetSearchMeta = void 0;
/**
 * Metadata about the search results.
 */
class WidgetSearchMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WidgetSearchMeta.attributeTypeMap;
    }
}
exports.WidgetSearchMeta = WidgetSearchMeta;
/**
 * @ignore
 */
WidgetSearchMeta.attributeTypeMap = {
    createdByAnyoneTotal: {
        baseName: "created_by_anyone_total",
        type: "number",
        format: "int64",
    },
    createdByYouTotal: {
        baseName: "created_by_you_total",
        type: "number",
        format: "int64",
    },
    favoritedByYouTotal: {
        baseName: "favorited_by_you_total",
        type: "number",
        format: "int64",
    },
    filteredTotal: {
        baseName: "filtered_total",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
