// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WidgetAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidgetAttributes = void 0;
/**
 * Attributes of a widget resource.
 */
class WidgetAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WidgetAttributes.attributeTypeMap;
    }
}
exports.WidgetAttributes = WidgetAttributes;
/**
 * @ignore
 */
WidgetAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "string",
        required: true,
    },
    definition: {
        baseName: "definition",
        type: "WidgetDefinition",
        required: true,
    },
    isFavorited: {
        baseName: "is_favorited",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
