// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WidgetRelationshipData.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WidgetRelationshipData = void 0;
/**
 * Relationship data referencing a user resource.
 */
class WidgetRelationshipData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WidgetRelationshipData.attributeTypeMap;
    }
}
exports.WidgetRelationshipData = WidgetRelationshipData;
/**
 * @ignore
 */
WidgetRelationshipData.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
